<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.336">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Damian Oswald">
<meta name="dcterms.date" content="2023-04-28">

<title>Remote Sensing of Foliar Nitrogen in Californian Almonds</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="MasterThesis_files/libs/clipboard/clipboard.min.js"></script>
<script src="MasterThesis_files/libs/quarto-html/quarto.js"></script>
<script src="MasterThesis_files/libs/quarto-html/popper.min.js"></script>
<script src="MasterThesis_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="MasterThesis_files/libs/quarto-html/anchor.min.js"></script>
<link href="MasterThesis_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="MasterThesis_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="MasterThesis_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="MasterThesis_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="MasterThesis_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<style>

      .quarto-title-block .quarto-title-banner {
        background-image: url(resources_html/almonds.png);
background-size: cover;
      }
</style>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body>

<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Remote Sensing of Foliar Nitrogen in Californian Almonds</h1>
                      </div>
  </div>
    
  <div class="quarto-title-meta-author">
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-heading">Affiliation</div>
    
      <div class="quarto-title-meta-contents">
      <p class="author">Damian Oswald </p>
    </div>
      <div class="quarto-title-meta-contents">
          <p class="affiliation">
              ETH Zürich
            </p>
        </div>
      </div>

  <div class="quarto-title-meta">

        
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">April 28, 2023</p>
      </div>
    </div>
    
      
    </div>
    
  <div>
    <div class="abstract">
      <div class="abstract-title">Abstract</div>
      Nitrogen is a crucial plant nutrient, but both its under-availability and over-abundance can harm crop yields as well as the environment and reduce water supply quality. This is particularly important for almond orchards in California, where nitrogen is applied in large quantities due to the crop’s profitability. Remote sensing techniques show promise for quantifying foliar nitrogen concentration in crops, which can inform fertilization decisions and potentially reduce the risks of over-fertilization. In this thesis, I explore and test various remote sensing approaches for estimating the foliar nitrogen concentration in Californian almonds using spectroradiometric measurements at the leaf level and hyperspectral imaging at the canopy level. I evaluate numerous classical vegetation indices, various machine learning models, and two physics-informed radiative transfer models. Radiative transfer modelling, although less investigated in scientific literature, demonstrates potential as both a robust and generalizable technique, yielding results comparable to the best-performing machine learning models.
    </div>
  </div>
  
  </header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#background" id="toc-background" class="nav-link active" data-scroll-target="#background"><span class="header-section-number">1</span> Background</a>
  <ul class="collapse">
  <li><a href="#sec-almonds" id="toc-sec-almonds" class="nav-link" data-scroll-target="#sec-almonds"><span class="header-section-number">1.1</span> Almond production in California</a></li>
  <li><a href="#sec-nitrogen" id="toc-sec-nitrogen" class="nav-link" data-scroll-target="#sec-nitrogen"><span class="header-section-number">1.2</span> Challenges with balancing crop nitrogen</a></li>
  <li><a href="#sec-remotesensing" id="toc-sec-remotesensing" class="nav-link" data-scroll-target="#sec-remotesensing"><span class="header-section-number">1.3</span> The role of remote sensing in precision agriculture</a></li>
  <li><a href="#sec-nitrogenRS" id="toc-sec-nitrogenRS" class="nav-link" data-scroll-target="#sec-nitrogenRS"><span class="header-section-number">1.4</span> Remote sensing of nitrogen</a></li>
  <li><a href="#sec-aims" id="toc-sec-aims" class="nav-link" data-scroll-target="#sec-aims"><span class="header-section-number">1.5</span> Aims and structure of this thesis</a></li>
  </ul></li>
  <li><a href="#sec-VI" id="toc-sec-VI" class="nav-link" data-scroll-target="#sec-VI"><span class="header-section-number">2</span> Vegetation-indices-based methods</a>
  <ul class="collapse">
  <li><a href="#evolution-of-vegetation-indices-and-their-limitations" id="toc-evolution-of-vegetation-indices-and-their-limitations" class="nav-link" data-scroll-target="#evolution-of-vegetation-indices-and-their-limitations"><span class="header-section-number">2.1</span> Evolution of vegetation indices and their limitations</a></li>
  <li><a href="#sec-nitrogenVI" id="toc-sec-nitrogenVI" class="nav-link" data-scroll-target="#sec-nitrogenVI"><span class="header-section-number">2.2</span> Nitrogen monitoring with vegetation indices</a></li>
  </ul></li>
  <li><a href="#sec-ML" id="toc-sec-ML" class="nav-link" data-scroll-target="#sec-ML"><span class="header-section-number">3</span> Machine learning methods for hyperspectral data analysis</a>
  <ul class="collapse">
  <li><a href="#sec-stepwise" id="toc-sec-stepwise" class="nav-link" data-scroll-target="#sec-stepwise"><span class="header-section-number">3.1</span> Ordinary least squares and stepwise selection</a></li>
  <li><a href="#sec-regularization" id="toc-sec-regularization" class="nav-link" data-scroll-target="#sec-regularization"><span class="header-section-number">3.2</span> Regularization</a></li>
  <li><a href="#sec-dimensionalityreduction" id="toc-sec-dimensionalityreduction" class="nav-link" data-scroll-target="#sec-dimensionalityreduction"><span class="header-section-number">3.3</span> Dimensionality reduction regression</a></li>
  <li><a href="#sec-decisiontrees" id="toc-sec-decisiontrees" class="nav-link" data-scroll-target="#sec-decisiontrees"><span class="header-section-number">3.4</span> Decision-tree-based learning</a></li>
  <li><a href="#sec-deeplearning" id="toc-sec-deeplearning" class="nav-link" data-scroll-target="#sec-deeplearning"><span class="header-section-number">3.5</span> Deep learning</a></li>
  </ul></li>
  <li><a href="#sec-RTM" id="toc-sec-RTM" class="nav-link" data-scroll-target="#sec-RTM"><span class="header-section-number">4</span> Radiative transfer models</a>
  <ul class="collapse">
  <li><a href="#leaf-level-radiative-transfer-models" id="toc-leaf-level-radiative-transfer-models" class="nav-link" data-scroll-target="#leaf-level-radiative-transfer-models"><span class="header-section-number">4.1</span> Leaf-level radiative transfer models</a></li>
  <li><a href="#canopy-level-radiative-transfer-models" id="toc-canopy-level-radiative-transfer-models" class="nav-link" data-scroll-target="#canopy-level-radiative-transfer-models"><span class="header-section-number">4.2</span> Canopy-level radiative transfer models</a>
  <ul class="collapse">
  <li><a href="#one-dimensional-canopy-reflectance-models" id="toc-one-dimensional-canopy-reflectance-models" class="nav-link" data-scroll-target="#one-dimensional-canopy-reflectance-models">One-dimensional canopy reflectance models</a></li>
  <li><a href="#ray-tracing" id="toc-ray-tracing" class="nav-link" data-scroll-target="#ray-tracing">Ray tracing</a></li>
  <li><a href="#three-dimensional-canopy-reflectance-models" id="toc-three-dimensional-canopy-reflectance-models" class="nav-link" data-scroll-target="#three-dimensional-canopy-reflectance-models">Three-dimensional canopy reflectance models</a></li>
  </ul></li>
  <li><a href="#the-inversion-problem" id="toc-the-inversion-problem" class="nav-link" data-scroll-target="#the-inversion-problem"><span class="header-section-number">4.3</span> The inversion problem</a>
  <ul class="collapse">
  <li><a href="#mathematical-optimization" id="toc-mathematical-optimization" class="nav-link" data-scroll-target="#mathematical-optimization">Mathematical optimization</a></li>
  <li><a href="#look-up-tables" id="toc-look-up-tables" class="nav-link" data-scroll-target="#look-up-tables">Look-up tables</a></li>
  <li><a href="#model-inversion-via-machine-learning" id="toc-model-inversion-via-machine-learning" class="nav-link" data-scroll-target="#model-inversion-via-machine-learning">Model inversion via machine learning</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#sec-data" id="toc-sec-data" class="nav-link" data-scroll-target="#sec-data"><span class="header-section-number">5</span> Describing, processing and inspecting the data</a>
  <ul class="collapse">
  <li><a href="#description-of-the-data-set" id="toc-description-of-the-data-set" class="nav-link" data-scroll-target="#description-of-the-data-set"><span class="header-section-number">5.1</span> Description of the data set</a></li>
  <li><a href="#sec-SRMprocessing" id="toc-sec-SRMprocessing" class="nav-link" data-scroll-target="#sec-SRMprocessing"><span class="header-section-number">5.2</span> Pre-processing the spectroradiometric data</a></li>
  <li><a href="#sec-HSIprocessing" id="toc-sec-HSIprocessing" class="nav-link" data-scroll-target="#sec-HSIprocessing"><span class="header-section-number">5.3</span> Pre-processing of the hyperspectral images</a></li>
  <li><a href="#sec-normalization" id="toc-sec-normalization" class="nav-link" data-scroll-target="#sec-normalization"><span class="header-section-number">5.4</span> Z-score normalization</a></li>
  <li><a href="#sec-exploratory" id="toc-sec-exploratory" class="nav-link" data-scroll-target="#sec-exploratory"><span class="header-section-number">5.5</span> Exploratory data analysis</a></li>
  </ul></li>
  <li><a href="#sec-modelling" id="toc-sec-modelling" class="nav-link" data-scroll-target="#sec-modelling"><span class="header-section-number">6</span> Technical description of the modelling</a>
  <ul class="collapse">
  <li><a href="#modelling-with-hand-crafted-features" id="toc-modelling-with-hand-crafted-features" class="nav-link" data-scroll-target="#modelling-with-hand-crafted-features"><span class="header-section-number">6.1</span> Modelling with hand-crafted features</a></li>
  <li><a href="#sec-hypertuning" id="toc-sec-hypertuning" class="nav-link" data-scroll-target="#sec-hypertuning"><span class="header-section-number">6.2</span> Training the machine learning models</a></li>
  <li><a href="#running-the-radiative-transfer-models" id="toc-running-the-radiative-transfer-models" class="nav-link" data-scroll-target="#running-the-radiative-transfer-models"><span class="header-section-number">6.3</span> Running the radiative transfer models</a>
  <ul class="collapse">
  <li><a href="#leaf-level-radiative-modelling" id="toc-leaf-level-radiative-modelling" class="nav-link" data-scroll-target="#leaf-level-radiative-modelling">Leaf level radiative modelling</a></li>
  <li><a href="#canopy-level-radiative-modelling" id="toc-canopy-level-radiative-modelling" class="nav-link" data-scroll-target="#canopy-level-radiative-modelling">Canopy level radiative modelling</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#sec-results" id="toc-sec-results" class="nav-link" data-scroll-target="#sec-results"><span class="header-section-number">7</span> Results of the model performances</a>
  <ul class="collapse">
  <li><a href="#performance-on-the-leaf-level" id="toc-performance-on-the-leaf-level" class="nav-link" data-scroll-target="#performance-on-the-leaf-level"><span class="header-section-number">7.1</span> Performance on the leaf level</a>
  <ul class="collapse">
  <li><a href="#sec-ResultsAreaBased" id="toc-sec-ResultsAreaBased" class="nav-link" data-scroll-target="#sec-ResultsAreaBased">Area-based nitrogen</a></li>
  <li><a href="#sec-ResultsMassBased" id="toc-sec-ResultsMassBased" class="nav-link" data-scroll-target="#sec-ResultsMassBased">Mass-based nitrogen</a></li>
  </ul></li>
  <li><a href="#performance-on-the-canopy-level" id="toc-performance-on-the-canopy-level" class="nav-link" data-scroll-target="#performance-on-the-canopy-level"><span class="header-section-number">7.2</span> Performance on the canopy level</a></li>
  </ul></li>
  <li><a href="#sec-discussion" id="toc-sec-discussion" class="nav-link" data-scroll-target="#sec-discussion"><span class="header-section-number">8</span> Discussion</a>
  <ul class="collapse">
  <li><a href="#different-modelling-approaches" id="toc-different-modelling-approaches" class="nav-link" data-scroll-target="#different-modelling-approaches"><span class="header-section-number">8.1</span> Different modelling approaches</a></li>
  <li><a href="#how-radiative-transfer-models-might-be-improved" id="toc-how-radiative-transfer-models-might-be-improved" class="nav-link" data-scroll-target="#how-radiative-transfer-models-might-be-improved"><span class="header-section-number">8.2</span> How radiative transfer models might be improved</a></li>
  <li><a href="#general-possibilities-for-improvement" id="toc-general-possibilities-for-improvement" class="nav-link" data-scroll-target="#general-possibilities-for-improvement"><span class="header-section-number">8.3</span> General possibilities for improvement</a></li>
  <li><a href="#addressing-the-quality-and-properties-of-the-data" id="toc-addressing-the-quality-and-properties-of-the-data" class="nav-link" data-scroll-target="#addressing-the-quality-and-properties-of-the-data"><span class="header-section-number">8.4</span> Addressing the quality and properties of the data</a></li>
  <li><a href="#range-of-the-nitrogen-measurements" id="toc-range-of-the-nitrogen-measurements" class="nav-link" data-scroll-target="#range-of-the-nitrogen-measurements"><span class="header-section-number">8.5</span> Range of the nitrogen measurements</a></li>
  </ul></li>
  <li><a href="#sec-conclusion" id="toc-sec-conclusion" class="nav-link" data-scroll-target="#sec-conclusion"><span class="header-section-number">9</span> Conclusion</a></li>
  <li><a href="#acknowledgement" id="toc-acknowledgement" class="nav-link" data-scroll-target="#acknowledgement">Acknowledgement</a></li>
  <li><a href="#appendix" id="toc-appendix" class="nav-link" data-scroll-target="#appendix">Appendix</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
</div>
<main class="content quarto-banner-title-block page-columns page-full" id="quarto-document-content">




<section id="background" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Background</h1>
<section id="sec-almonds" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="sec-almonds"><span class="header-section-number">1.1</span> Almond production in California</h2>
<p>The United States of America is by far the world’s largest producer of almonds (<em>Prunus amygdalus</em> <span class="smallcaps">Batsch</span>) with a production share of roughly 80% <span class="citation" data-cites="almonds2020">(<a href="#ref-almonds2020" role="doc-biblioref">CDFA 2020</a>)</span>. Almost all of this production takes place in the state of California, whose Mediterranean climate and well-developed water infrastructure make it possible to grow the crop commercially, even though almonds are not native to the state <span class="citation" data-cites="bjerga2018california">(<a href="#ref-bjerga2018california" role="doc-biblioref">Bjerga et al. 2018</a>)</span>. Almond trees have an average life span of 20 to 25 years, whereby it typically takes three to four years for a tree to begin producing fruits. The almond trees in California begin blooming mid-February through March and harvest takes place in October before the wet season starts. The crops are alternate bearing, which means that strong harvests are succeeded by weaker ones in the following year <span class="citation" data-cites="almonds2016">(<a href="#ref-almonds2016" role="doc-biblioref">WIFSS 2016</a>)</span>.</p>
<p>One could argue that the expansion of its almond industry is California’s biggest agricultural success story <span class="citation" data-cites="bruno2021outlook">(<a href="#ref-bruno2021outlook" role="doc-biblioref">Bruno et al. 2021</a>)</span>. Over the last three decades, the production volume increased almost eight-fold: the state produced about 1.27 million tons of almonds on 408’000 ha in 2020 <span class="citation" data-cites="almonds2021">(<a href="#ref-almonds2021" role="doc-biblioref">USDA 2021</a>)</span>. The majority of the growing sites stretch along the Central Valley, most notably the counties <em>Fresno</em>, <em>Kern</em>, <em>Stanislaus</em>, <em>Merced</em>, and <em>Madera</em> (<a href="#fig-MapCalifornia">Figure&nbsp;1</a>). The economic value of Californian almond production sums up to a staggering 5.62 billion U.S. dollars annually. Thus, almonds rank third place among California’s agricultural commodities, trailing only the dairy and grape industries <span class="citation" data-cites="bruno2021outlook">(<a href="#ref-bruno2021outlook" role="doc-biblioref">Bruno et al. 2021</a>)</span>.</p>
<p>This immense production comes at a cost. California is a comparatively dry state, where agriculture uses around 40% of the total available water supply — although that percentage varies dramatically between wet and dry years <span class="citation" data-cites="mount2016water">(<a href="#ref-mount2016water" role="doc-biblioref">Mount and Hanak 2016</a>)</span>. It is estimated that Californian almond orchards alone use approximately 10% of the state’s water supply <span class="citation" data-cites="holthaus2015">(<a href="#ref-holthaus2015" role="doc-biblioref">Holthaus 2015</a>)</span>. The scarcity of water is only expected to increase due to reduced amount of snowpack in the Sierras, depleting groundwater levels and climate change — the California Department of Water Resources expects a 10% reduction in water supply by 2040 <span class="citation" data-cites="watersupply2022">(<a href="#ref-watersupply2022" role="doc-biblioref">DWR 2022</a>)</span>. Although discerning strong trends from climate projections is difficult, a drying trend for the Sacramento River Basin during the 21<sup>st</sup> century is observed through different climate scenarios developed <span class="citation" data-cites="california2015california">(<a href="#ref-california2015california" role="doc-biblioref">DWR 2015</a>)</span>. As a consequence to these environmental changes, almond yields are predicted to decrease in the coming decades <span class="citation" data-cites="pathak2018climate">(<a href="#ref-pathak2018climate" role="doc-biblioref">Pathak et al. 2018</a>)</span>.</p>
<div id="fig-MapCalifornia" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/MapOfCalifornia.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;1: Map of California (Mercator projection). The shading of the counties relates to their almond production relative to the total Californian almond production. The counties with the highest almond production are <em>Fresno</em> (1), <em>Kern</em> (2), <em>Stanislaus</em> (3), <em>Merced</em> (4), and <em>Madera</em> (5). The area encircled by the dashed line indicates the approximate location of the Californian Central Valley. The data used to produce this map was obtained from the U.S. Department of Agriculture <span class="citation" data-cites="almonds2021">(<a href="#ref-almonds2021" role="doc-biblioref">USDA 2021</a>)</span>.</figcaption>
</figure>
</div>
<p>These trends are likely to increase the economic pressure on almond farmers and provide incentives to take up water-saving measures. While some older orchards in California may still be irrigated using flood irrigation, 85% of the orchards use low-volume irrigation, which is much preferable for water conservation <span class="citation" data-cites="almondsustainability2020">(<a href="#ref-almondsustainability2020" role="doc-biblioref">ABC 2020</a>)</span>. In addition to that, some farmers already adopted the use of remote sensing for a more resourceful irrigation, either using water-stress-specific vegetation indices or thermal imaging <span class="citation" data-cites="ihuoma2017recent ceresimaging vineview almondsustainability2020">(<a href="#ref-ihuoma2017recent" role="doc-biblioref">Ihuoma and Madramootoo 2017</a>, <a href="#ref-almondsustainability2020" role="doc-biblioref">ABC 2020</a>, <a href="#ref-ceresimaging" role="doc-biblioref">Ceres Imaging 2022</a>, <a href="#ref-vineview" role="doc-biblioref">VineView 2022</a>)</span>.</p>
<p>Another serious problem stemming from the intensive agriculture in California is nitrogen leaching due to over-fertilization. The subsequent excesses of nitrate in water resources have been found to contribute to environmental degradation in the state <span class="citation" data-cites="nitrogen2012">(<a href="#ref-nitrogen2012" role="doc-biblioref">Viers et al. 2012</a>)</span>. For almond orchards, it is officially recommended to apply nitrogen in four rates: 20% of the total annual application at the end of the blooming stage, 30% during fruit growth, 30% during kernel fill and at last, 20% after the hull-split or after harvest <span class="citation" data-cites="nitrogen2020">(<a href="#ref-nitrogen2020" role="doc-biblioref">Brown et al. 2020</a>)</span>. Thereby, it is not uncommon in intensive almond orchards to apply up to 300 kg nitrogen per hectare and year <span class="citation" data-cites="smart2014nitrogen">(<a href="#ref-smart2014nitrogen" role="doc-biblioref">Smart et al. 2014</a>)</span>, although the Almond Board of California (ABC) states that the fertilization should consider both the almond tree’s age as well as the expected kernel yield <span class="citation" data-cites="nitrogen2020">(<a href="#ref-nitrogen2020" role="doc-biblioref">Brown et al. 2020</a>)</span>. As such, recommendations for best management practices range between 130 and 285 kg nitrogen per hectare and year. Thus, under a new regulation of the Californian state to avoid environmental contamination, farmers are required to apply nitrogen fertilizations appropriately <span class="citation" data-cites="zhang2019california">(<a href="#ref-zhang2019california" role="doc-biblioref">Zhang et al. 2019</a>)</span>.</p>
</section>
<section id="sec-nitrogen" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="sec-nitrogen"><span class="header-section-number">1.2</span> Challenges with balancing crop nitrogen</h2>
<p>Nitrogen (<span class="math inline">\(\text{N}\)</span>) is a vital element for plant growth. It is essential for the plant’s metabolism and is found in proteins, nucleic acids, chlorophyll, co-enzymes, phytohormones, and secondary metabolites. Plants take up nitrogen in the form of ammonium or nitrate and convert it into amino acids, either in their roots or shoots. Once inside the plant, nitrogen is transported as nitrate or amino acids <span class="citation" data-cites="hawkesford2012functions">(<a href="#ref-hawkesford2012functions" role="doc-biblioref">Hawkesford et al. 2012</a>)</span>. The use of synthetic nitrogen fertilizers has drastically increased since the green revolution and has been a major driver for higher food production <span class="citation" data-cites="vitousek1997human">(<a href="#ref-vitousek1997human" role="doc-biblioref">Vitousek et al. 1997</a>)</span>. However, the recovery of nitrogen from these fertilizers by plants is rather low and rarely exceeds 50% of the fertilizer input <span class="citation" data-cites="raun1999fertilizer">(<a href="#ref-raun1999fertilizer" role="doc-biblioref">Raun et al. 1999</a>)</span>. The proportional uptake by plants is usually even lower in the case of organic fertilizers when compared to mineral fertilizers <span class="citation" data-cites="webb2013assessment">(<a href="#ref-webb2013assessment" role="doc-biblioref">Webb et al. 2013</a>)</span>.</p>
<p>The resulting loss of nitrogen from agricultural systems negatively impacts non-agricultural systems, such as terrestrial, ground, and freshwater ecosystems. Through leaching, nitrogen can move to other ecosystems in the form of nitrate (<span class="math inline">\(\text{NO}_3^-\)</span>) and cause eutrophication of surface water or contamination of drinking water <span class="citation" data-cites="yang2008mechanisms kapoor1997nitrate">(<a href="#ref-kapoor1997nitrate" role="doc-biblioref">Kapoor and Viraraghavan 1997</a>, <a href="#ref-yang2008mechanisms" role="doc-biblioref">Yang et al. 2008</a>)</span>. Agricultural production also contributes heavily to the emission of gaseous nitrogen in the form of ammonia (<span class="math inline">\(\text{NH}_3\)</span>), molecular nitrogen (<span class="math inline">\(\text{N}_2\)</span>) or nitrogen oxide (<span class="math inline">\(\text{NO}_2\)</span>) and nitrous oxide (<span class="math inline">\(\text{N}_2 \text O\)</span>), the latter two of which are greenhouse gases and thus accelerate climate change <span class="citation" data-cites="stark2008continuing">(<a href="#ref-stark2008continuing" role="doc-biblioref">Stark and Richards 2008</a>)</span>. On the other hand, nitrogen deficiency negatively affects crop yields both in terms of quantity as well as quality. Therefore, there is also a substantial economic interest to continuously monitor the nitrogen status of crops during all growth stages <span class="citation" data-cites="baret2007quantification hank2019spaceborne">(<a href="#ref-baret2007quantification" role="doc-biblioref">Baret et al. 2007</a>, <a href="#ref-hank2019spaceborne" role="doc-biblioref">Hank et al. 2019</a>)</span>.</p>
<p>Foliar nitrogen concentrations vary a lot among plant species, phenological stages and environmental conditions, ranging between 0.2 to 6.4% of the leaf dry weight <span class="citation" data-cites="wright2004worldwide">(<a href="#ref-wright2004worldwide" role="doc-biblioref">Wright et al. 2004</a>)</span>. The optimal nitrogen concentration in almond leaves lies between 2.2 and 2.5%—the exact value depends on the time and the targeted yield <span class="citation" data-cites="micke1996">(<a href="#ref-micke1996" role="doc-biblioref">Micke 1996</a>)</span>. In C<sub>3</sub> plant leaves, nitrogen is indispensable for photosynthesis, biosynthesis, carbon fixation and bioenergetics, since is a crucial element to both proteins and chlorophyll a and b (<a href="#fig-NitrogenInPlants">Figure&nbsp;2</a>). Chlorophyll is a key component in the light harvesting process of plants and its porphyrin core contains four nitrogen atoms, which act as a stabilizer of the central magnesium ion <span class="citation" data-cites="evans1989photosynthesis">(<a href="#ref-evans1989photosynthesis" role="doc-biblioref">Evans 1989</a>)</span>. However, in C<sub>3</sub> plants, chlorophyll a and b only bind 1.7% of the total leaf nitrogen <span class="citation" data-cites="chapin1987plant">(<a href="#ref-chapin1987plant" role="doc-biblioref">Chapin et al. 1987</a>)</span>. Precise knowledge of the nitrogen concentration in plant leaves can be used for a more targeted fertilization and might even pave the way for variable rate fertilization <span class="citation" data-cites="guerrero2021current">(<a href="#ref-guerrero2021current" role="doc-biblioref">Guerrero et al. 2021</a>)</span>. For example, one might consider the nitrogen nutrition index (NNI) introduced by <span class="citation" data-cites="justes1994determination">Justes et al. (<a href="#ref-justes1994determination" role="doc-biblioref">1994</a>)</span>. <span id="eq-NNI"><span class="math display">\[
\text {NNI} = \frac{\text {N}_\%}{\text {N}_\text{crit.}}
\tag{1}\]</span></span> Here, N<sub>%</sub> is the measured nitrogen concentration and N<sub>crit.</sub> the critical nitrogen concentration. If NNI <span class="math inline">\(\geq\)</span> 1, the crop is sufficiently (or even excessively) provided with nitrogen. Vice versa NNI <span class="math inline">\(&lt;\)</span> 1 implies a nitrogen deficiency. Unfortunately, the retrieval of information about the foliar nitrogen concentration via chemical analyses is generally labour and cost intensive and thus not fit for monitoring the nitrogen concentration in plants throughout their seasonal development <span class="citation" data-cites="berger2020crop">(<a href="#ref-berger2020crop" role="doc-biblioref">Berger et al. 2020</a>)</span>. This calls for surrogate strategies of estimating the foliar nitrogen concentration — for example via remote sensing. Remote sensing of crop nitrogen for more adequate fertilization is in fact one of the core promises of precision agriculture and smart farming <span class="citation" data-cites="zhang2002precision walter2017smart sishodia2020applications">(<a href="#ref-zhang2002precision" role="doc-biblioref">Zhang et al. 2002</a>, <a href="#ref-walter2017smart" role="doc-biblioref">Walter et al. 2017</a>, <a href="#ref-sishodia2020applications" role="doc-biblioref">Sishodia et al. 2020</a>)</span>.</p>
<div id="fig-NitrogenInPlants" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/NitrogenInPlants.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;2: This figure shows the proportional functional allocation of nitrogen in an average C<sub>3</sub> plant leaf. The percentages indicate what share of nitrogen is bound for a specific purpose. Adapted from <span class="citation" data-cites="chapin1987plant">Chapin et al. (<a href="#ref-chapin1987plant" role="doc-biblioref">1987</a>)</span>.</figcaption>
</figure>
</div>
</section>
<section id="sec-remotesensing" class="level2" data-number="1.3">
<h2 data-number="1.3" class="anchored" data-anchor-id="sec-remotesensing"><span class="header-section-number">1.3</span> The role of remote sensing in precision agriculture</h2>
<p>According to <span class="citation" data-cites="pierce1999aspects">Pierce and Nowak (<a href="#ref-pierce1999aspects" role="doc-biblioref">1999</a>)</span>, precision agriculture involves using various technologies and principles to address the spatial and temporal variability in all elements of agricultural production, with the ultimate goal of enhancing crop performance and maintaining environmental quality. As such, precision agriculture is a part of smart farming, which aims at the optimization of agricultural systems driven by digitally available information <span class="citation" data-cites="wolfert2017big">(<a href="#ref-wolfert2017big" role="doc-biblioref">Wolfert et al. 2017</a>)</span>. A core driver of precision agriculture is the increasing availability of remotely sensed data as well as the means of analyzing this data <span class="citation" data-cites="sishodia2020applications">(<a href="#ref-sishodia2020applications" role="doc-biblioref">Sishodia et al. 2020</a>)</span>. Remote sensing describes the process of acquiring information about properties of an object without physical contact to said object, i.e., the attempt to measure something from distance rather than <em>in situ</em> <span class="citation" data-cites="schowengerdt2006remote">(<a href="#ref-schowengerdt2006remote" role="doc-biblioref">Schowengerdt 2006</a>)</span>. This type of data is typically measured by satellites, aircraft, or unmanned aerial vehicles <span class="citation" data-cites="sishodia2020applications">(<a href="#ref-sishodia2020applications" role="doc-biblioref">Sishodia et al. 2020</a>)</span>.</p>
<p>For numerous agricultural applications, remote sensing focuses on measuring the electromagnetic radiation reflected or emitted from the Earth’s surface, primarily the reflection of sunlight in the visible and near-infrared spectrum <span class="citation" data-cites="jafarbiglu2022comprehensive">(<a href="#ref-jafarbiglu2022comprehensive" role="doc-biblioref">Jafarbiglu and Pourreza 2022</a>)</span>. Sensors more effectively capture signals in the visible light spectrum between 380 and 750 nm wavelength because the electromagnetic radiation of the sun is the peaking there, whereas it tapers off in the ultraviolet and near- to short-wave-infrared regions (<a href="#fig-Electromagnetic">Figure&nbsp;3</a>, A).</p>
<p>The intensity of this electromagnetic radiation — its flux density — is typically measured in watts per square metre <span class="citation" data-cites="ISO9288">(W m<sup>−2</sup>, <a href="#ref-ISO9288" role="doc-biblioref">ISO 2022</a>)</span>. If this radiation is received by a physical object, it is referred to as irradiance, if it is emitted by the object, it is called radiance. The spectral irradiance and radiance on the other hand are a measure of irradiance or radiance over a specific range of wavelengths (or frequencies) rather than across the entire electromagnetic spectrum, and thus typically measured in watts per square metre per nanometre <span class="citation" data-cites="ISO9288">(W m<sup>−2</sup> nm<sup>−1</sup>, <a href="#ref-ISO9288" role="doc-biblioref">ISO 2022</a>)</span>. The solar irradiance measured at a particular time and location depends on many parameters such as the solar incidence angle, the height above mean sea level, or the current atmospheric conditions <span class="citation" data-cites="stickler2016educational blumthaler1997increase">(<a href="#ref-blumthaler1997increase" role="doc-biblioref">Blumthaler et al. 1997</a>, <a href="#ref-stickler2016educational" role="doc-biblioref">Stickler and Kyle 2016</a>)</span>. Because of that, remote sensing applications do not directly analyse the radiance — which is just as dependent on those parameters as the irradiance — but rather the so-called reflectance (<span class="math inline">\(\rho\)</span>). Reflectance is the share of electromagnetic radiation which is reflected by a body. Consequentially, the spectral reflectance is defined as the ratio between the spectral radiance <span class="math inline">\(\Phi_{\text e, \lambda} ^\text r\)</span> and the spectral irradiance <span class="math inline">\(\Phi_{\text e, \lambda} ^\text i\)</span> <span class="citation" data-cites="ISO9288">(<a href="#ref-ISO9288" role="doc-biblioref">ISO 2022</a>)</span>. <span id="eq-reflectance"><span class="math display">\[
\rho_\lambda = \frac{\Phi_{\text e, \lambda} ^\text r}{\Phi_{\text e, \lambda} ^\text i}
\tag{2}\]</span></span></p>
<p>At which exact wavelengths <span class="math inline">\(\lambda\)</span> this reflectance is calculated depends on the available data. Data from an ordinary digital camera usually covers three bands: red (620-740 nm), green (500-560 nm), and blue (450-490 nm), which is why it is referred to as RGB imagery <span class="citation" data-cites="bruno2005crc">(<a href="#ref-bruno2005crc" role="doc-biblioref">Bruno and Svoronos 2005</a>)</span>. In contrast to that, multispectral images are a type of remotely sensed data that consist of more than three bands beyond just the visible spectrum. For agricultural purposes, multispectral imagery typically combines RGB imagery with bands in the near-infrared spectrum <span class="citation" data-cites="hagen2013review">(<a href="#ref-hagen2013review" role="doc-biblioref">Hagen and Kudenov 2013</a>)</span>. Because the infrared spectrum will be quite important within the scope of this thesis, I will often refer to certain subgroups of that range: near-infrared (NIR, 750-1400 nm), short-wavelength-infrared (SWIR, 1400-3000 nm), mid-wavelength infrared (MWIR, 3-8 <span class="math inline">\(\mu\)</span>m), long-wavelength infrared (LWIR, 8-15 <span class="math inline">\(\mu\)</span>m) and far infrared <span class="citation" data-cites="Byrnes2009">(FIR, 15-1000 <span class="math inline">\(\mu\)</span>m, <a href="#ref-Byrnes2009" role="doc-biblioref">Byrnes 2009</a>)</span>.</p>
<div id="fig-Electromagnetic" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/Electromagnetic.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;3: This figure illustrates the spectral flux density of solar irradiance (<strong>A</strong>) in relation to the broader electromagnetic spectrum (<strong>B</strong>). A portion of the initial irradiance is absorbed by various gases or scattered by Rayleigh scattering, resulting in only a part of it reaching the Earth’s surface at sea level. The irradiance and radiance data at a given surface location (<strong>C</strong>) can be utilized to calculate the reflectance based on <a href="#eq-reflectance">equation&nbsp;2</a> (<strong>D</strong>). This reflectance enables the determination of various physical and biochemical properties of the surface. This figure was adapted from work by <span class="citation" data-cites="Rohde2013">Rohde (<a href="#ref-Rohde2013" role="doc-biblioref">2013</a>)</span> and created using data from the <span class="citation" data-cites="NREL">National Renewable Energy Laboratory (<a href="#ref-NREL" role="doc-biblioref">2022</a>)</span>.</figcaption>
</figure>
</div>
<p>When incorporating numerous additional spectral bands into the imaging process, it is referred to as hyperspectral imaging. Hyperspectral images (HSI) usually consist of hundreds of narrow, contiguous wavelength bands, which together with the spatial resolution form a hypercube <span class="math inline">\(\mathbf X \in \mathbb R^{w \times l \times d}\)</span> with the width <span class="math inline">\(w\)</span>, length <span class="math inline">\(l\)</span> and the depth <span class="math inline">\(d\)</span> (that is the number of spectral bands). The high spectral resolution is the main advantage of hyperspectral imaging, it is what allows for the detailed analysis of the chemical and physical properties of the imaged surface <span class="citation" data-cites="bioucas2013hyperspectral">(<a href="#ref-bioucas2013hyperspectral" role="doc-biblioref">Bioucas-Dias et al. 2013</a>)</span>. This high spectral resolution however stands in a trade-off relationship with the spatial resolution, which is why hyperspectral imaging usually yields a lower spatial resolution than multispectral imaging <span class="citation" data-cites="yokoya2013hyperspectral">(<a href="#ref-yokoya2013hyperspectral" role="doc-biblioref">Yokoya and Iwasaki 2013</a>)</span>. Sometimes, the continuous information about an object’s reflectance found solely in the spectral dimension of hyperspectral images is also referred to as its spectral signature. A very important property of the spectral dimension of hyperspectral images is the strong autocorrelation among spectral bands, which is expressed in extremely non-diagonal covariance matrices <span class="citation" data-cites="camps2011remote">(<a href="#ref-camps2011remote" role="doc-biblioref">Camps-Valls et al. 2011</a>)</span>.</p>
<p>In order to to capture hyperspectral images, specialized hyperspectral cameras are used. They typically consist of a spectrograph, which separates the incoming light into its individual wavelength bands, and of an array of detectors that measure the intensity of the light in each band <span class="citation" data-cites="elmasry2010principles">(<a href="#ref-elmasry2010principles" role="doc-biblioref">El Masry and Sun 2010</a>)</span>. The spectrograph and detectors are combined into a single, integrated unit, which is mounted on the carrying platform. These sensors work well in the visible spectrum, but the quality starts to drop in the near-infrared region and especially in the short-wave-infrared region (after 900 nm wavelength, see <a href="#fig-Electromagnetic">figure&nbsp;3</a>).</p>
<p>There are several reasons for this. One reason is that the near-infrared and short-wavelength infrared regions are typically characterized by lower levels of radiance compared to the visible light spectrum, which can make it difficult to obtain high-quality, noise-free data. In addition, the short-wavelength infrared region is often affected by atmospheric absorption of different gases, especially water vapour, which can cause the signal strength to vary significantly depending on the atmospheric conditions (<a href="#fig-Electromagnetic">Figure&nbsp;3</a> A). This noise is sometimes removed using atmospheric correction algorithms <span class="citation" data-cites="gao2009atmospheric">(<a href="#ref-gao2009atmospheric" role="doc-biblioref">Gao et al. 2009</a>)</span>. A third challenge in the short-wavelength infrared region is the need for specialized detectors that are sensitive to the wavelengths of light in this region. These detectors are typically made of alloys such as indium gallium arsenide or mercury cadmium telluride, which are more expensive and less sensitive than the detectors used in the visible and near-infrared regions <span class="citation" data-cites="dyson2021 ciani2020comparison">(<a href="#ref-ciani2020comparison" role="doc-biblioref">Ciani et al. 2020</a>, <a href="#ref-dyson2021" role="doc-biblioref">Dyson 2021</a>)</span>.</p>
<p>Due to these reasons, hyperspectral imaging of sufficient quality is often limited to the visible- and near-infrared spectrum. Acquiring exact spectral information in the short-wave-infrared spectrum is mostly reserved to spectroradiometric measurements, although this starts to change with the advancement of hyperspectral sensors <span class="citation" data-cites="weber2016extended">(<a href="#ref-weber2016extended" role="doc-biblioref">Weber et al. 2016</a>)</span>.</p>
</section>
<section id="sec-nitrogenRS" class="level2" data-number="1.4">
<h2 data-number="1.4" class="anchored" data-anchor-id="sec-nitrogenRS"><span class="header-section-number">1.4</span> Remote sensing of nitrogen</h2>
<p>The first attempt to directly model crop nitrogen based on spectrophotometrically measured reflectance data has been made by <span class="citation" data-cites="thomas1972estimating">Thomas and Oerther (<a href="#ref-thomas1972estimating" role="doc-biblioref">1972</a>)</span>, who worked on sweet potatoes. The authors stated:</p>
<blockquote class="blockquote">
<p><em>“The regression equation at the 550 nm wavelength suggests that near the 4% nitrogen level a 1% change in reflectance would change the predicted nitrogen content by 0.2%.”</em></p>
</blockquote>
<p>In other words, they proposed a simple linear regression model (on logarithmically transformed variables) to quickly estimate the nitrogen concentration — solely based on the reflectance at 550 nm wavelength. Back then, this relationship had to be found by calculating all band-wise correlations and simply choosing a suitable band for the regression. Ever since, much more advanced methods of modelling crop nitrogen have been developed. There are many ways to divide the modelling approaches into groups. For example, <span class="citation" data-cites="verrelst2019quantifying">Verrelst et al. (<a href="#ref-verrelst2019quantifying" role="doc-biblioref">2019</a>)</span> proposed grouping the methods in parametric, non-parametric, physics-based and hybrid methods. <span class="citation" data-cites="berger2020crop">Berger et al. (<a href="#ref-berger2020crop" role="doc-biblioref">2020</a>)</span> followed up on that division, but further distinguished between linear and non-linear methods — which they equate to the terms chemometrics and machine learning.</p>
<p>However, the terms parametric and non-parametric as used by <span class="citation" data-cites="berger2020crop">Berger et al. (<a href="#ref-berger2020crop" role="doc-biblioref">2020</a>)</span> collide with pre-occupied definitions in the field of machine learning. Furthermore, the formal definition of machine learning also applies to many of the other groups proposed by the authors. Therefore, I will use the following simplified classification throughout this thesis:</p>
<ol type="1">
<li>Hand-crafted features</li>
<li>Machine learning</li>
<li>Radiative transfer models</li>
</ol>
<p>The primary distinction between the first two methods lies in how the bands which are used to make predictions are selected. Hand-crafted features describe all methods which select a subset of bands or a mathematical combination thereof (as in the case of vegetation indices) which is made prior to the fitting of the regression model. Thereby, the band selection happens according to a certain imposed logic, or according to what is already used in the scientific literature. In contrast to that, the models classified as machine learning have access to the entire data, i.e., all available bands. The model automatically learns which bands or combinations of bands are the most important for accurately predicting crop nitrogen without prior information or manual feature engineering. Radiative transfer models, conversely, are primarily designed to simulate the detailed physical processes governing the transport of light and its interaction with a leaf or canopy. As a result, these models incorporate a substantial amount of prior knowledge. At this point, it is important to acknowledge that ultimately, these classifications possess a degree of arbitrariness and exhibit considerable overlap in their respective methodologies.</p>
<p>All three listed methods have been explored in the scientific literature, but not in comparable detail. <span class="citation" data-cites="berger2020crop">Berger et al. (<a href="#ref-berger2020crop" role="doc-biblioref">2020</a>)</span> found that 78 out of 125 evaluated studies that aimed to predict crop nitrogen via remote sensing relied on vegetation indices. Out of these, only a few studies have highlighted the limitations of indices-based approaches, such as their limited transferability to different sites. The studies mostly relied on the empirical relationship of certain vegetation indices to the chlorophyll content of crops, which in turn is related to nitrogen <span class="citation" data-cites="penuelas1994reflectance baret2007quantification">(<a href="#ref-penuelas1994reflectance" role="doc-biblioref">Peñuelas et al. 1994</a>, <a href="#ref-baret2007quantification" role="doc-biblioref">Baret et al. 2007</a>)</span>. The application of machine learning for crop nitrogen estimation was also investigated by numerous studies, where the most often applied model was the partial least squares regression model. Radiative transfer models on the other hand have only been used in three studies for modelling crop nitrogen <span class="citation" data-cites="berger2020crop">(<a href="#ref-berger2020crop" role="doc-biblioref">Berger et al. 2020</a>)</span>.</p>
<p>Crop nitrogen or nitrogen content are both ambiguous terms, although very commonly used in the scientific literature. Ever since <span class="citation" data-cites="thomas1972estimating">Thomas and Oerther (<a href="#ref-thomas1972estimating" role="doc-biblioref">1972</a>)</span> pioneered remote sensing of nitrogen, the focus lied on two different measurements: mass- and area-based nitrogen <span class="citation" data-cites="wright2004worldwide">(<a href="#ref-wright2004worldwide" role="doc-biblioref">Wright et al. 2004</a>)</span>. Mass-based nitrogen (N<sub>%</sub>) is the nitrogen concentration in the leaves, typically relative to the leaf dry weight. Area-based nitrogen (N<sub>area</sub>) describes how much nitrogen is present in the biomass over a certain area. The two measures are linked together by the leaf mass per area (LMA), which is a widely used morphological trait that indicates good plant growth and functioning <span class="citation" data-cites="de2016leaf">(<a href="#ref-de2016leaf" role="doc-biblioref">de la Riva et al. 2016</a>)</span>. <span id="eq-LMA"><span class="math display">\[
\text N_{\text{area}} = \text{LMA} \times \text N_{\%}
\tag{3}\]</span></span></p>
<p>The majority of the contemporary studies focus on multi- or hyperspectral data. However, there is still some ongoing research about the usage of broadly available and low cost RGB imagery for predicting nitrogen content in crops, although with no great success <span class="citation" data-cites="schirrmann2016monitoring zheng2018evaluation fu2020winter">(<a href="#ref-schirrmann2016monitoring" role="doc-biblioref">Schirrmann et al. 2016</a>, <a href="#ref-zheng2018evaluation" role="doc-biblioref">Zheng et al. 2018</a>, <a href="#ref-fu2020winter" role="doc-biblioref">Fu et al. 2020</a>)</span>.</p>
<p>Within the visible light spectrum, three types of plant pigments mainly determine the absorption process and thus the leaf colour: chlorophylls, carotenoids, and anthocyanins. Chlorophylls include chlorophyll a and b with their respective absorption bands, while carotenoids (e.g., <span class="math inline">\(\beta\)</span>-carotene and xanthophyll) and anthocyanins include numerous compounds <span class="citation" data-cites="feret2008prospect">(<a href="#ref-feret2008prospect" role="doc-biblioref">Féret et al. 2008</a>)</span>. Out of these pigments, only chlorophyll is directly connected to nitrogen. It accounts for roughly 1.7% of total leaf nitrogen, which explains the rather loose Pearson correlation (<span class="math inline">\(\rho = 0.65 \pm 0.15\)</span>) between the chlorophyll and the nitrogen concentration in leaves <span class="citation" data-cites="kokaly2009characterizing homolova2013review">(<a href="#ref-kokaly2009characterizing" role="doc-biblioref">Kokaly et al. 2009</a>, <a href="#ref-homolova2013review" role="doc-biblioref">Homolová et al. 2013</a>)</span>. It remains to be noted that the Pearson correlation might not be the best metric to compare nitrogen and chlorophyll, as <span class="citation" data-cites="padilla2018different">Padilla et al. (<a href="#ref-padilla2018different" role="doc-biblioref">2018</a>)</span> have found a second order polynomial model to slightly improve the model fit.</p>
<p>Fully relying on chlorophyll for the remote sensing of crop nitrogen is highly limited and can even be misleading. First, the prediction will always be limited to the correlation between the two traits. Also, the relationship between chlorophyll and nitrogen changes over time, as nitrogen is reallocated from the vegetative organs to reproductive structures during the reproductive stage <span class="citation" data-cites="ohyama2010nitrogen">(<a href="#ref-ohyama2010nitrogen" role="doc-biblioref">Ohyama 2010</a>)</span>. Finally, a lack of chlorophyll cannot certainly be ascribed to a nitrogen deficiency but also any other chlorosis-causing nutrient deficiency <span class="citation" data-cites="asner2009airborne bloem2005influence">(<a href="#ref-bloem2005influence" role="doc-biblioref">Bloem et al. 2005</a>, <a href="#ref-asner2009airborne" role="doc-biblioref">Asner and Martin 2009</a>)</span>.</p>
<p>Due to these reasons, leaf protein content would be a more stable proxy for the foliar nitrogen concentration <span class="citation" data-cites="berger2020crop">(<a href="#ref-berger2020crop" role="doc-biblioref">Berger et al. 2020</a>)</span>. The majority of plant nitrogen is bound to protein (<a href="#fig-NitrogenInPlants">Figure&nbsp;2</a>), thus there is less room for decoupling between the two traits <span class="citation" data-cites="kokaly2009characterizing">(<a href="#ref-kokaly2009characterizing" role="doc-biblioref">Kokaly et al. 2009</a>)</span>.</p>
</section>
<section id="sec-aims" class="level2" data-number="1.5">
<h2 data-number="1.5" class="anchored" data-anchor-id="sec-aims"><span class="header-section-number">1.5</span> Aims and structure of this thesis</h2>
<p>In this master’s thesis, I pursue two primary objectives: First, I aim to gain a comprehensive understanding of the various methods for modelling crop nitrogen, as outlined in the previous section. Second, I search for a model capable of effectively mapping the spectral signature of almond canopies or leaves with their respective nitrogen concentrations, while maintaining generalizability across diverse temporal variations, including different times of the year, distinct years, and varied geographical locations.</p>
<p>With these goals in mind, the thesis is structured as follows: The first part delves into each of the approaches previously introduced. <a href="#sec-VI">Chapter&nbsp;2</a> examines remote sensing of nitrogen using vegetation indices. <a href="#sec-ML">Chapter&nbsp;3</a> is dedicated to selected machine learning algorithms suitable for hyperspectral data analysis. <a href="#sec-RTM">Chapter&nbsp;4</a> investigates physics-based radiative transfer models, beginning at the leaf level and progressing to the canopy level.</p>
<p>Please note that this thesis does not include a systematic literature review and, as such, does not claim to be complete with respect of the modelling methods regarded. For instance, the radiative transfer models discussed in <a href="#sec-RTM">chapter&nbsp;4</a> represent merely a small subset of the available models. Instead, the primary objective is to understand the underlying principles as well as the distinctions among the three techniques.</p>
<p>This thesis resulted in a collaboration with the Digital Agriculture Laboratory at the University of California, Davis, which is led by Dr.&nbsp;Alireza Pourreza. The lab’s members collected data from a Californian almond orchard prior to my involvement; however, the data was highly unstructured. Consequently, a portion of this thesis’ workload involved cleaning and pre-processing the data, which is described in detail in <a href="#sec-data">chapter&nbsp;5</a>.</p>
<p>The modelling stage is documented in <a href="#sec-modelling">chapter&nbsp;6</a>, and the model performance results are presented in <a href="#sec-results">chapter&nbsp;7</a>. An in-depth discussion regarding the model’s generalizability and other relevant topics can be found in <a href="#sec-discussion">chapter&nbsp;8</a>, with concluding remarks in <a href="#sec-conclusion">chapter&nbsp;9</a>.</p>
</section>
</section>
<section id="sec-VI" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Vegetation-indices-based methods</h1>
<p>Vegetation indices are mathematical constructs — typically expressed as ratios — that are derived from the measurement of vegetation reflectance at two or more specific wavelengths <span class="citation" data-cites="huete2002overview">(<a href="#ref-huete2002overview" role="doc-biblioref">Huete et al. 2002</a>)</span>. These indices are frequently employed to evaluate the health, biophysical properties, and biochemical characteristics of vegetation. Notably, vegetation indices have been utilized for applications such as estimating plant parameters such as the leaf area index, detecting water stress in crops, and classifying various types of vegetation for weed mapping purposes <span class="citation" data-cites="jackson1981canopy zheng2009retrieving">(<a href="#ref-jackson1981canopy" role="doc-biblioref">Jackson et al. 1981</a>, <a href="#ref-zheng2009retrieving" role="doc-biblioref">Zheng and Moskal 2009</a>)</span>. Of particular relevance to the subject of this thesis, they have also been commonly employed for estimating chlorophyll and nitrogen content in crops <span class="citation" data-cites="berger2020crop">(<a href="#ref-berger2020crop" role="doc-biblioref">Berger et al. 2020</a>)</span>. Precision agriculture technologies relying on vegetation indices have been well-integrated into commercial applications, an outcome that can likely be attributed to the simplicity of their implementation and their comparatively early development <span class="citation" data-cites="xue2017significant sishodia2020applications">(<a href="#ref-xue2017significant" role="doc-biblioref">Xue and Su 2017</a>, <a href="#ref-sishodia2020applications" role="doc-biblioref">Sishodia et al. 2020</a>)</span>.</p>
<section id="evolution-of-vegetation-indices-and-their-limitations" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="evolution-of-vegetation-indices-and-their-limitations"><span class="header-section-number">2.1</span> Evolution of vegetation indices and their limitations</h2>
<p>Historically, one of the very first vegetation indices developed was the simple ratio index (RVI). It was proposed by <span class="citation" data-cites="jordan1969derivation">Jordan (<a href="#ref-jordan1969derivation" role="doc-biblioref">1969</a>)</span>. <span id="eq-RVI"><span class="math display">\[
\text{RVI} = \frac{\rho_\text{R}}{\rho_\text{NIR}}
\tag{4}\]</span></span> In this and the following equations, <span class="math inline">\(\rho_\text{R}\)</span> stands for the red band reflectance, whereas <span class="math inline">\(\rho_\text{NIR}\)</span> stands for the near-infrared band reflectance. Not too soon after introduction of the RVI, <span class="citation" data-cites="rouse1973monitoring">Rouse Jr et al. (<a href="#ref-rouse1973monitoring" role="doc-biblioref">1973</a>)</span> proposed the normalized difference vegetation index (NDVI), which until today remains the most widely used vegetation index <span class="citation" data-cites="xue2017significant">(<a href="#ref-xue2017significant" role="doc-biblioref">Xue and Su 2017</a>)</span>. <span id="eq-NDVI"><span class="math display">\[
\text{NDVI} = \frac{\rho_\text{NIR}-\rho_\text{R}}{\rho_\text{NIR}+\rho_\text{R}}
\tag{5}\]</span></span> Compared to the RVI, the NDVI is normalized between the values of −1 and +1. It is quite sensitive to green vegetation, even in sparsely covered areas. However, the NDVI is also influenced by soil colour, brightness, and shadows cast by canopy and clouds, which are undesirable traits for a vegetation index, as it should ideally be affected solely by the presence of vegetation <span class="citation" data-cites="xue2017significant">(<a href="#ref-xue2017significant" role="doc-biblioref">Xue and Su 2017</a>)</span>. Consequently, numerous vegetation indices have been developed subsequent to the NDVI, aiming to address its limitations and incorporate atmospheric corrections <span class="citation" data-cites="kaufman1992atmospherically ren1996approach">(<a href="#ref-kaufman1992atmospherically" role="doc-biblioref">Kaufman and Tanre 1992</a>, <a href="#ref-ren1996approach" role="doc-biblioref">Ren-hua et al. 1996</a>)</span> or adjustments for soil colour and brightness <span class="citation" data-cites="huete1988soil qi1994modified richardson1977distinguishing baret1993soil">(<a href="#ref-richardson1977distinguishing" role="doc-biblioref">Richardson and Wiegand 1977</a>, <a href="#ref-huete1988soil" role="doc-biblioref">Huete 1988</a>, <a href="#ref-baret1993soil" role="doc-biblioref">Baret et al. 1993</a>, <a href="#ref-qi1994modified" role="doc-biblioref">Qi et al. 1994</a>)</span>. A recent review by <span class="citation" data-cites="xue2017significant">Xue and Su (<a href="#ref-xue2017significant" role="doc-biblioref">2017</a>)</span> examined over one hundred vegetation indices, with many specifically targeting the shortcomings of the NDVI. The authors discovered that, in various specific scenarios — considering factors such as a particular crop, soil type, and atmospheric conditions — certain vegetation indices demonstrated superior performance over others.</p>
<p>Early development of vegetation indices primarily relied on bands frequently captured by multispectral cameras, such as red, red edge, and near-infrared wavelengths <span class="citation" data-cites="chlingaryan2018machine">(<a href="#ref-chlingaryan2018machine" role="doc-biblioref">Chlingaryan et al. 2018</a>)</span>. Blue bands were rarely employed, as blue light is more susceptible to atmospheric scattering, which can introduce greater uncertainty in measurements — an aspect of particular significance in satellite-based remote sensing applications. However, the advent of hyperspectral imaging has ushered in the creation of vegetation indices tailored to very specific narrow bands, expanding the capabilities of these tools — but at the same time massively increasing the risk of over-fitting <span class="citation" data-cites="thenkabail2013selection">(<a href="#ref-thenkabail2013selection" role="doc-biblioref">Thenkabail et al. 2013</a>)</span>.</p>
<p>In the end, relationships between vegetation indices and the trait of interest are purely empirical in nature. Typically, data on the plant trait is collected, and various vegetation indices are tested to determine the ones that correlate well with it. This approach can be critiqued as being somewhat unscientific, as it relies more on coincidentally found correlations than on fundamental principles or theories.</p>
<p><span class="citation" data-cites="atzberger2011confining">Atzberger et al. (<a href="#ref-atzberger2011confining" role="doc-biblioref">2011</a>)</span> explicitly outlines five disadvantages associated with using vegetation indices-based methods for crop nitrogen estimation. The first disadvantage is the inverse problem, wherein high-dimensional data is reduced to a single value, which could potentially be derived from varying forms of reflectance. The second drawback involves the partial exploitation of multi- or hyperspectral data, as only a limited number of spectral bands are utilized. This issue is especially relevant to narrow-band vegetation indices, which may also be affected by measurement noise. The third challenge concerns the lack of transferability in the calibration of vegetation indices to specific sensors, as it does not extend to instruments with different characteristics. The fourth issue relates to overfitting, implying that vegetation index-based nitrogen models may be tailored too specifically to the dataset used for model calibration. This can lead to sensitivity regarding crop type, growth stage, underlying soil type, sun-sensor constellation, as well as time and date. Finally, the fifth drawback highlights the need to collect <em>in situ</em> reference data for model establishment, which demands manpower, instrument availability, and therefore imposes a high financial burden on the development.</p>
<p>Despite this criticism, new vegetation indices keep being proposed in the scientific literature. Some of the drawbacks of vegetation indices are addressed by the usage of radiative transfer models: Synthetic data sets are generated, where new vegetation indices aim to correlate with the parameter used to generate the respective data sets <span class="citation" data-cites="hunt2013visible">(<a href="#ref-hunt2013visible" role="doc-biblioref">Hunt Jr et al. 2013</a>)</span>. For example, the radiative transfer model <span class="smallcaps">Prosail</span> has been used to design new vegetation indices <span class="citation" data-cites="cheng2014estimating jin2014newly">(<a href="#ref-cheng2014estimating" role="doc-biblioref">Cheng et al. 2014</a>, <a href="#ref-jin2014newly" role="doc-biblioref">Jin et al. 2014</a>)</span>. This search for new vegetation indices is sometimes even implemented algorithmically, for example by the adoption of = algorithms <span class="citation" data-cites="costa2020new">(<a href="#ref-costa2020new" role="doc-biblioref">Costa et al. 2020</a>)</span>.</p>
</section>
<section id="sec-nitrogenVI" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="sec-nitrogenVI"><span class="header-section-number">2.2</span> Nitrogen monitoring with vegetation indices</h2>
<p>Vegetation indices represent the most prevalent method for remote detection of crop nitrogen content <span class="citation" data-cites="berger2020crop">(<a href="#ref-berger2020crop" role="doc-biblioref">Berger et al. 2020</a>)</span>. Numerous indices have been intentionally designed to estimate crop nitrogen, while others have simply been found to correlate with it <span class="citation" data-cites="herrmann2010swir">(<a href="#ref-herrmann2010swir" role="doc-biblioref">Herrmann et al. 2010</a>)</span>. These indices mostly capitalize on the correlation between crop nitrogen and chlorophyll content <span class="citation" data-cites="baret2007quantification">(<a href="#ref-baret2007quantification" role="doc-biblioref">Baret et al. 2007</a>)</span>. For instance, the transformed chlorophyll absorption in reflectance index (TCARI) has demonstrated a strong correlation with chlorophyll <span class="citation" data-cites="cui2017comparison">(<a href="#ref-cui2017comparison" role="doc-biblioref">Cui and Zhou 2017</a>)</span>. Some of the most notable vegetation indices employed for crop nitrogen detection can be found in <a href="#tbl-VegetationIndices">table&nbsp;1</a>.</p>
<p>But how exactly is a regression model employing vegetation indices constructed? The utilization of vegetation indices on multi- or hyperspectral data results in a dimensionality reduction of the original data. While the initial data possesses multiple dimensions, vegetation indices are one-dimensional. Ideally, a vegetation index employed for crop nitrogen monitoring correlates more strongly with crop nitrogen than any individual band. However, the index itself lacks a unit and, as a consequence, cannot be used directly to estimate crop nitrogen. Therefore, any vegetation index must be linked to the trait of interest <span class="math inline">\(y\)</span> through a connecting function <span class="math inline">\(f\)</span> — a so-called link function. <span id="eq-VIlinkfunction"><span class="math display">\[
f: g(\mathbf x) \rightarrow y
\tag{6}\]</span></span> In this case, <span class="math inline">\(g(\mathbf x)\)</span> represents the vegetation index applied to the spectral data. Typically, a simple linear regression is employed to identify the linking function <span class="math inline">\(f\)</span>. Nevertheless, <span class="math inline">\(f\)</span> could represent any function and might even accept multiple vegetation indices as inputs. For instance, <span class="citation" data-cites="johnson2016crop">Johnson et al. (<a href="#ref-johnson2016crop" role="doc-biblioref">2016</a>)</span> found that a multiple linear regression model yielded better predictions based on multiple vegetation indices than relying on a single vegetation index at a time.</p>
<p>The linking function <span class="math inline">\(f\)</span> could even take the form of a more flexible machine learning model able to represent intricate interactions, such as a random forest or support vector machines <span class="citation" data-cites="sonobe2018crop">(<a href="#ref-sonobe2018crop" role="doc-biblioref">Sonobe et al. 2018</a>)</span>. This practice, however, raises the question of the necessity for vegetation indices in the first place. Machine learning models can be applied directly to the data, enabling the direct selection of relevant features. One could, so to say, “cut out the middleman”. This effectively eliminates several of the issues addressed by <span class="citation" data-cites="atzberger2011confining">Atzberger et al. (<a href="#ref-atzberger2011confining" role="doc-biblioref">2011</a>)</span>. Hence, the application of machine learning algorithms to hyperspectral data in order to estimate crop nitrogen will be the focus of the next chapter.</p>
<div id="tbl-VegetationIndices" class="anchored">
<table class="table">
<caption>Table&nbsp;1: Overview of various vegetation indices that were used to estimate foliar nitrogen concentrations. <span class="math inline">\(\rho_\lambda\)</span> for the reflectance at a specific wavelength <span class="math inline">\(\lambda\)</span> (<span class="math inline">\(\rho_{530}\)</span> = green band, <span class="math inline">\(\rho_{685}\)</span> = red band, <span class="math inline">\(\rho_{720}\)</span> = red edge, <span class="math inline">\(\rho_{790}\)</span> = near-infrared). This table was adapted from <span class="citation" data-cites="herrmann2010swir">Herrmann et al. (<a href="#ref-herrmann2010swir" role="doc-biblioref">2010</a>)</span> and <span class="citation" data-cites="tian2011assessing">Tian et al. (<a href="#ref-tian2011assessing" role="doc-biblioref">2011</a>)</span>.</caption>
<colgroup>
<col style="width: 35%">
<col style="width: 45%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"><strong>Index</strong></th>
<th style="text-align: left;"><strong>Formula</strong></th>
<th style="text-align: left;"><strong>Author</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Normalised Difference Vegetation Index (NDVI)</td>
<td style="text-align: left;"><span class="math display">\[\frac{\rho_{\text{790}}-\rho_{\text{685}}}{\rho_{\text{790}}+\rho_{\text{685}}}\]</span></td>
<td style="text-align: left;"><span class="citation" data-cites="rouse1973monitoring">Rouse Jr et al. (<a href="#ref-rouse1973monitoring" role="doc-biblioref">1973</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">Normalized Difference Red Edge (NDRE)</td>
<td style="text-align: left;"><span class="math display">\[\frac{\rho_{790} - \rho_{720}}{\rho_{790} + \rho_{720}}\]</span></td>
<td style="text-align: left;"><span class="citation" data-cites="barnes2000coincident">Barnes et al. (<a href="#ref-barnes2000coincident" role="doc-biblioref">2000</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Modified Chlorophyll Absorption in Reflectance Index (MCARI)</td>
<td style="text-align: left;"><span class="math display">\[\frac{\rho_{700}}{\rho_{670}} \left[ 0.8 \cdot \rho_{700} - \rho_{670} - 0.2 \cdot \rho_{550} \right]\]</span></td>
<td style="text-align: left;"><span class="citation" data-cites="daughtry2000estimating">Daughtry et al. (<a href="#ref-daughtry2000estimating" role="doc-biblioref">2000</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">Transformed Chlorophyll Absorption in Reflectance Index (TCARI)</td>
<td style="text-align: left;"><span class="math display">\[3 \left[ \rho_{700} - \rho_{670} - \frac{0.2\cdot\rho_{700}}{\rho_{670}} \left(\rho_{700} - \rho_{550} \right) \right]\]</span></td>
<td style="text-align: left;"><span class="citation" data-cites="haboudane2002integrated">Haboudane et al. (<a href="#ref-haboudane2002integrated" role="doc-biblioref">2002</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Green Chlorophyll Index (GCI)</td>
<td style="text-align: left;"><span class="math display">\[\frac{\rho_{\text{790}}}{\rho_{530}}-1\]</span></td>
<td style="text-align: left;"><span class="citation" data-cites="gitelson2003relationships">Gitelson et al. (<a href="#ref-gitelson2003relationships" role="doc-biblioref">2003</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">Red Edge Chlorophyll Index (RECI)</td>
<td style="text-align: left;"><span class="math display">\[\frac{\rho_{\text{790}}}{\rho_{\text{720}}}-1\]</span></td>
<td style="text-align: left;"><span class="citation" data-cites="gitelson2003relationships">Gitelson et al. (<a href="#ref-gitelson2003relationships" role="doc-biblioref">2003</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Green Normalised Difference Vegetation Index (GNDVI)</td>
<td style="text-align: left;"><span class="math display">\[\frac{\rho_{\text{790}}-\rho_{530}}{\rho_{\text{790}}+\rho_{530}}\]</span></td>
<td style="text-align: left;"><span class="citation" data-cites="gitelson1996use">Gitelson et al. (<a href="#ref-gitelson1996use" role="doc-biblioref">1996</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">Enhanced Vegetation Index (EVI)</td>
<td style="text-align: left;"><span class="math display">\[\frac{\frac{5}{2}(\rho_{\text{790}} - \rho_{\text{720}})}{\rho_{\text{790}} + 6 \ \rho_{\text{720}} + \frac{15}{2} \rho_{\text{470}} + 1}\]</span></td>
<td style="text-align: left;"><span class="citation" data-cites="huete2002overview">Huete et al. (<a href="#ref-huete2002overview" role="doc-biblioref">2002</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Wide Dynamic Range Vegetation Index (WDRVI)</td>
<td style="text-align: left;"><span class="math display">\[\frac{\alpha \rho_{\text{790}}-\rho_{\text{720}}}{\alpha \rho_{\text{790}}+\rho_{\text{720}}}\]</span></td>
<td style="text-align: left;"><span class="citation" data-cites="gitelson2004wide">Gitelson (<a href="#ref-gitelson2004wide" role="doc-biblioref">2004</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">Normalized Difference Nitrogen Index (NDNI)</td>
<td style="text-align: left;"><span class="math display">\[\frac{\log_{10} \rho_{1510}^{-1} - \log_{10} \rho_{1680}^{-1}}{\log_{10} \rho_{1510}^{-1} + \log_{10} \rho_{1680}^{-1}}\]</span></td>
<td style="text-align: left;"><span class="citation" data-cites="serrano2002remote">Serrano et al. (<a href="#ref-serrano2002remote" role="doc-biblioref">2002</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">R434</td>
<td style="text-align: left;"><span class="math display">\[\frac{\rho_{\text{434}}}{\rho_{\text{496}} - \rho_{\text{401}}}\]</span></td>
<td style="text-align: left;"><span class="citation" data-cites="tian2011assessing">Tian et al. (<a href="#ref-tian2011assessing" role="doc-biblioref">2011</a>)</span></td>
</tr>
</tbody>
</table>
</div>
</section>
</section>
<section id="sec-ML" class="level1 page-columns page-full" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Machine learning methods for hyperspectral data analysis</h1>
<div class="page-columns page-full"><p>The term machine learning was first used by <span class="citation" data-cites="samuel1959machine">Samuel (<a href="#ref-samuel1959machine" role="doc-biblioref">1959</a>)</span>. He described it as the “field of study that gives computers the ability to learn without being explicitly programmed”. Back then, only few people were interested in this particular field. Today, however, this could not be further from the truth: A quick <span class="smallcaps">Scopus</span> search reveals, that in 2022 alone, 18’464 scientific publications use the term “machine learning” in their title — that is more than 50 publications a day!<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a></p><div class="no-row-height column-margin column-container"><li id="fn1"><p><sup>1</sup>&nbsp;The <span class="smallcaps">Scopus</span> search for <code>machine AND learning</code> was conducted on March 7, 2023. Only the document type “article” was searched for. 449’663 publications of all time contain “machine learning” in their title, abstract or keywords. 57’926 contain “machine learning” in the title only. Out of all the articles ever using “machine learning” in their title, 32% were published in 2022.</p></li></div></div>
<p>The substantial advancements witnessed in machine learning over the past two decades can be largely ascribed to the surge in computational power, the sudden availability of copious amounts of data, and the open-source culture that enables applications through readily accessible machine learning libraries and frameworks <span class="citation" data-cites="jordan2015machine fridman2019">(<a href="#ref-jordan2015machine" role="doc-biblioref">Jordan and Mitchell 2015</a>, <a href="#ref-fridman2019" role="doc-biblioref">Fridman 2019</a>)</span>.</p>
<p>In this thesis, I will use the term “machine learning” as originally defined by Arthur Samuels. As per his definition, machine learning includes a variety of mathematical models from chemometrics and statistics, despite the fact that many of these models were developed prior to the coining of the term itself. The essential aspect of these models lies in their primary function for making predictions, rather than serving as statistical models. Moreover, the machine learning models evaluated in this thesis are differentiated from hand-crafted features such as vegetation indices and radiative transfer models by their fully automatic process of selecting and combining spectral bands.</p>
<p>Owing to their exceptional predictive capabilities and handling of high-dimensional input data, machine learning algorithms have emerged as a widely adopted tool for modern hyperspectral image analysis <span class="citation" data-cites="gewali2018machine">(<a href="#ref-gewali2018machine" role="doc-biblioref">Gewali et al. 2018</a>)</span>. In this chapter, various machine learning algorithms that have been extensively used on hyperspectral data will be examined. To do this, I will loosely follow the route taken by <span class="citation" data-cites="james2013introduction">James et al. (<a href="#ref-james2013introduction" role="doc-biblioref">2013</a>)</span>, commencing with the ordinary least squares model and addressing why it is a poor choice for hyperspectral data, and subsequently slowly relaxing the assumptions of the linear model.</p>
<section id="sec-stepwise" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="sec-stepwise"><span class="header-section-number">3.1</span> Ordinary least squares and stepwise selection</h2>
<p>The simplest and also probably most prevalently used machine learning model is the linear regression model <span class="citation" data-cites="yan2009linear">(<a href="#ref-yan2009linear" role="doc-biblioref">Yan and Su 2009</a>)</span>. In its univariate form, it can be represented as follows. <span id="eq-generalLinearModel"><span class="math display">\[
\mathbf y = \mathbf X \boldsymbol \beta + \boldsymbol \varepsilon
\tag{7}\]</span></span> Here, <span class="math inline">\(\mathbf y\)</span> represents the labels, <span class="math inline">\(\mathbf X\)</span> denotes the design matrix comprising all predictors, <span class="math inline">\(\boldsymbol \beta\)</span> signifies the coefficients, and <span class="math inline">\(\boldsymbol \varepsilon\)</span> corresponds to the model’s errors. The customary approach for estimating the coefficients <span class="math inline">\({\boldsymbol \beta}\)</span> in <a href="#eq-generalLinearModel">equation&nbsp;7</a> employs the analytical solution using the ordinary least squares estimator. <span id="eq-ols"><span class="math display">\[
\hat {\boldsymbol \beta} = (\mathbf X^\top\mathbf X)^{-1} \mathbf X^\top \mathbf y
\tag{8}\]</span></span> This method was (allegedly) introduced independently by <span class="citation" data-cites="legendre1806nouvelles">Legendre (<a href="#ref-legendre1806nouvelles" role="doc-biblioref">1806</a>)</span> in <em>Nouvelles méthodes pour la détermination des orbites des comètes</em> (new methods for the determination of comet orbits) and <span class="citation" data-cites="gauss1809theoria">Gauss (<a href="#ref-gauss1809theoria" role="doc-biblioref">1809</a>)</span> in <em>Theoria Motus Corporum Coelestium</em> (theory of the motion of the heavenly bodies), eventually being formalized at a later date by <span class="citation" data-cites="markov1912wahrscheinlichkeitsrechnung">Markov (<a href="#ref-markov1912wahrscheinlichkeitsrechnung" role="doc-biblioref">1912</a>)</span>. However, it is worth noting that Markov did not contribute any novel proof <span class="citation" data-cites="plackett1949historical yan2009linear">(<a href="#ref-plackett1949historical" role="doc-biblioref">Plackett 1949</a>, <a href="#ref-yan2009linear" role="doc-biblioref">Yan and Su 2009</a>)</span>.</p>
<p>In what is now known as the Gauss-Markov theorem, Markov asserts that the ordinary least squares estimator is the best linear unbiased estimator under the specific conditions that the error’s expectation is zero (<span class="math inline">\(\mathbb E(\varepsilon_i) = 0\)</span>), the error exhibits equal variance (<span class="math inline">\(\text{Var}(\varepsilon_i) = \sigma^2\)</span>), and all distinct error terms are uncorrelated (<span class="math inline">\(\text{Cov}(\varepsilon_i, \varepsilon_j) = 0\)</span>).</p>
<p>Unprocessed hyperspectral reflectance data frequently violates the assumptions of the Gauss-Markov theorem — particularly the third assumption regarding uncorrelated errors. While equal variance can be achieved through z-score normalization (a technique described in more detail in <a href="#sec-normalization">section&nbsp;5.4</a>), and the expectation of zero error might be satisfied if the response is investigated on a small enough scale, the assumption of uncorrelated error terms cannot be fulfilled while retaining all available variables or information about all bands. The crux of the problem lies in so-called multicollinearity: the reflectance of neighbouring spectral bands is typically very strongly correlated. Consequently, the reflectance at one band can be predicted linearly from others with a very high precision <span class="citation" data-cites="chlingaryan2018machine">(<a href="#ref-chlingaryan2018machine" role="doc-biblioref">Chlingaryan et al. 2018</a>)</span>. Because of multicollinearity, minor changes in the data or model specification can have a disproportional effect on parameter estimates <span class="citation" data-cites="ravikanth2017extraction">(<a href="#ref-ravikanth2017extraction" role="doc-biblioref">Ravikanth et al. 2017</a>)</span>.</p>
<p>Ordinary least squares also encounter difficulties in hyperspectral data analysis due to the <em>small n, large p problem</em>. In many agricultural applications of hyperspectral data, there is often a limit to the number of samples taken, since the acquisition of observations can be both costly and labour-intensive <span class="citation" data-cites="ravikanth2017extraction">(<a href="#ref-ravikanth2017extraction" role="doc-biblioref">Ravikanth et al. 2017</a>)</span>. When the design matrix <span class="math inline">\(\mathbf X\)</span> has fewer rows (observations, <span class="math inline">\(n\)</span>) than columns (parameters, <span class="math inline">\(p\)</span>), the inversion of <span class="math inline">\(\mathbf X^\top\mathbf X\)</span> in <a href="#eq-ols">equation&nbsp;8</a> becomes impossible.</p>
<p>If one wanted to strictly stick to linear regression using ordinary least squares, a potential workaround evading the <em>small n, large p problem</em> would be the application of a prior feature selection strategy to tackle multicollinearity (remember, vegetation indices as discussed in the chapter before essentially constituted such a feature selection method). Provided that there are more observations than parameters to estimate, feature selection methods like stepwise selection can also further create a subset of variables for the linear regression model such that it is no longer over-specified. These methods minimize a so-called information criterion. This metric balances both the model’s likelihood and penalizes over-parametrization simultaneously. The Akaike information criterion (AIC), introduced by <span class="citation" data-cites="akaike1998information">Akaike (<a href="#ref-akaike1998information" role="doc-biblioref">1998</a>)</span>, serves as one such criterion for comparing the fit of various regression models. It can be calculated as follows. <span id="eq-AIC"><span class="math display">\[
\text{AIC} = k(p - \ln \hat {L})
\tag{9}\]</span></span> In this equation, <span class="math inline">\(p\)</span> represents the number of parameters, <span class="math inline">\(k\)</span> is a scalar, and <span class="math inline">\(\ln \hat {L}\)</span> denotes the estimated log-likelihood of the model. A true AIC is obtained when <span class="math inline">\(k=2\)</span>, while <span class="math inline">\(k=\ln(n)\)</span> (where <span class="math inline">\(n\)</span> is the number of observations) is sometimes called the Bayesian information criterion (BIC) or Schwarz information criterion (SIC). Through stepwise selection, variables are iteratively added to and removed from the model so as to minimize the information criterion.</p>
<p>A practical application of this approach can be seen in the research conducted by <span class="citation" data-cites="castaldi2016data">Castaldi et al. (<a href="#ref-castaldi2016data" role="doc-biblioref">2016</a>)</span>. The authors focused on predicting wheat grain nitrogen uptake from satellite data. To identify a suitable subset of spectral bands, they executed a stepwise selection process on the variables of a multiple linear regression model.</p>
<p>However, the application of stepwise selection to hyperspectral data has faced substantial criticism <span class="citation" data-cites="bolster1996determination grossman1996critique">(<a href="#ref-bolster1996determination" role="doc-biblioref">Bolster et al. 1996</a>, <a href="#ref-grossman1996critique" role="doc-biblioref">Grossman et al. 1996</a>)</span>. The key issue is that its performance markedly declines as the number of utilized bands increases. When dealing with a too large number of variables, stepwise selection begins to select a seemingly random subset of bands for linear regression, which subsequently results in overfitting <span class="citation" data-cites="hastie2017extended">(<a href="#ref-hastie2017extended" role="doc-biblioref">Hastie et al. 2017</a>)</span>. In situations where a linear regression model incorporating more than a handful of bands is sought, regularization of the model may prove to be a more suitable solution.</p>
</section>
<section id="sec-regularization" class="level2 page-columns page-full" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="sec-regularization"><span class="header-section-number">3.2</span> Regularization</h2>
<div class="page-columns page-full"><p>Regularization is a technique extensively employed in machine learning in order to avoid overfitting and enhance the generalization of models. Applicable to linear models as well as more intricate ones, such as neural networks or decision-tree-based models, regularization operates by adding a penalty term to the original, unregularized cost function <span class="math inline">\(J\)</span>, which in turn penalizes the magnitude of the estimated coefficient. Consequently, larger coefficient estimates lead to a greater penalty <span class="citation" data-cites="james2013introduction">(<a href="#ref-james2013introduction" role="doc-biblioref">James et al. 2013</a>)</span>. This process forces the regression model to strike a balance between small (or few) coefficients and an effective explanation of the data. In that sense, regularization can be understood as Occam’s Razor<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a> applied to machine learning models, favouring simpler models over their more complex counterparts — even if the more complex ones could better explain the data. The underlying reasoning for this approach is that, when given enough flexibility, models inherently tend to over-explain the data. As a countermeasure, regularization restricts some of that flexibility, promoting a more balanced model.</p><div class="no-row-height column-margin column-container"><li id="fn2"><p><sup>2</sup>&nbsp;Occam’s razor is a philosophical principle that suggests that, among competing hypotheses, the simplest one with the fewest assumptions is preferable or more likely to be true <span class="citation" data-cites="van2018occam">(<a href="#ref-van2018occam" role="doc-biblioref">Van den Berg 2018</a>)</span>.</p></li></div></div>
<p>The penalty term is typically based on various vector norms of the model coefficients, such as the <span class="math inline">\(\ell^1\)</span>-norm (referred to as lasso regularization) or the <span class="math inline">\(\ell^2\)</span>-norm (known as ridge regularization). The cost function <span class="math inline">\(J\)</span> for both lasso and ridge regression can be expressed in a single equation as follows. <span id="eq-RLSloss"><span class="math display">\[
J(\boldsymbol \beta, \mathbf X, \mathbf y) =
\| \mathbf {y -X\boldsymbol\beta} \|_2^2 +
\lambda \left[ \frac{1}{2}(1-\alpha)\| \boldsymbol \beta \|_2^2 + \alpha \| \boldsymbol \beta \| \right]
\tag{10}\]</span></span> Here, <span class="math inline">\(\| \cdot \|\)</span> represents the absolute value norm or the <span class="math inline">\(\ell^1\)</span>-norm, while <span class="math inline">\(\| \cdot \|_2\)</span> denotes the Euclidean norm or the <span class="math inline">\(\ell^2\)</span>-norm. It is worth noting that the first term in <a href="#eq-RLSloss">equation&nbsp;10</a> (<span class="math inline">\(\| \mathbf {y -X\boldsymbol\beta} \|_2^2\)</span>) is simply the standard cost function for linear least squares regression, as it equates to the sum of squared errors. The second half constitutes the penalty term that regularizes the coefficients. The parameter <span class="math inline">\(\lambda\)</span> determines the magnitude of the penalty, as it is multiplied by a mixture of norms of the coefficient vector <span class="math inline">\(\boldsymbol \beta\)</span>. If one exclusively selects <span class="math inline">\(\alpha = 0\)</span>, that results in ridge regression, since only the <span class="math inline">\(\ell^2\)</span>-norm is employed. Conversely, choosing <span class="math inline">\(\alpha = 1\)</span> relies solely on the <span class="math inline">\(\ell^1\)</span>-norm, resulting in lasso regression. Any value between <span class="math inline">\(0 &lt; \alpha &lt; 1\)</span> yields a combination of the two, referred to as the elastic net <span class="citation" data-cites="zou2005regularization james2013introduction">(<a href="#ref-zou2005regularization" role="doc-biblioref">Zou and Hastie 2005</a>, <a href="#ref-james2013introduction" role="doc-biblioref">James et al. 2013</a>)</span>. Consequently, we can regard <span class="math inline">\(\lambda \in [0, \infty)\)</span> as a parameter controlling the penalty’s magnitude and <span class="math inline">\(\alpha \in [0,1]\)</span> as a parameter directing the type of regularization applied.</p>
<p>In general, regularization aids in mitigating multicollinearity issues, consequently reducing the variance of coefficient estimates. Ridge regression, based on the Euclidean norm, progressively decreases the penalty on coefficient estimates as they approach zero. This means, that for ridge regression, the variables in the model will always exert an effect, albeit a tiny one in the case of useless variables.</p>
<p>Lasso regression on the other hand, initially introduced by <span class="citation" data-cites="tibshirani1996">Tibshirani (<a href="#ref-tibshirani1996" role="doc-biblioref">1996</a>)</span>, begins to induce sparsity in the data matrix as <span class="math inline">\(\lambda\)</span> increases <span class="citation" data-cites="james2013introduction">(<a href="#ref-james2013introduction" role="doc-biblioref">James et al. 2013</a>)</span>. This property means that lasso regression leads to a selection of variables that effectively explain the response. Unlike stepwise selection, lasso regression does not suffer from a high number of simultaneously used variables <span class="citation" data-cites="hastie2017extended">(<a href="#ref-hastie2017extended" role="doc-biblioref">Hastie et al. 2017</a>)</span>. Lasso regression emerges as a particularly appealing tool for hyperspectral data analysis, as it facilitates the selection of a limited number of bands that best explain the response variable <span class="citation" data-cites="takayama2016optimal">(<a href="#ref-takayama2016optimal" role="doc-biblioref">Takayama and Iwasaki 2016</a>)</span>.</p>
<p>The number of selected bands can be effectively controlled via <span class="math inline">\(\lambda\)</span>. Lasso regression has already been employed for spectral band selection in spectroradiometric leaf-level data, specifically in the context of foliar nitrogen prediction in vineyards <span class="citation" data-cites="omidi2020ensemble">(<a href="#ref-omidi2020ensemble" role="doc-biblioref">Omidi et al. 2020</a>)</span>.</p>
</section>
<section id="sec-dimensionalityreduction" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="sec-dimensionalityreduction"><span class="header-section-number">3.3</span> Dimensionality reduction regression</h2>
<p>Thus far, the focus has been on regression methods that address multicollinearity and the small <span class="math inline">\(n\)</span>, large <span class="math inline">\(p\)</span> problem by selecting features, penalizing coefficients, or both, as in the case of lasso regression. An alternative approach to tackling these issues involves formulating the regression after applying a dimensionality reduction technique. Principal component regression (PCR) and partial least squares regression (PLSR) stand out as the two most prominent methods for achieving this. Partial least squares, originally introduced by <span class="citation" data-cites="wold1984collinearity">Wold et al. (<a href="#ref-wold1984collinearity" role="doc-biblioref">1984</a>)</span>, is particularly prevalent in chemometrics, even though it was initially introduction as an econometric tool for addressing multicollinearity <span class="citation" data-cites="geladi1986partial mehmood2016diversity">(<a href="#ref-geladi1986partial" role="doc-biblioref">Geladi and Kowalski 1986</a>, <a href="#ref-mehmood2016diversity" role="doc-biblioref">Mehmood and Ahmed 2016</a>)</span>.</p>
<p>The two related methods can be mathematically expressed in a similar manner. Any design matrix <span class="math inline">\(\mathbf X\)</span> containing the original data can be represented by its score matrix <span class="math inline">\(\mathbf T\)</span>. <span id="eq-pca"><span class="math display">\[
\mathbf T = \mathbf{XW}
\tag{11}\]</span></span> Here, <span class="math inline">\(\mathbf W\)</span> is a <span class="math inline">\(p \times p\)</span> rotation matrix of weights. In the case of principal component analysis (PCA), the columns of <span class="math inline">\(\mathbf W\)</span> are the eigenvectors of <span class="math inline">\(\text{Cov}(\mathbf X)\)</span>. The full matrix <span class="math inline">\(\mathbf T\)</span> is thus a linear re-combination of <span class="math inline">\(\mathbf X\)</span>, but it conveniently contains maximal information in its first column, with the remaining information successively distributed in subsequent columns. The constructed variables in the score matrix, called principal components, are completely uncorrelated with each other. These principal components, also referred to as latent variables, are not directly measured but are constructed from the observed variables.</p>
<p>The lack of multicollinearity among the latent variables in the score matrix makes them better suited for ordinary least squares regression than the observed data. The more collinear the original data, the fewer principal components are needed to capture the essence of the data. This effectively eliminates the small <span class="math inline">\(n\)</span>, large <span class="math inline">\(p\)</span> problem. After performing linear dimensionality reduction on <span class="math inline">\(\mathbf X\)</span>, linear regression can be applied to a limited number of variables in the score matrix <span class="math inline">\(\mathbf T\)</span> to avoid overfitting issues typically associated with stepwise selection <span class="citation" data-cites="hansen2003reflectance grossman1996critique">(<a href="#ref-grossman1996critique" role="doc-biblioref">Grossman et al. 1996</a>, <a href="#ref-hansen2003reflectance" role="doc-biblioref">Hansen and Schjoerring 2003</a>)</span>. Both principal component and partial least squares regression can thus be formulated as follows. <span id="eq-pcr"><span class="math display">\[
\mathbf y = \mathbf T_{[1,l]} \: \boldsymbol \beta + \boldsymbol \varepsilon
\tag{12}\]</span></span> Here, <span class="math inline">\(l\)</span> is the number of variables from <span class="math inline">\(\mathbf T\)</span> used for the regression, and the rest is analogous to <a href="#eq-generalLinearModel">equation&nbsp;7</a>. The difference between principal component and partial least square regression lies in how the score matrix <span class="math inline">\(\mathbf T\)</span> and the rotation matrix <span class="math inline">\(\mathbf W\)</span> are estimated. In principal component regression, a principal component analysis is performed without considering the response <span class="math inline">\(\mathbf y\)</span>, but instead maximizing the variance in the columns of the score matrix <span class="math inline">\(\mathbf T\)</span> successively. However, this strategy could theoretically find principal components which explain maximal variance in the data, but do not in fact correlate with the labels <span class="citation" data-cites="geladi1986partial">(<a href="#ref-geladi1986partial" role="doc-biblioref">Geladi and Kowalski 1986</a>)</span>.</p>
<p>Partial least squares regression, introduced by <span class="citation" data-cites="wold1984collinearity">Wold et al. (<a href="#ref-wold1984collinearity" role="doc-biblioref">1984</a>)</span>, addresses this issue by informing the estimation of the rotation matrix with both the labels and the data at the same time, thus maximizing the predictive power of the latent variables. Specifically, partial least squares estimate <span class="math inline">\(\mathbf W\)</span> to maximize the covariance between <span class="math inline">\(\mathbf y\)</span> and <span class="math inline">\(\mathbf T\)</span>: <span id="eq-plsr"><span class="math display">\[
\mathbf W = \arg \max_{\mathbf W} \left\{ \text{Cov} \left( \mathbf y, \mathbf {XW} \right) \right\} \qquad\text{s.t. } \mathbf W ^\top \mathbf W = \mathbf I
\tag{13}\]</span></span> In this equation, <span class="math inline">\(\mathbf I\)</span> represents the identity matrix <span class="citation" data-cites="bennett2003optimization">(<a href="#ref-bennett2003optimization" role="doc-biblioref">Bennett and Embrechts 2003</a>)</span>. Note that <a href="#eq-plsr">equation&nbsp;13</a> describes the maximization task for univariate partial least squares, applicable to cases with a single response variable, although the model can also be applied to multivariate cases, where a rotation matrix is found to maximize the covariance of the principal components with respect to multiple responses simultaneously.</p>
<div style="page-break-after: always;"></div>
<p>In summary, principal component regression maximizes the variance of the principal components with respect to <span class="math inline">\(\mathbf X\)</span> and subsequently performs a regression on <span class="math inline">\(\mathbf y\)</span>. In contrast, partial least squares regression maximizes the covariance between the components <span class="math inline">\(\mathbf T\)</span> and the independent variable <span class="math inline">\(\mathbf y\)</span>. Therefore, if the primary goal is to make predictions using linear regression analysis, partial least squares regression is generally a better alternative compared to principal component regression <span class="citation" data-cites="inoue2012diagnostic liu2022partial">(<a href="#ref-inoue2012diagnostic" role="doc-biblioref">Inoue et al. 2012</a>, <a href="#ref-liu2022partial" role="doc-biblioref">Liu et al. 2022</a>)</span>.</p>
<p>Due to these properties, partial least squares regression is an excellent method for predicting a response based on hyperspectral data. It is also the most widely tested model for predicting crop nitrogen <span class="citation" data-cites="berger2020crop">(<a href="#ref-berger2020crop" role="doc-biblioref">Berger et al. 2020</a>)</span>. For instance, partial least squares were successfully used by <span class="citation" data-cites="wang2017non">Wang et al. (<a href="#ref-wang2017non" role="doc-biblioref">2017</a>)</span> to predict nitrogen in leaves from pear trees, with a validation R<sup>2</sup> value of 0.85.</p>
<p>However, there are two major weaknesses to the partial least squares approach when it comes to hyperspectral data that must be considered. First, the regression method utilizes the entire spectrum of the original variables to construct the latent variables. While this can provide some insights into which bands are more explanatory for the response, it is not as straightforward as other methods for selecting a minimum number of relevant bands that can adequately explain the response. Band selection via partial least squares requires additional analysis of the model <span class="citation" data-cites="mehmood2016diversity">(<a href="#ref-mehmood2016diversity" role="doc-biblioref">Mehmood and Ahmed 2016</a>)</span>. Second, partial least squares regression is inherently linear in nature. As a result, despite its powerful capabilities in modelling linear relationships, it may fail to capture complex, non-linear relationships present in the data — relationships that are known to exist between the foliar nitrogen concentration and the leaf’s reflectance.</p>
<p>When the underlying structure of the data is strongly non-linear, principal component and partial least squares regression may not produce satisfactory results. In such cases, their kernelized counterparts can be explored, which allow for non-linear dimensionality reduction before the regression stage <span class="citation" data-cites="rosipal2006overview">(<a href="#ref-rosipal2006overview" role="doc-biblioref">Rosipal and Krämer 2006</a>)</span>. Alternatively, other machine learning techniques, such as decision-tree-based models, Gaussian process regression, support vector regression, or artificial neural networks, might be more appropriate for modelling intrinsically non-linear relationships.</p>
</section>
<section id="sec-decisiontrees" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="sec-decisiontrees"><span class="header-section-number">3.4</span> Decision-tree-based learning</h2>
<p>Decision-tree-based methods are vastly different from the machine learning algorithms examined thus far. A decision tree can be constructed by recursively partitioning the data into subsets based on the values of individual features, and then establishing individual rules for the created subsets. The paths in these trees are called branches, and the terminal nodes are called leaves <span class="citation" data-cites="james2013introduction">(<a href="#ref-james2013introduction" role="doc-biblioref">James et al. 2013</a>)</span>. Decision trees can be employed for both classification and regression; however, since the focus of this thesis is the prediction of foliar nitrogen concentration, the discussion will centre on regression trees. Decision-tree-based learning has been widely utilized in various applications, including remote sensing applications for nitrogen detection <span class="citation" data-cites="chlingaryan2018machine">(<a href="#ref-chlingaryan2018machine" role="doc-biblioref">Chlingaryan et al. 2018</a>)</span>.</p>
<p>When used individually, decision trees often either overfit the data or fail to effectively approximate the underlying function. However, their power is significantly enhanced when employed in ensemble learning, which combines multiple trees to improve performance. Two popular ensemble methods include bagging, where trees are trained on randomly selected subsets of the data, and boosting, where trees are iteratively trained to correct the errors of their predecessors <span class="citation" data-cites="james2013introduction">(<a href="#ref-james2013introduction" role="doc-biblioref">James et al. 2013</a>)</span>.</p>
<p>One of the most prevalent bagging methods is random forests, a model introduced by <span class="citation" data-cites="breiman2001">Breiman (<a href="#ref-breiman2001" role="doc-biblioref">2001</a>)</span>. Each tree within a random forest is trained on only a subset of the observations and a limited number of variables, which allows less informative variables to influence the outcome as well. Predictions made by a random forest are simply the mean of all individual predictions of decision trees <span class="citation" data-cites="james2013introduction">(<a href="#ref-james2013introduction" role="doc-biblioref">James et al. 2013</a>)</span>. Random forests have been employed by <span class="citation" data-cites="jin2020advancing">Jin et al. (<a href="#ref-jin2020advancing" role="doc-biblioref">2020</a>)</span> due to their exceptional capability of handling non-linear relationships in order to determine which variables are essential for yield determination in Californian almonds.</p>
<p>As an alternative to bagging and random forests, the technique of boosting combines multiple weak learners, which are decision trees that perform only slightly better than random guessing, into a robust model capable of making accurate predictions <span class="citation" data-cites="james2013introduction">(<a href="#ref-james2013introduction" role="doc-biblioref">James et al. 2013</a>)</span>. The key aspect of this approach is that weak learners are fitted sequentially, with new trees attempting to reduce the error of their predecessors. Consequently, subsequent learners focus on the aspects that previous learners failed to capture. The final model is a weighted sum of the individual weak learners, with the weights determined by each learner’s relative performance. One of the latest examples of boosting is the extreme gradient boosting (XGBoost) framework, introduced by <span class="citation" data-cites="chen2016xgboost">Chen and Guestrin (<a href="#ref-chen2016xgboost" role="doc-biblioref">2016</a>)</span>. This particular algorithm has proven particularly useful for predicting the foliar nitrogen concentration of grapevines in California <span class="citation" data-cites="moghimi2020novel">(<a href="#ref-moghimi2020novel" role="doc-biblioref">Moghimi et al. 2020</a>)</span>.</p>
<p>A third variation of ensemble learning with decision trees are cubist trees, which incorporate linear regression models as node rules. This adds flexibility to the model, making the response a linear function at any point — unlike the prior two models, which are essentially limited to always be step functions.</p>
<p>In general, decision-tree-based learning can be particularly effective when applied to hyperspectral data, as the trees can manage numerous dimensions and automatically select the most relevant bands at each node. They can adeptly handle non-linear relationships, a capability that some previously discussed methods lack. Additionally, their hierarchical structure is easy to understand and interpret, reflecting the decision-making process in a clear and comprehensible manner. This advantage enables drawing conclusions about the importance of specific variables for prediction, making decision-tree-based learning useful for spectral band selection, especially in classification tasks <span class="citation" data-cites="omidi2020ensemble">(<a href="#ref-omidi2020ensemble" role="doc-biblioref">Omidi et al. 2020</a>)</span>.</p>
<p>Decision-tree-based methods have the notable drawback in that they tend to be considerably slower during the inference stage compared to other models. Remember that when making predictions with an ensemble method, all the trained models need to calculate the result, which can contribute to a longer computation time compared to other models. This disadvantage becomes especially apparent when predictions must be for every pixel in a hyperspectral image which may potentially require billions of predictions. Additionally, in regression, decision-tree-based learning may suffer from the fact that they generate inherently non-smooth functions. If smooth non-linear functions are required, methods such as support vector regression, Gaussian process regression or even artificial neural networks might be more suitable alternatives.</p>
</section>
<section id="sec-deeplearning" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="sec-deeplearning"><span class="header-section-number">3.5</span> Deep learning</h2>
<p>Much of today’s remarkable advances in the field of machine learning are largely based on a group of models called artificial neural networks. These models power the most advanced artificial intelligence systems, which effortlessly outperform humans in chess, Go, or video games <span class="citation" data-cites="stockfish2023 silver2017mastering vinyals2019grandmaster">(<a href="#ref-silver2017mastering" role="doc-biblioref">Silver et al. 2017</a>, <a href="#ref-vinyals2019grandmaster" role="doc-biblioref">Vinyals et al. 2019</a>, <a href="#ref-stockfish2023" role="doc-biblioref">Noda 2023</a>)</span>. Artificial neural networks are also applied in various scientific domains, such as chaos control in fusion reactors <span class="citation" data-cites="degrave2022magnetic">(<a href="#ref-degrave2022magnetic" role="doc-biblioref">Degrave et al. 2022</a>)</span> or for highly accurate protein structure prediction <span class="citation" data-cites="jumper2021highly">(<a href="#ref-jumper2021highly" role="doc-biblioref">Jumper et al. 2021</a>)</span>. In addition, artificial neural networks have made a significant impact in recent years in the form of generative artificial intelligence that can produce images, audio, or text <span class="citation" data-cites="openai2023 rombach2021highresolution">(<a href="#ref-rombach2021highresolution" role="doc-biblioref">Rombach et al. 2021</a>, <a href="#ref-openai2023" role="doc-biblioref">OpenAI 2023</a>)</span>. Some researchers even believe that the largest neural networks exhibit sparks of artificial general intelligence — that is artificial intelligence with human-like understanding and adaptability <span class="citation" data-cites="bubeck2023sparks">(<a href="#ref-bubeck2023sparks" role="doc-biblioref">Bubeck et al. 2023</a>)</span>.</p>
<p>Artificial neural networks are a type of machine learning algorithm that is loosely inspired by the structure and function of the human brain. The concept of artificial neural networks pre-dates many of the other models discussed so far: the so-called perceptron was formally introduced by <span class="citation" data-cites="rosenblatt1958perceptron">Rosenblatt (<a href="#ref-rosenblatt1958perceptron" role="doc-biblioref">1958</a>)</span>, with similar ideas dating as far back as to <span class="citation" data-cites="mcculloch1943logical">McCulloch and Pitts (<a href="#ref-mcculloch1943logical" role="doc-biblioref">1943</a>)</span>. Neural networks consist of multiple layers of interconnected nodes, known as neurons. Each neuron essentially functions as a generalized linear model. These neurons are organized into input, output, and hidden layers. The stacking of multiple hidden layers was referred to as deep neural networks, which gave rise to the term deep learning — basically referring to the usage of neural networks in order to solve a task.</p>
<p>Despite their numerous benefits, neural networks have some significant drawbacks: they only tend to excel in domains with large sample sizes, extremely non-linear functions, and very high-dimensional data. While the last of these characteristics applies to hyperspectral data, the first two are rarely satisfied in this context. A suitable application of neural networks in hyperspectral data analysis would therefore be one where ample data is available, such as in the case of synthetic data from simulations. Hence, artificial neural networks are commonly applied to approximate the inverse of radiative transfer models <span class="citation" data-cites="verger2011optimal">(<a href="#ref-verger2011optimal" role="doc-biblioref">Verger et al. 2011</a>)</span>.</p>
<p>Moreover, training artificial neural networks is more challenging due to the large number of tuning parameters that must be set adequately <span class="citation" data-cites="chlingaryan2018machine">(<a href="#ref-chlingaryan2018machine" role="doc-biblioref">Chlingaryan et al. 2018</a>)</span>. And more than any of the previously discussed machine learning models, artificial neural networks exemplify a common issue with all data-driven methods: while they may excel at recognizing patterns and making predictions, they do not necessarily enhance human understanding of processes. This is why <span class="citation" data-cites="berger2020crop">Berger et al. (<a href="#ref-berger2020crop" role="doc-biblioref">2020</a>)</span> advocates for further exploration of physics-based models, particularly those modelling radiative transfer.</p>
</section>
</section>
<section id="sec-RTM" class="level1 page-columns page-full" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Radiative transfer models</h1>
<p>Radiative transfer describes the energy exchange through electromagnetic radiation, with varying influences from different chemical compounds <span class="citation" data-cites="verrelst2017global">(<a href="#ref-verrelst2017global" role="doc-biblioref">Verrelst and Rivera 2017</a>)</span>. Radiative transfer models leverage this physical knowledge by simulating energy transfer from plant leaves or vegetation canopies, enabling the retrieval of vegetation properties from remotely sensed data <span class="citation" data-cites="yang2020spart">(<a href="#ref-yang2020spart" role="doc-biblioref">Yang et al. 2020</a>)</span>. Compared to machine learning approaches, radiative transfer models provide enhanced generalizability and robustness, as they embody a physical process rather than merely extracting a signal from data <span class="citation" data-cites="berger2018evaluation">(<a href="#ref-berger2018evaluation" role="doc-biblioref">Berger et al. 2018</a>)</span>.</p>
<div id="fig-PlateModel" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/PlateModel.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;4: This image depicts an anonymous artist’s conception of cross section of a maple leaf on the left, and a graphical illustration of a simple radiative transfer model on the right. The model consists of 100 circular arcs simulating the scattering of light through the leaf. The illustration was originally published by <span class="citation" data-cites="allen1973willstatter">Allen et al. (<a href="#ref-allen1973willstatter" role="doc-biblioref">1973</a>)</span>.</figcaption>
</figure>
</div>
<section id="leaf-level-radiative-transfer-models" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="leaf-level-radiative-transfer-models"><span class="header-section-number">4.1</span> Leaf-level radiative transfer models</h2>
<p>One of the most prevalent radiative transfer models is <span class="smallcaps">Prospect</span> (leaf optical <u>pro</u>perties <u>spect</u>ra), initially introduced by <span class="citation" data-cites="jacquemoud1990prospect">Jacquemoud and Baret (<a href="#ref-jacquemoud1990prospect" role="doc-biblioref">1990</a>)</span>. Drawing upon the plate model by <span class="citation" data-cites="allen1969interaction">Allen et al. (<a href="#ref-allen1969interaction" role="doc-biblioref">1969</a>)</span>, it simulates both reflectance and transmittance of a plant leaf across wavelengths ranging from 400 to 2500 nm. The first version of the <span class="smallcaps">Prospect</span> model featured only three input variables: leaf structure (<span class="math inline">\(N\)</span>), leaf chlorophyll content (<span class="math inline">\(c_{a+b}\)</span>), and leaf water content (<span class="math inline">\(c_w\)</span>). Based on that, the model returns the expected reflectance (<span class="math inline">\(\boldsymbol \rho\)</span>) and transmittance (<span class="math inline">\(\boldsymbol \tau\)</span>) for each simulated wavelength (<span class="math inline">\(\lambda\)</span>). <span id="eq-prospect"><span class="math display">\[
f: [N, c_{a+b}, c_w] \in \mathbb R^{3} \rightarrow [\boldsymbol \rho, \boldsymbol \tau] \in \mathbb R^{2\times 2101}
\tag{14}\]</span></span> Over the past four decades, the initial <span class="smallcaps">Prospect</span> model has undergone several updates to encompass leaf mass, tannins, anthocyanins, and carotenoids <span class="citation" data-cites="fourty1996leaf jacquemoud1996estimating baret1997estimation fourty1998spectral">(<a href="#ref-fourty1996leaf" role="doc-biblioref">Fourty et al. 1996</a>, <a href="#ref-jacquemoud1996estimating" role="doc-biblioref">Jacquemoud et al. 1996</a>, <a href="#ref-baret1997estimation" role="doc-biblioref">Baret and Fourty 1997</a>, <a href="#ref-fourty1998spectral" role="doc-biblioref">Fourty and Baret 1998</a>)</span>. For a long period, protein and other non-protein carbon-based compounds were consolidated into a single variable due to the challenge of disentangling them <span class="citation" data-cites="jacquemoud2000comparison">(<a href="#ref-jacquemoud2000comparison" role="doc-biblioref">Jacquemoud et al. 2000</a>)</span>.</p>
<p>In a notable development, the newest iteration of the <span class="smallcaps">Prospect</span> model, <span class="smallcaps">Prospect-Pro</span>, <span class="citation" data-cites="feret2021prospect">Féret et al. (<a href="#ref-feret2021prospect" role="doc-biblioref">2021</a>)</span> successfully disentangled these components by splitting the earlier used variable of leaf mass into protein content and other carbon-based content, which include cellulose, lignin, hemicellulose, starch, and sugars. <span class="smallcaps">Prospect-Pro</span> was calibrated and validated using the leaf optical properties experiment (<span class="smallcaps">Lopex</span>) dataset, which comprises measurements from over fifty plant species collected in northern Italy <span class="citation" data-cites="hosgood1995leaf">(<a href="#ref-hosgood1995leaf" role="doc-biblioref">Hosgood et al. 1995</a>)</span>. This model incorporates seven distinct classes of absorbing compounds, collectively referred to as vector <span class="math inline">\(\mathbf c\)</span>. <span id="eq-prospectpro"><span class="math display">\[
f: [N, \mathbf c, \alpha] \in \mathbb R^{9} \rightarrow [\boldsymbol \rho, \boldsymbol \tau] \in \mathbb R^{2\times 2101}
\tag{15}\]</span></span> <span class="smallcaps">Prospect-Pro</span> also enables the adjustment of the solid angle <span class="math inline">\(\alpha\)</span> for incident light on the leaf surface, initially determined to be <span class="math inline">\(\alpha = 60^\circ\)</span> by <span class="citation" data-cites="jacquemoud1990prospect">Jacquemoud and Baret (<a href="#ref-jacquemoud1990prospect" role="doc-biblioref">1990</a>)</span> using a limited dataset. Subsequently, <span class="citation" data-cites="feret2008prospect">Féret et al. (<a href="#ref-feret2008prospect" role="doc-biblioref">2008</a>)</span> discovered that <span class="math inline">\(\alpha = 40^\circ\)</span> was a more accurate value. An overview of the parameters used in <span class="smallcaps">Prospect-Pro</span> can be found in <a href="#tbl-prospect">table&nbsp;2</a>.</p>
<div id="tbl-prospect" class="anchored">
<table class="table">
<caption>Table&nbsp;2: Input variables of the <span class="smallcaps">Prospect-Pro</span> model as documented by <span class="citation" data-cites="feret2022package">Féret and de Boissieu (<a href="#ref-feret2022package" role="doc-biblioref">2022a</a>)</span>. The rows 2 - 7 show the elements of the constituents vector <span class="math inline">\(\mathbf c\)</span>.</caption>
<colgroup>
<col style="width: 4%">
<col style="width: 15%">
<col style="width: 67%">
<col style="width: 15%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Abbreviation</strong></th>
<th></th>
<th><strong>Parameter</strong></th>
<th><strong>Unit</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(N\)</span></td>
<td><code>N</code></td>
<td>Leaf mesophyll structure parameter</td>
<td></td>
</tr>
<tr class="even">
<td><br>
<br>
<br>
<span class="math inline">\(\mathbf c\)</span></td>
<td><code>CHL</code><br>
<code>CAR</code><br>
<code>ANT</code><br>
<code>BROWN</code><br>
<code>EWT</code><br>
<code>PROT</code><br>
<code>CBC</code></td>
<td>Chlorophyll content<br>
Carotenoid content<br>
Anthocyanin content<br>
Tannin content<br>
Equivalent water thickness<br>
Protein content<br>
(Non-protein) carbon-based constituent content</td>
<td><span class="math inline">\(\mu \text{g cm}^{-2}\)</span><br>
<span class="math inline">\(\mu \text{g cm}^{-2}\)</span><br>
<span class="math inline">\(\mu \text{g cm}^{-2}\)</span><br>
<span class="math inline">\(-\)</span><br>
<span class="math inline">\(\text{g cm}^{-2}\)</span><br>
<span class="math inline">\(\text{g cm}^{-2}\)</span><br>
<span class="math inline">\(\text{g cm}^{-2}\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\alpha\)</span></td>
<td><code>alpha</code></td>
<td>Solid angle for incident light at surface of leaf</td>
<td><span class="math inline">\(^{\circ}\)</span></td>
</tr>
</tbody>
</table>
</div>
<p>The <span class="smallcaps">Prospect</span> calculations rely on well-established laws of physics, with the Fresnel equations serving as the foundation. These equations describe the reflection and transmission of electromagnetic radiation between distinct optical media <span class="citation" data-cites="fresnel1821">(<a href="#ref-fresnel1821" role="doc-biblioref">Fresnel 1821</a>)</span>. Both <span class="math inline">\(s\)</span> and <span class="math inline">\(p\)</span> polarizations are included in the equations, which necessitate a refractive index <span class="math inline">\(n\)</span> — assumed to be <span class="math inline">\(n &gt; 1\)</span> for the leaf and <span class="math inline">\(n = 1\)</span> for air — and an angle of incidence <span class="math inline">\(\theta\)</span>. <span id="eq-fresnel"><span class="math display">\[
t_s = \frac{4\sqrt{1-\sin^2 \theta} \sqrt{n^2-\sin^2 \theta}}{\left(\sqrt{1-\sin^2 \theta} + \sqrt{n^2-\sin^2 \theta}\right)^2}
\quad
t_p = \frac{4n^2\sqrt{1-\sin^2 \theta} \sqrt{n^2-\sin^2 \theta}}{\left(n^2\sqrt{1-\sin^2 \theta} + \sqrt{n^2-\sin^2 \theta}\right)^2}
\tag{16}\]</span></span> As the focus lies on the transmissivity for diffuse light originating from all angles <span class="math inline">\(\theta\)</span> between 0 and <span class="math inline">\(\frac{\pi}{2}\)</span>, it is necessary to integrate across all possible angles to obtain the average transmissivity <span class="math inline">\(\overline{T}\)</span>. <span id="eq-TAVunintegrated"><span class="math display">\[
\overline{T} = \frac{\int_0^{2\pi} d\varphi \int_0^{\alpha} t(\theta,n) \cos \theta \sin\theta d \theta}{\int_0^{2\pi} d\varphi \int_0^{\alpha}  \cos \theta \sin\theta d \theta}
\tag{17}\]</span></span> The integration for the special case where the maximum incidence angle <span class="math inline">\(\alpha\)</span> equals <span class="math inline">\(\frac{\pi}{2}\)</span> was performed by <span class="citation" data-cites="stern1964transmission">Stern (<a href="#ref-stern1964transmission" role="doc-biblioref">1964</a>)</span>. Subsequently, <span class="citation" data-cites="allen1973transmission">Allen (<a href="#ref-allen1973transmission" role="doc-biblioref">1973</a>)</span> conducted the integration of the average transmissivity <span class="math inline">\(\overline{T}\)</span> for a generalized angle of incidence <span class="math inline">\(\alpha\)</span>. This approach is currently employed in the <span class="smallcaps">Prospect-Pro</span> computation <span class="citation" data-cites="feret2022package">(<a href="#ref-feret2022package" role="doc-biblioref">Féret and de Boissieu 2022a</a>)</span>. The expression for average transmissivity is rather complex, as to be seen in <a href="#eq-TavTsTp">equation&nbsp;18</a>. A visual representation of how the average transmissivity varies as a function of the refractive index <span class="math inline">\(n\)</span> and the angular range <span class="math inline">\(\alpha\)</span> can be found in <a href="#fig-SAC">figure&nbsp;6</a>, displayed at the bottom.</p>
<div id="fig-LeafStructure" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/LeafStructure.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;5: The <span class="smallcaps">Prospect</span> model simplifies a leaf’s structure by representing it as a flat plane, rather than accounting for its complex three-dimensional form, which consists of various components and a textured surface (<strong>A</strong>, created using content by <span class="citation" data-cites="wheeler2011">Wheeler (<a href="#ref-wheeler2011" role="doc-biblioref">2011</a>)</span>). In this sketch inspired by <span class="citation" data-cites="jacquemoud1992utilisation">Jacquemoud (<a href="#ref-jacquemoud1992utilisation" role="doc-biblioref">1992</a>)</span> (<strong>B</strong>), the different angles incorporated in <a href="#eq-TAVunintegrated">equation&nbsp;17</a> are illustrated: the general angles for azimuth (<span class="math inline">\(\varphi\)</span>) and zenith (<span class="math inline">\(\theta\)</span>) as well as the angle of maximum incident (<span class="math inline">\(\alpha\)</span>). In addition to that, the arrows indicate the reflectance (<span class="math inline">\(\rho\)</span>) and transmittance (<span class="math inline">\(\tau\)</span>).</figcaption>
</figure>
</div>
<div style="page-break-after: always;"></div>
<p><span class="math display">\[
\begin{aligned}
T_s(\alpha, n) =
\frac{1}{96}\frac{(n^2-1)^4}{\left[\sqrt{\left(\sin^2 \alpha -\frac{1}{2} (n^2 + 1)\right)^2 -\frac{1}{4} (n^2 - 1)^2} - \sin^2 \alpha +\frac{1}{2}  (n^2 + 1)\right]^3} -
\\\
\frac{1}{4}\frac{n^4 - 2n^2 +1}{\sqrt{\left(\sin^2 \alpha -\frac{1}{2} (n^2 + 1)\right)^2 -\frac{1}{4} (n^2 - 1)^2} - \sin^2 \alpha + \frac{1}{2} (n^2 + 1)} -
\\\
\frac{1}{2} \left( \sqrt{\left(\sin^2 \alpha -\frac{1}{2} (n^2 + 1)\right)^2 -\frac{1}{4} (n^2 - 1)^2} - \sin^2 \alpha + \frac{1}{2} (n^2 + 1) \right) +
\\\
\frac{1}{12} \frac{(n^2 - 1)^4}{(n + 1)^6} + \frac{1}{2} \frac{(n^2-1)^2}{(n+1)^2} + \frac{1}{4} (n^2+2n+1)
\end{aligned}
\]</span></p>
<p><span class="math display">\[
\begin{aligned}
T_p(\alpha, n) =
\frac{-2 n^2}{(n^2 + 1)^2} \left[ \sqrt{\left(\sin^2 \alpha - \frac{1}{2} (n^2 + 1)\right)^2 - \frac{1}{4} (n^2 - 1)^2} - \sin^2 \alpha + \frac{1}{2} (n^2 + 1) - \frac{1}{2} (n + 1)^2 \right] -
\\\
\frac{2  n^2  (n^2 + 1)}{(n^2-1)^2}  \cdot \ln \left[ \sqrt{\left(\sin^2 \alpha - \frac{1}{2} (n^2 + 1)\right)^2 - \frac{1}{4} (n^2 - 1)^2} -
\frac{\sin^2 \alpha - \frac{1}{2}(n^2 + 1)}{(n+1)^2}
\right] +
\\\
\frac{1}{2} n^2 \left\{ \left[ \sqrt{\left(\sin^2 \alpha - \frac{1}{2} (n^2 + 1)\right)^2 - \frac{1}{4} (n^2 - 1)^2} - \sin^2\alpha - \frac{1}{2} (n^2 + 1)\right]^{-1} - \frac{2}{(n + 1)^2} \right\} +
\\
\frac{16n^4 (n^4 + 1)}{(n^2+1)^3(n^2-1)^2} \cdot \ln \left[\frac{
2 (n^2 + 1)  \left(\sqrt{\left(\sin^2 \alpha - \frac{1}{2} (n^2 + 1)\right)^2} - \sin^2 \alpha + \frac{1}{2}  (n^2 + 1)\right)
- (n^2 - 1)^2}{(n^2 + 1)(n + 1)^2  - (n^2-1)^2} \right] +
\\\
\frac{16 n^6}{(n^2 + 1)^3} \left[
{2 (n^2 + 1)  \left(\sqrt{\left(\sin^2 \alpha - \frac{1}{2} (n^2 + 1)\right)^2} - \sin^2 \alpha + \frac{1}{2}  (n^2 + 1)\right)
- (n^2 - 1)^2} \right]^{-1} -
\\\
\frac{16n^6}{(n^2 + 1)^3\left( (n^2 + 1)   (n + 1)^2 - (n^2 - 1)^2 \right)}
\end{aligned}
\]</span></p>
<p><span id="eq-TavTsTp"><span class="math display">\[
\overline{T}(\alpha, n) = \frac{T_s(\alpha, n) + T_p(\alpha, n)}{2 \cdot \sin^2 \alpha}
\tag{18}\]</span></span></p>
<p>In the <span class="smallcaps">Prospect-Pro</span> model, reflectance and transmittance depend on multiple variables, as well as numerous fixed, empirically estimated parameters. These parameters encompass the refractive index (<span class="math inline">\(n\)</span>) and specific absorption coefficients (<span class="math inline">\(s\)</span>), which all are individually fitted for each absorbing compound in <span class="math inline">\(\mathbf c\)</span> across a diverse range of plant species. Additionally, these parameters are estimated for every wavelength <span class="math inline">\(\lambda\)</span> between 400 and 2500 nm. As a result, <span class="smallcaps">Prospect-Pro</span> relies on a total of 8 × 2101 = 16’808 empirically estimated internal parameters. With these parameters in hand, the overall absorption coefficient can be computed as follows. <span id="eq-k"><span class="math display">\[
k_\lambda = \frac{1}{N} \sum_i c_i s_{i,\lambda}
\tag{19}\]</span></span> Here, <span class="math inline">\(c_i\)</span> represents the content of the <span class="math inline">\(i\)</span><sup>th</sup> constituent in the vector <span class="math inline">\(\mathbf c\)</span> (as detailed in <a href="#tbl-prospect">table&nbsp;2</a>), and <span class="math inline">\(s_i\)</span> denotes the specific absorption coefficient of the <span class="math inline">\(i\)</span><sup>th</sup> constituent and the <span class="math inline">\(\lambda\)</span><sup>th</sup> wavelength. Although the computation of reflectance and transmittance is not linear with respect to <span class="math inline">\(k\)</span>, it is still possible to discern the bandwidths in which a particular compound’s presence would have an impact based on the specific absorption coefficients <span class="math inline">\(s\)</span> displayed in <a href="#fig-SAC">figure&nbsp;6</a>. Utilizing the overall absorption coefficient <span class="math inline">\(k\)</span>, the transmission for isotropic light <span class="math inline">\(\varphi\)</span> can be calculated for each wavelength <span class="math inline">\(\lambda\)</span> with <span class="math inline">\(k = k_\lambda\)</span> using the upper incomplete gamma function in accordance with the Beer-Lambert law <span class="citation" data-cites="lambert1760photometria beer1852bestimmung">(<a href="#ref-lambert1760photometria" role="doc-biblioref">Lambert 1760</a>, <a href="#ref-beer1852bestimmung" role="doc-biblioref">Beer 1852</a>)</span>. <span id="eq-phi"><span class="math display">\[
\varphi =
(1 - k) \cdot e^{-k} + k^2 \cdot \Gamma(k) = \\
(1 - k) \cdot e^{-k} + k^2 \cdot \int_k^\infty \frac{e^{-t}}{t} dt
\tag{20}\]</span></span> In the case of <span class="smallcaps">Prospect-Pro</span>, the upper incomplete Gamma function <span class="math inline">\(\Gamma(k)\)</span> is simplified and computed according to <span class="citation" data-cites="expint">Goulet (<a href="#ref-expint" role="doc-biblioref">2016</a>)</span>. Using this information, the final equations for transmittance <span class="math inline">\(\tau\)</span> and reflectance <span class="math inline">\(\rho\)</span> can be expressed (<a href="#eq-rhoAndTau">Equation&nbsp;22</a>). Due to space limitations, the substituting terms <span class="math inline">\(A, B, C, D, E, F\)</span>, and <span class="math inline">\(G\)</span> are employed (<a href="#eq-substitutions">Equation&nbsp;21</a>). <span id="eq-substitutions"><span class="math display">\[
\begin{aligned}
A = \frac{ \varphi \: \overline T(90, n)} { n^2\left[1 - \varphi^2 \: \left(1 - n^{-2} \: \overline T(90, n) \right)^2\right] }
\qquad
B = \overline T(90, n) \: A
\\[15pt]
C = 1 - \overline T(90, n) + \varphi \: B \: \left(1 - \frac{\overline T(90, n)}{n^2}\right)
\\[15pt]
D = \sqrt{(1 + C + B) \: (1 + C - B) \: (1 - C + B) \: (1 - C - B)}
\qquad
E = \frac{1 + C^2 - B^2 + D}{2\,C}
\\[15pt]
F = \frac{1 - C^2 + B^2 + D}{2\,B}
\qquad
G = F^{N-1}
\qquad
H = \frac{E \, (G^2 - 1)}{E^2 \, G^2 - 1}
\end{aligned}
\tag{21}\]</span></span> <span id="eq-rhoAndTau"><span class="math display">\[
\rho = 1 - \overline T(\alpha, n) \left[ 1 + \varphi \, A \left(1 - \frac{\overline T(90, n)}{n^2}  \right) + \frac{A \, H \, B}{1 - H \, C}  \right]
\quad
\tau =  \frac{\overline T(\alpha, n) \, A \, G \, (E^2 - 1)}{(E^2 \, G^2 - 1) \, (1 - H \, C)}
\tag{22}\]</span></span></p>
<p>The impact of distinct compounds on absorption is demonstrated in <a href="#fig-ProspectModelOutput">figure&nbsp;7</a>. This figure vividly portrays the absorption of electromagnetic radiation by proteins in the short-wave-infrared spectrum, specifically within the range of 1500 to 2500 nm wavelength. In contrast to carbon-based compounds, proteins exhibit no absorption in the near-infrared or visible light spectra.</p>
<div id="fig-SAC" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/SpecPROSPECT.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;6: <em>Top:</em> This figure shows the refractive index (<strong>n</strong>) and the eight specific absorption coefficients (<span class="math inline">\(s\)</span>) for different constituents of the <span class="smallcaps">Prospect-Pro</span> model as functions of the wavelength <span class="math inline">\(\lambda\)</span>. This encompasses the parameters chlorophyll content (<strong>CHL</strong>), carotenoid content (<strong>CAR</strong>), anthocyanin content (<strong>ANT</strong>), brown pigment content (<strong>BROWN</strong>), equivalent water thickness (<strong>EWT</strong>), leaf mass per area (<strong>LMA</strong>), protein content (<strong>PROT</strong>), and (non-protein) carbon-based constituents content (<strong>CBC</strong>). <em>Bottom:</em> The average transmissivity <span class="math inline">\(\overline{T}\)</span> of a plane dielectric surface for different values of the refractive index <span class="math inline">\(n\)</span> over the angular range <span class="math inline">\(\alpha\)</span>. Adapted from <span class="citation" data-cites="jacquemoud1990prospect">Jacquemoud and Baret (<a href="#ref-jacquemoud1990prospect" role="doc-biblioref">1990</a>)</span>.</figcaption>
</figure>
</div>
<div id="fig-ProspectModelOutput" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/ProspectModelOutput.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;7: This figure illustrates the output of the <span class="smallcaps">Prospect-Pro</span> model. Every plot shows an overlay of 20 model runs where only one variable is changing <em>ceteris paribus</em>. The three variables illustrated are chlorophyll content (<strong>A</strong>: <code>CHL</code> <span class="math inline">\(\in [10,70]\; \mu\text{g} \: \text{cm}^{-2}\)</span>), (non-protein) carbon-based content (<strong>B</strong>: <code>CBC</code> <span class="math inline">\(\in [0,0.02] \; \text{g} \: \text{cm}^{-2}\)</span>), and protein content (<strong>C</strong>: <code>PROT</code> <span class="math inline">\(\in [0,0.02] \; \text{g} \: \text{cm}^{-2}\)</span>).</figcaption>
</figure>
</div>
</section>
<section id="canopy-level-radiative-transfer-models" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="canopy-level-radiative-transfer-models"><span class="header-section-number">4.2</span> Canopy-level radiative transfer models</h2>
<p>The <span class="smallcaps">Prospect</span> model is specifically designed to simulate the reflectance and transmittance of an individual leaf. However, simulating the reflectance of an entire canopy proves to be far more intricate. Canopy reflectance is affected by numerous factors such as leaf inclination, solar irradiance angle, viewing angle of the canopy, and the interference of soil reflectance as well as atmospheric effects. To effectively account for these diverse variables, substantial extensions are necessary, which ultimately results in a significantly more complex simulation process. Consequently, this section outlines several approaches for utilizing the <span class="smallcaps">Prospect</span> model at the canopy level.</p>
<section id="one-dimensional-canopy-reflectance-models" class="level3">
<h3 class="anchored" data-anchor-id="one-dimensional-canopy-reflectance-models">One-dimensional canopy reflectance models</h3>
<p>The radiative transfer model <span class="smallcaps">Sail</span> (<u>s</u>cattering by <u>a</u>rbitrary <u>i</u>nclined <u>l</u>eaves) was one of the earliest models to simulate the bidirectional reflectance of a canopy. <span class="smallcaps">Sail</span> makes reflectance a function of both solar incident and viewing angles <span class="citation" data-cites="verhoef1984light verhoef1985earth">(<a href="#ref-verhoef1984light" role="doc-biblioref">Verhoef 1984</a>, <a href="#ref-verhoef1985earth" role="doc-biblioref">1985</a>)</span>. As implied by its name, <span class="smallcaps">Sail</span> represents the canopy through the simulation of numerous randomly inclined leaves situated above the soil. These leaves are assumed to be uniformly distributed in space but inclined according to a specific leaf inclination angle distribution function.</p>
<p>The most recent iteration of the <span class="smallcaps">Sail</span> model is known as <span class="smallcaps">4Sail</span>. While it simulates the same process as the original version, <span class="smallcaps">4Sail</span> is more numerically robust and optimized for enhanced speed <span class="citation" data-cites="verhoef2007unified">(<a href="#ref-verhoef2007unified" role="doc-biblioref">Verhoef et al. 2007</a>)</span>. The parameters used in the <span class="smallcaps">Sail</span> model can be found in <a href="#tbl-SAIL">table&nbsp;3</a>.</p>
<div id="tbl-SAIL" class="anchored">
<table class="table">
<caption>Table&nbsp;3: Input variables of the <span class="smallcaps">4Sail</span> model as documented by <span class="citation" data-cites="feret2022prosail">Féret and de Boissieu (<a href="#ref-feret2022prosail" role="doc-biblioref">2022b</a>)</span>.</caption>
<colgroup>
<col style="width: 20%">
<col style="width: 70%">
<col style="width: 10%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;"><strong>Parameter</strong></th>
<th style="text-align: left;"><strong>Unit</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>lai</code><br>
<code>q</code><br>
<span class="math inline">\(\rho_{\text{soil}}\)</span><br>
<span class="math inline">\(\alpha_{\text{soil}}\)</span><br>
<code>skyl</code><br>
<span class="math inline">\(\theta_{\text{s}}\)</span><br>
<span class="math inline">\(\theta_{\text{v}}\)</span><br>
<code>rAA</code></td>
<td style="text-align: left;">Leaf area index<br>
Hot spot parameter<br>
Soil reflectance<br>
Soil brightness factor<br>
Fraction of diffuse illumination<br>
Sun zenith angle<br>
Viewing (observer) zenith angle<br>
Relative azimuth angle between sun and sensor</td>
<td style="text-align: left;"><span class="math inline">\(\text{m}^2 \; \text{m}^{-2}\)</span><br>
<span class="math inline">\(\text{m} \; \text{m}^{-1}\)</span><br>
<span class="math inline">\(\%\)</span><br>
<br>
<br>
<span class="math inline">\(^\circ\)</span><br>
<span class="math inline">\(^\circ\)</span><br>
<span class="math inline">\(^\circ\)</span></td>
</tr>
</tbody>
</table>
</div>
<p>In order to achieve accurate modelling with the <span class="smallcaps">Sail</span> model, specifying the reflectance and transmittance of leaves is crucial. These values can either be measured directly and integrated into the model or derived from the <span class="smallcaps">Prospect</span> model’s predictions. The coupling of these models was first accomplished by <span class="citation" data-cites="baret1992modeled">Baret et al. (<a href="#ref-baret1992modeled" role="doc-biblioref">1992</a>)</span> following the development of the initial <span class="smallcaps">Prospect</span> model. This combined model is known as <span class="smallcaps">Prosail</span>, and its latest implementation unifies <span class="smallcaps">4Sail</span> and <span class="smallcaps">Prospect-Pro</span> <span class="citation" data-cites="jacquemoud2009prospect">(<a href="#ref-jacquemoud2009prospect" role="doc-biblioref">Jacquemoud et al. 2009</a>)</span>. <span class="smallcaps">Prosail</span> provides an effective means of extending plant pigment retrieval techniques from leaf-level analysis to canopy-level assessment, thereby enriching the scope of vegetation studies.</p>
<p>The <span class="smallcaps">Prosail</span> model simplifies light transport through a canopy into a one-dimensional process, in which light can travel either downwards or upwards. The canopy is assumed to consist of a homogeneous distribution of leaves. This simplification imposes certain limitations on the applicability of <span class="smallcaps">Prosail</span> and similar one-dimensional radiative transfer models <span class="citation" data-cites="bailey2020one">(<a href="#ref-bailey2020one" role="doc-biblioref">Bailey et al. 2020</a>)</span>. Although the assumption proves reasonably effective for modelling canopies that can be approximated as a homogeneous mass, such as grass, wheat, or soybeans, it may not be well-suited for more heterogeneous canopies. As a result, the <span class="smallcaps">Prosail</span> model is not the most suitable choice for examining crops with intricate canopy structures such as almond trees — alternative models should be considered and explored for these specific applications.</p>
</section>
<section id="ray-tracing" class="level3">
<h3 class="anchored" data-anchor-id="ray-tracing">Ray tracing</h3>
<p>In order to simulate the impact of a specific three-dimensional canopy structure on the reflectance, a technique called ray tracing is required. Ray tracing is a method for modelling light transport and is widely employed in rendering algorithms to create realistic digital images. It is not the only rendering algorithm, nor is it the fastest, but it tends to deliver the most realistic results <span class="citation" data-cites="peddie2019ray">(<a href="#ref-peddie2019ray" role="doc-biblioref">Peddie 2019</a>)</span>.</p>
<p>The underlying reason for this is that ray tracing truthfully replicates the behaviour of light, offering a highly accurate simulation of its interactions within various environments. Within an environment containing various objects of interest, single rays of light are simulated, each assigned a specified (spectral) reflectance and transmittance. Each ray of light is represented by a direction and an energy level provided for every simulated spectral band. When a ray of light encounters an object, its direction changes according to a bidirectional reflectance distribution function (BRDF), and its energy level is reduced by multiplying it with the object’s surface reflectance. The ray of light is continuously losing energy, as an object’s reflectance is always <span class="math inline">\(\leq 1\)</span>. Ultimately, the ray either exits the modelled environment or strikes the simulated sensor, contributing to the values of simulated pixels.</p>
<p>Forward path tracing is a technique in which ray tracing originates at the light source and may conclude at the simulated camera. The primary drawback of this approach is that most light rays leave the modelled environment without impacting the simulated image at all. Hence, a more computationally efficient strategy begins at the sensor position, sending light rays into the scene while calculating the appropriate spectral intensity for each individual ray. This process is known as backwards path tracing <span class="citation" data-cites="LessUserManual2022">(<a href="#ref-LessUserManual2022" role="doc-biblioref">Jianbo 2022</a>)</span>.</p>
<p>Ray tracing has advanced significantly over the last few decades, particularly due to the improvement efforts of the video game and computer animation industries. The implementation of many tricks such as backwards path tracing has constantly improved the performance of ray tracing algorithms. However, research has primarily focused on methods for increasing algorithm speed <span class="citation" data-cites="wyman2021rearchitecting lin2022generalized">(<a href="#ref-wyman2021rearchitecting" role="doc-biblioref">Wyman and Panteleev 2021</a>, <a href="#ref-lin2022generalized" role="doc-biblioref">Lin et al. 2022</a>)</span>. Accurate simulation of optical processes has been of lesser importance, as long as the resulting images appear visually pleasing to humans. Consequently, many ray tracing algorithms developed for computer graphics cannot be directly applied to simulating canopy reflectance <span class="citation" data-cites="LessUserManual2022">(<a href="#ref-LessUserManual2022" role="doc-biblioref">Jianbo 2022</a>)</span>.</p>
</section>
<section id="three-dimensional-canopy-reflectance-models" class="level3">
<h3 class="anchored" data-anchor-id="three-dimensional-canopy-reflectance-models">Three-dimensional canopy reflectance models</h3>
<p>Specialized programs are necessary to carry out the ray tracing calculations required for scientific applications. One such program is the Less{.smallcaps} model, specifically designed for simulating forest canopy reflectance. <span class="smallcaps">Less</span> (<u>l</u>arge-scale <u>e</u>mulation <u>s</u>ystem for forest <u>s</u>imulation) was developed by <span class="citation" data-cites="qi2017large">Qi et al. (<a href="#ref-qi2017large" role="doc-biblioref">2017</a>)</span> in 2017 and has been continuously maintained and updated in subsequent years <span class="citation" data-cites="qi2019less qi20223d qi2023fast">(<a href="#ref-qi2019less" role="doc-biblioref">Qi et al. 2019</a>, <a href="#ref-qi20223d" role="doc-biblioref">2022</a>, <a href="#ref-qi2023fast" role="doc-biblioref">2023</a>)</span>. The model employs a backward path tracing method to generate large-scale spectral images, ranging from the visible to the thermal infrared spectral domain <span class="citation" data-cites="LessUserManual2022">(<a href="#ref-LessUserManual2022" role="doc-biblioref">Jianbo 2022</a>)</span>.</p>
<p>Access to <span class="smallcaps">Less</span> is provided through a freely available graphical user interface (GUI), which facilitates the arrangement of three-dimensional objects within a scene and allows for pre-rendering inspection. Each component in the modelled scene, such as leaves, stems, or soil, can be assigned specific optical properties — both reflectance and transmittance — that influence ray tracing. However, all simulated surfaces are assumed to be Lambertian, or perfectly matte, meaning that no bidirectional reflectance distribution function can be implemented for leaves in <span class="smallcaps">Less</span> <span class="citation" data-cites="qi2017large">(<a href="#ref-qi2017large" role="doc-biblioref">Qi et al. 2017</a>)</span>.</p>
<p>Nevertheless, <span class="smallcaps">Less</span> enables the configuration of numerous parameters, including zenith and azimuth angles for both sensor view and solar irradiance, sensor height, cloudiness type, and the desired quality of the produced image.</p>
<p>It is worth noting that <span class="smallcaps">Less</span> is not the only model capable of rendering a three-dimensional canopy for scientific purposes. Another example is the ray tracing model <span class="smallcaps">Dart</span> (<u>d</u>iscrete <u>a</u>nisotropic <u>r</u>adiative <u>t</u>ransfer), introduced by <span class="citation" data-cites="malenovsky2013retrieval">Malenovskỳ et al. (<a href="#ref-malenovsky2013retrieval" role="doc-biblioref">2013</a>)</span>.</p>
</section>
</section>
<section id="the-inversion-problem" class="level2 page-columns page-full" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="the-inversion-problem"><span class="header-section-number">4.3</span> The inversion problem</h2>
<p>Thus far, the discussion has centred on radiative transfer models as a method for simulating leaf or canopy reflectance and transmittance, assuming that parameters such as chlorophyll, anthocyanin, or carotenoid content are already known. This is referred to as the forward mode of a radiative transfer model. In some cases, the forward mode of a model might be useful in and by itself, most notably for generating synthetic data. Such data can be used for the intercomparison of other models, to conduct a sensitivity analysis, or even to find new vegetation indices <span class="citation" data-cites="cheng2014estimating jin2014newly">(<a href="#ref-cheng2014estimating" role="doc-biblioref">Cheng et al. 2014</a>, <a href="#ref-jin2014newly" role="doc-biblioref">Jin et al. 2014</a>)</span>.</p>
<p>However, in many cases, reflectance information is already available, and the objective shifts to identifying the set of parameters that, given a specific model, best explain the observed data. This is referred to as the <em>backward mode</em> of a radiative transfer model. The goal is to discover the inverse function <span class="math inline">\(f^{-1}\)</span> of the actual model <span class="math inline">\(f\)</span>. <span id="eq-inversion"><span class="math display">\[
f: x \to y \qquad \qquad f^{-1}: y\to x
\tag{23}\]</span></span> Upon successful model inversion, the output of the model becomes its input and vice versa. Unfortunately, inverting radiative transfer models such as <span class="smallcaps">Prospect</span> is considered mathematically ill-posed and cannot be accomplished algebraically by simply re-writing the equations <span class="citation" data-cites="daun2017">(<a href="#ref-daun2017" role="doc-biblioref">Daun 2017</a>)</span>. Consequently, alternative methods must be employed to achieve the model inversion. Specifically, the inversion of a radiative transfer model can be achieved through mathematical optimization, look-up tables, or machine learning techniques.</p>
<section id="mathematical-optimization" class="level3 page-columns page-full">
<h3 class="anchored" data-anchor-id="mathematical-optimization">Mathematical optimization</h3>
<div class="page-columns page-full"><p>The conventional approach to inverting <span class="smallcaps">Prospect-Pro</span> involves mathematical optimization, specifically sequential quadratic programming <span class="citation" data-cites="feret2022package borchers2022pracma">(<a href="#ref-borchers2022pracma" role="doc-biblioref">Borchers 2022</a>, <a href="#ref-feret2022package" role="doc-biblioref">Féret and de Boissieu 2022a</a>)</span>. This iterative optimization algorithm seeks to identify the parameters <span class="math inline">\(\boldsymbol \theta\)</span> for the model <span class="math inline">\(f\)</span> in order to minimize a cost function <span class="math inline">\(J\)</span>. Typically, the mean squared error is employed as the cost function, which represents the sum of the squared difference between the model’s predictions (<span class="math inline">\(f(\boldsymbol {\theta})_\lambda\)</span>) and the observed reflectance and transmittance data. <span id="eq-minimization"><span class="math display">\[
\boldsymbol {\hat \theta}
= \arg \min_{\boldsymbol {\theta}} J(\boldsymbol \theta, \boldsymbol \rho) =
\arg \min_{\boldsymbol {\theta}} \sum_{\lambda=1}^{n} \left( f(\boldsymbol {\theta})_\lambda - \rho_\lambda \right)^2
\tag{24}\]</span></span> Note that the formula in <a href="#eq-minimization">equation&nbsp;24</a> only relates to reflectance but works analogous for transmittance as well. The optimization algorithm operates by iteratively running the model, numerically approximating the gradient (<span class="math inline">\(\nabla_J\)</span>) and the Hessian (<span class="math inline">\(\mathbf H_J\)</span>) of the cost function <span class="math inline">\(J\)</span>, and utilizing this information to adjust the parameters. This iteration is stopped when the gradient becomes sufficiently small (<span class="math inline">\(\nabla_J \approx \mathbf 0\)</span>). Inverting the <span class="smallcaps">Prospect-Pro</span> model necessitates approximately 132 model runs on average<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a> to achieve a tolerance level of 10<sup>−6</sup>.</p><div class="no-row-height column-margin column-container"><li id="fn3"><p><sup>3</sup>&nbsp;This statement highlights that the average number of model evaluations required for inverting the <span class="smallcaps">Prospect-Pro</span> model on the almond dataset was 132. While this number may vary for different datasets, the differences are not expected to be extreme, remaining within the same order of magnitude.</p></li></div></div>
<p>As an alternative to the maximum likelihood estimation effectively executed via the minimization task in <a href="#eq-minimization">equation&nbsp;24</a>, a Bayesian framework could be employed to identify the model’s parameters <span class="citation" data-cites="daun2017">(<a href="#ref-daun2017" role="doc-biblioref">Daun 2017</a>)</span>. This method requires information on the prior distribution of the parameter to compute the posterior probability distribution. Although it is computationally more demanding than the maximum likelihood estimation, it offers the advantage of producing credibility intervals for the parameters based on their joint distribution. For instance, <span class="citation" data-cites="zhang2005estimating">Zhang et al. (<a href="#ref-zhang2005estimating" role="doc-biblioref">2005</a>)</span> and more recently <span class="citation" data-cites="shiklomanov2021cutting">Shiklomanov et al. (<a href="#ref-shiklomanov2021cutting" role="doc-biblioref">2021</a>)</span> employed a Bayesian calibration method to invert the <span class="smallcaps">Prosail-2</span> and <span class="smallcaps">Prospect-5</span> models, respectively. Both teams utilized a Markov Chain Monte Carlo (MCMC) algorithm to achieve this inversion.</p>
<p>There are three primary drawbacks associated with employing mathematical optimization as an inversion technique for radiative transfer models. The first drawback concerns the feasibility of the approach, which diminishes as the computational cost for a single evaluation of the model becomes too large. This typically occurs when the radiative transfer model grows increasingly complex and computationally demanding, thereby reducing the efficacy of mathematical optimization as an inversion method. The second disadvantage emerges when handling a substantial number of observations that necessitate individual parameter retrieval. In these instances, mathematical optimization loses its effectiveness, as the information obtained from inverting the model for one observation cannot be utilized for another. The final concern involves the risk of becoming trapped in a local minimum, contingent upon the selection of initial parameters. If these limitations prove to be insurmountable, it is prudent to explore alternative solutions such as look-up tables or machine learning techniques, which might offer greater efficiency and effectiveness in addressing such challenges. Consequently, for the inversion of more intricate models like <span class="smallcaps">Less</span>, mathematical optimization is not the most suitable approach.</p>
</section>
<section id="look-up-tables" class="level3">
<h3 class="anchored" data-anchor-id="look-up-tables">Look-up tables</h3>
<p>Look-up tables serve as a particularly valuable model inversion technique when the model in question is computationally expensive, yet its results can be cached easily <span class="citation" data-cites="selan2005using">(<a href="#ref-selan2005using" role="doc-biblioref">Selan 2005</a>)</span>. Typically implemented as a fixed-size array, a look-up table consists of elements, each of which corresponds to a specific set of parameters. In the case of <span class="smallcaps">Prospect-Pro</span>, these elements represent the combined reflectance and transmittance data. This structure enables relatively rapid data access; however, it also necessitates pre-calculation and storage of the entire look-up table, encompassing all possible parameter combinations <span class="citation" data-cites="berger2018evaluation">(<a href="#ref-berger2018evaluation" role="doc-biblioref">Berger et al. 2018</a>)</span>.</p>
<p>The primary disadvantage of look-up tables is the potentially extensive time required for pre-calculating all elements, as well as the necessity of accessing the entire table — which may represent an exceptionally large data structure — for making predictions. It is important to highlight that the data within look-up tables often displays a high degree of collinearity. As a result, a machine learning model can be trained on the look-up table, significantly reducing its size, and thus facilitating the distribution of the inverted model and also the inference process <span class="citation" data-cites="verrelst2016active">(<a href="#ref-verrelst2016active" role="doc-biblioref">Verrelst et al. 2016</a>)</span>.</p>
</section>
<section id="model-inversion-via-machine-learning" class="level3">
<h3 class="anchored" data-anchor-id="model-inversion-via-machine-learning">Model inversion via machine learning</h3>
<p>Inverting models through machine learning can be achieved by first generating a synthetic data set using the radiative transfer model, and then training a machine learning model on that synthetic data set. In this regard, artificial neural networks have gained particular prominence. As the number of observations rises, neural networks tend to outperform other machine learning models. Given the synthetic nature of the data, there is virtually no constraint on the number of observations <span class="citation" data-cites="kimes2000inversion verger2011optimal">(<a href="#ref-kimes2000inversion" role="doc-biblioref">Kimes et al. 2000</a>, <a href="#ref-verger2011optimal" role="doc-biblioref">Verger et al. 2011</a>)</span>. Moreover, artificial neural networks offer the advantage of producing relatively quick predictions during the inference stage once they are trained. For instance, the canopy level radiative transfer model <span class="smallcaps">Prosail</span> has been successfully inverted for crop trait retrieval using neural networks <span class="citation" data-cites="verger2011optimal">(<a href="#ref-verger2011optimal" role="doc-biblioref">Verger et al. 2011</a>)</span>, but also using random forest regression <span class="citation" data-cites="doktor2014extraction">(<a href="#ref-doktor2014extraction" role="doc-biblioref">Doktor et al. 2014</a>)</span>.</p>
<p><span class="citation" data-cites="verrelst2015replacing">Verrelst et al. (<a href="#ref-verrelst2015replacing" role="doc-biblioref">2015</a>)</span> refer to machine learning models that approximate the inverse of a radiative transfer model as emulators. The authors argue that the most significant advantages of emulators are their ability to interpolate effectively in regions where the radiative transfer model has not been executed and their rapid prediction capabilities compared to methods like look-up tables.</p>
</section>
</section>
</section>
<section id="sec-data" class="level1 page-columns page-full" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Describing, processing and inspecting the data</h1>
<p>The data utilized for this thesis originates from the Westfield almond orchard, situated in Woodland, California, USA (38.66722 N, -121.88989 E). This orchard houses two major almond cultivars, specifically <em>Nonpareil</em> and <em>Monterey</em>, both of which were planted in 2010. An aerial image of the entire Westfield almond orchard spanning 60 ha is displayed in <a href="#fig-orthomosaic">figure&nbsp;8</a>. In the context of the project, ground truth data concerning foliar nitrogen concentration was gathered from 125 target trees sampled across the site, thereby supplying the information for the response variable.</p>
<div class="page-columns page-full"><p>This chapter outlines the methodologies implemented for data capture<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a> and offers an in-depth account of the data characteristics. In addition, I will clarify techniques and strategies employed in the pre-processing of both spectroradiometric data and hyperspectral images.</p><div class="no-row-height column-margin column-container"><li id="fn4"><p><sup>4</sup>&nbsp;It should be noted that the data capture occurred during the summer of 2022, prior to me joining the research group. However, for the sake of comprehensiveness, this chapter will provide a description of the data capture as well.</p></li></div></div>
<div id="fig-orthomosaic" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/Orthomosaic.jpg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;8: Orthomosaic of the almond orchard in Woodland, California, United States of America. This image was taken by Hamid Jafarbiglu in July 2022.</figcaption>
</figure>
</div>
<section id="description-of-the-data-set" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="description-of-the-data-set"><span class="header-section-number">5.1</span> Description of the data set</h2>
<p>Within the Westwind orchard, 125 trees were identified as target trees of interest. From each of these trees, Momtanu Chakraborty collected five to six leaves, which were subsequently weighed, photographed, dried, and weighed once more. The leaf area was calculated from the photographs. Both leaf area and leaf dry mass are crucial measurements for converting mass-based nitrogen (N<sub>%</sub>) to area-based nitrogen (N<sub>area</sub>) via the leaf mass per area (LMA, <a href="#eq-LMA">Equation&nbsp;3</a>). Subsequently, the dry leaves from each target tree were combined and ground using a Mixer Mill MM 400 laboratory grinder manufactured by Retsch (Haan, North Rhine-Westphalia, Germany). The pulverized samples were sent to the A&amp;L Western Laboratories (Modesto, CA 95351, USA) for nitrogen concentration analysis. In order to determine the nitrogen concentration of the samples, the laboratory employed Kjeldahl digestion.</p>
<p>Regarding the leaf level data, the Digital Agriculture Laboratory team measured the reflectance and transmittance of all 741 collected leaves utilizing a HR-1024i spectroradiometer manufactured by Spectra Vista Corporation (Poughkeepsie, NY 12603, USA). Consequently, spectroradiometric data was accessible for the five to six individual leaves per tree, while the nitrogen concentration was only available for the combined set of leaves.</p>
<p>The hyperspectral images were captured on the 12<sup>th</sup> of July 2022 by Momtanu Chakraborty and Hamid Jafarbiglu. They utilized the hyperspectral camera Pika L, manufactured by Resonon Inc.&nbsp;(Bozeman, MT 59715, USA), which was mounted on a <span class="smallcaps">Matrice 600</span> drone produced by DJI Technology Co.&nbsp;(Nanshan, Shenzhen, China). The hyperspectral camera gathers data via line-scanning, whereby hyperspectral images are recorded one line of pixels at a time using a two-dimensional dispersing element and a two-dimensional detector array <span class="citation" data-cites="elmasry2010principles">(<a href="#ref-elmasry2010principles" role="doc-biblioref">El Masry and Sun 2010</a>)</span>. Although line-scanning makes it more challenging to obtain a stable image, it helps avoid the hot spot effect, which may arise in multispectral images captured around noon <span class="citation" data-cites="jafarbiglu2023impact">(<a href="#ref-jafarbiglu2023impact" role="doc-biblioref">Jafarbiglu and Pourreza 2023</a>)</span>. The line-scanning sensor circumvents the hot-spot effect when the scanned line is perpendicular to the angle of solar incidence, i.e., in the east-west direction. Consequently, the hyperspectral images were captured in a north-south orientation.</p>
<p>An overview of both the leaf-level and canopy-level data is presented in <a href="#tbl-data">table&nbsp;4</a>. Initially, there were 125 target trees; however, one collection of leaves was lost, and the drone carrying the hyperspectral camera experienced a crash, leading to differing numbers of observations for the canopy-level data.</p>
<div id="tbl-data" class="anchored">
<table class="table">
<caption>Table&nbsp;4: Overview of the two datasets utilized. It is worth noting that the leaf-level data encompasses both reflectance and transmittance, with 1024 bands available for each.</caption>
<colgroup>
<col style="width: 19%">
<col style="width: 19%">
<col style="width: 19%">
<col style="width: 22%">
<col style="width: 22%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Level</strong></th>
<th><strong>Observations</strong></th>
<th><strong>Bands</strong></th>
<th><strong>Range</strong></th>
<th><strong>Sensor</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Leaf</td>
<td>124</td>
<td>1024</td>
<td>338 - 2512 nm</td>
<td>HR-1024i</td>
</tr>
<tr class="even">
<td>Canopy</td>
<td>107</td>
<td>151</td>
<td>385 - 900 nm</td>
<td>Pika L</td>
</tr>
</tbody>
</table>
</div>
</section>
<section id="sec-SRMprocessing" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="sec-SRMprocessing"><span class="header-section-number">5.2</span> Pre-processing the spectroradiometric data</h2>
<p>The leaf-level data acquired from the spectroradiometer required only minimal pre-processing. Any recorded values below zero were adjusted to zero in order to rectify discrepancies. Further processing included removing noisy and overlapping regions to enhance data clarity (as to be seen in <a href="#fig-SpectroradiometerSmoothing">figure&nbsp;9</a>). To align the reflectance and transmittance as closely as possible with the nitrogen labels, the weighted mean for every band was computed within each group of leaves from a single tree, whereby the measured leaf weights appropriately served as the statistical weights. The rationale behind this procedure is that heavier leaves have a more significant impact on the nitrogen label of their group, and thus should also exert a more substantial influence on the reflectance and transmittance data.</p>
<p>To conform to the requirements of the <span class="smallcaps">Prospect-Pro</span> model, the data was interpolated to specific wavelengths between 400 and 2500 nm. This interpolation was performed using local polynomial regression, utilizing the function <code>loess</code> from the R package <code>stats</code> with the parameter <code>span = 0.025</code> <span class="citation" data-cites="R2022">(<a href="#ref-R2022" role="doc-biblioref">R Core Team 2022</a>)</span>.</p>
<div style="page-break-after: always;"></div>
<div id="fig-SpectroradiometerSmoothing" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/SpectroradiometricDataPreProcessing.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;9: This example illustrates the interpolation process using local polynomial regression (<strong>A</strong>). The smoothing spline was fitted through the data captured by the spectroradiometer, which is unstable at certain points and overlapping at others. The bottom figures display the distribution of the interpolated data for reflectance (<strong>B</strong>) and transmittance (<strong>C</strong>), respectively — the red areas display the center 50%, 75%, 95%, and 99% of the data.</figcaption>
</figure>
</div>
</section>
<section id="sec-HSIprocessing" class="level2 page-columns page-full" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="sec-HSIprocessing"><span class="header-section-number">5.3</span> Pre-processing of the hyperspectral images</h2>
<p>The acquired hyperspectral images were saved as batches of ENVI-files (<code>.bip</code>, band inter-leaved by line). Each of these files is essentially a three-dimensional array containing up to 2000 × 900 × 200 values: there are 200 bands, 900 pixels per line, and (at most) 2000 lines scanned. Subsequent batches of lines scanned were written to another ENVI-file. Additionally, a single file of irradiance measurements was generated for each batch of ENVI-files captured. The raw data recorded by the sensors were unitless. To convert them to radiance and irradiance, the shutter speeds and the sensor’s sensitivity at each wavelength must be considered. Reflectance can then be calculated from irradiance and radiance as shown in <a href="#eq-reflectance">equation&nbsp;2</a>. All these calculations were performed using the <span class="smallcaps">Spectronon Pro</span> software by Resonon Inc.&nbsp;(Bozeman, MT 59715, USA). The resulting hyperspectral images contained reflectance data and thus were standardized between 0 and 1.</p>
<p>The following step involved the georeferencing the data. Georeferencing is to the process of manipulating geographic data such that it aligns with a known coordinate system, enabling analysis and comparison with other geographic data <span class="citation" data-cites="wade2006z">(<a href="#ref-wade2006z" role="doc-biblioref">Wade et al. 2006</a>)</span>. Consequently, each observational unit, in this case the hyperspectral data of every pixel, must be assigned a specific coordinate point. Orthorectification is particularly crucial in this context: aerial images must be processed to ensure they are geometrically accurate, akin to a geographic map. Orthorectification corrects for lens distortion, camera tilt, perspective, and topographic relief. This entire process was also conducted using <span class="smallcaps">Spectronon Pro</span>. However, the results were often unsatisfactory, as entire batches appeared distorted. The scanning line of the hyperspectral camera seemed to have varied its scanning angle relative to the flight direction during the drone flight, causing local image tilting. This tilting of the data was not something the georeferencing functions of <span class="smallcaps">Spectronon Pro</span> could address. To rectify the tilt, I opted to adjust the cropping polygons during the tree-cropping stage rather than improve the georeferencing, as this was a more straightforward technical solution.</p>
<div class="page-columns page-full"><p>The objective of tree cropping is to generate a separate hyperspectral image for each tree in the orchard. This step is essential for computational reasons, as it is impractical to work with the entire dataset simultaneously.<a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a> Dividing the hyperspectral data into multiple files allows subsequent processes to be applied to individual images of cropped trees. For this cropping procedure, polygons defining the precise cut-out area for each tree needed to be established. However, as the hyperspectral images were not always accurately georeferenced, they had to be manually compared with the orthomosaic shown in <a href="#fig-orthomosaic">figure&nbsp;8</a>, and the polygons had to be adjusted manually to align with the expected tree coordinates. Once all polygons were defined, the cropping was carried out using Python scripts originally authored by Ali Mohigimi and Yuto Kamiya, in addition to the QGIS software <span class="citation" data-cites="QGIS">(<a href="#ref-QGIS" role="doc-biblioref">QGIS Development Team 2022</a>)</span>.</p><div class="no-row-height column-margin column-container"><li id="fn5"><p><sup>5</sup>&nbsp;The combined hyperspectral images had an approximate file size of 150 GB, which is far too large for programs to handle all at once.</p></li></div></div>
<p>Once all images were cropped, the quality of the cropping was assessed. By employing a simple linear model on a transformation of the image data, it was possible to estimate whether the vegetation (determined by the NDVI) was centred in the cropped image. The fourth column in <a href="#fig-exampleCroppedImages">figure&nbsp;10</a> displays an instance where the tree is not centred, resulting in a linear model slope that is opposite in direction compared to the others. Consequently, the slope of this straightforward model could serve as an indicator of the centricity of the images, enabling to make necessary readjustments during the cropping stage.</p>
<p>The average cropped image had dimensions of 56 × 38 × 200, which translates to over 400,000 values per hyperspectral image. With only 124 observations, such a number of predictors is rather excessive. To address this, the dimensions of the hyperspectral images were reduced by eliminating all spatial dimensions, retaining only the spectral dimension. However, calculating the mean reflectance of the entire image for each spectral band would have diluted the signal due to interference from shadows and soil. Consequently, the cropped images were semantically segmented into bright leaves, obscure areas, and soil, with only the bright leaf areas contributing to the retrieved mean spectral reflectance. Semantic segmentation refers to the task of assigning a class to each pixel in an image, rather than to the image as a whole <span class="citation" data-cites="guo2018review">(<a href="#ref-guo2018review" role="doc-biblioref">Guo et al. 2018</a>)</span>. The semantic segmentation was performed using an extreme gradient boosting model with the R package <code>xgboost</code> <span class="citation" data-cites="chen2022">(<a href="#ref-chen2022" role="doc-biblioref">Chen et al. 2022</a>)</span>. To facilitate learning, various vegetation indices were calculated for each pixel and used as features alongside the greyscale values, as opposed to using the original features.</p>
<p>A training dataset comprising 690 observations from 23 cropped images was manually created for the machine learning model. The extreme gradient boosting model demonstrated an overall test accuracy of 0.94, with most of the error source stemming from confusion between soil and shadow. The bright leaf area exhibited a test precision of 0.99 and a test recall of 0.97, indicating that almost no non-leaf area was mislabelled as leaf, and very few leaf areas were labelled as non-leaf areas. Examples of the semantically segmented images can be viewed in <a href="#fig-exampleCroppedImages">figure&nbsp;10</a>.</p>
<p>The reflectance signal beyond 900 nm wavelength becomes highly unstable due to water vapour interference, as illustrated in <a href="#fig-Electromagnetic">figure&nbsp;3</a>, and does not align with the canopy’s expected magnitude of reflectance. Consequently, the data was further reduced by discarding any bands beyond 900 nm wavelength. This reduction meant that the reflectance of each tree was effectively condensed to a vector containing only 161 values.</p>
<div id="fig-exampleCroppedImages" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/Segmentation.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;10: Here, four examples of cropped trees are presented. From the original hyperspectral data, the NDVI, along with several other vegetation indices, was calculated (<strong>A</strong>). Using this information, an XGBoost model semantically segmented the cropped image (<strong>B</strong>). The slope of a linear regression model, fitted on the mean vertical NDVI values, indicated how well the cropped image was centred around the tree (<strong>C</strong>). From only the pixels classified as bright leaf area, the mean reflectance was calculated for further use (<strong>D</strong>).</figcaption>
</figure>
</div>
</section>
<section id="sec-normalization" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="sec-normalization"><span class="header-section-number">5.4</span> Z-score normalization</h2>
<p>In multiple instances so far, one could observe that the spectral signature of plants follows a specific pattern, consistently showing different magnitudes of reflectance for the same wavelengths (e.g., <a href="#fig-exampleCroppedImages">figure&nbsp;10</a> at the bottom). Light absorption is typically higher in the red and blue band areas, while it is lower in the green, red edge, and near-infrared areas. The variance of the collected data at each wavelength is not absolute but rather proportional to the reflectance’s magnitude. Importantly, the differing magnitudes of the variables do not correspond to their relative importance for predicting crop nitrogen. However, if left untreated, variables with larger magnitudes and variances may unjustly dominate certain models. A common strategy employed in machine learning to address this issue is z-score normalization. With respect to a single vector <span class="math inline">\(\mathbf x\)</span>, z-score normalization can be expressed as follows. <span id="eq-zScoreNormalization"><span class="math display">\[
\mathbf z = \frac{\mathbf x - \mu_x}{\sigma_x}
\tag{25}\]</span></span> Here, <span class="math inline">\(\mu_x\)</span> represents the mean of <span class="math inline">\(\mathbf x\)</span>, <span class="math inline">\(\sigma_x\)</span> denotes the standard deviation of <span class="math inline">\(\mathbf x\)</span>, and <span class="math inline">\(\mathbf z\)</span> is the z-score normalized vector. This procedure creates a vector <span class="math inline">\(\mathbf z\)</span> which has a mean of <span class="math inline">\(\mu_z = 0\)</span> and a variance of <span class="math inline">\(\sigma_z^2 = 1\)</span>, as illustrated graphically in <a href="#fig-zScoreNormalization">figure&nbsp;11</a>. Z-score normalization of each column of the input data matrix <span class="math inline">\(\mathbf X\)</span> offers multiple benefits. It results in a homogeneous treatment of the error and enables iterative optimization algorithms to converge more quickly in finding the parameters of machine learning models <span class="citation" data-cites="geladi1986partial ioffe2015batch">(<a href="#ref-geladi1986partial" role="doc-biblioref">Geladi and Kowalski 1986</a>, <a href="#ref-ioffe2015batch" role="doc-biblioref">Ioffe and Szegedy 2015</a>)</span>. Therefore, it streamlines the training process of machine learning models — particularly when gradient descent is employed in the training process <span class="citation" data-cites="nawi2013effect">(<a href="#ref-nawi2013effect" role="doc-biblioref">Nawi et al. 2013</a>)</span>.</p>
<p>In addition to addressing the issue of equal variable importance, z-score normalization enhances the interpretability of many models, as coefficients are no longer relative to the variable magnitude. For these reasons, z-score normalization was applied to both leaf-level and canopy-level data before proceeding to the modelling phase.</p>
<div id="fig-zScoreNormalization" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/zScoreNormalization.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;11: Visual explanation of z-score normalization of a dataset: The variables in the original data might exhibit significantly varying means and variances (<strong>A</strong>). Therefore, we can scale the variables with respect to their mean (<strong>B</strong>), their variance (<strong>C</strong>), or both (<strong>D</strong>). Adapted from <span class="citation" data-cites="geladi1986partial">Geladi and Kowalski (<a href="#ref-geladi1986partial" role="doc-biblioref">1986</a>)</span>.</figcaption>
</figure>
</div>
</section>
<section id="sec-exploratory" class="level2" data-number="5.5">
<h2 data-number="5.5" class="anchored" data-anchor-id="sec-exploratory"><span class="header-section-number">5.5</span> Exploratory data analysis</h2>
<p>In this section, a superficial exploration of the data sets, including both the predictor data listed in <a href="#tbl-data">table&nbsp;4</a> as well as the response, will be conducted.</p>
<p>All predictor data sets containing reflectance and transmittance data show extremely high multicollinearity. This becomes evident when examining the correlation matrices (<a href="#fig-exploratory">Figure&nbsp;12</a> A), where bright areas indicate highly collinear variables. The collinearity is further emphasized by a principal component analysis, in which the variance explained by additional principal components rapidly diminishes and falls below 1% after only the third or fourth principal component, depending on the data set (<a href="#fig-exploratory">Figure&nbsp;12</a> D).</p>
<p>An inspection of the correlations of single wavelength reflectance or transmittance to the nitrogen concentration in <a href="#fig-exploratory">figure&nbsp;12</a> C reveals that the strongest direct link occurs in the red light area at precisely 700 nm wavelength (<span class="math inline">\(r = -0.48\)</span>) for the leaf-level reflectance data. The correlation to nitrogen of the canopy-level reflectance data roughly matches the leaf-level reflectance data, although it is considerably less pronounced (<span class="math inline">\(r = -0.32\)</span>). The correlation in the red light area can be attributed to the absorption of red light by chlorophyll and its connection to nitrogen.</p>
<p>Interestingly, the near-infrared and particularly the short-wave-infrared areas (where the absorption spectra of protein reside) appear to be less informative about the foliar nitrogen concentration (<a href="#fig-exploratory">Figure&nbsp;12</a> C). Nevertheless, the information in these areas is independent of the information in the visible light area, as displayed by the correlation matrices for the leaf-level data (<a href="#fig-exploratory">Figure&nbsp;12</a> A).</p>
<p>Overall, the transmittance data exhibits a weaker correlation with nitrogen than the reflectance data. However, there are spikes of negative correlations in the areas where protein reflects minimally — consequently, the correlation is negative in these parts. A similar pattern can be observed for the reflectance data, albeit less pronounced.</p>
<div id="fig-exploratory" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/ExploratoryDataAnalysis.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;12: These plots present the correlation matrix (<strong>A</strong>), band-wise correlation with foliar nitrogen concentration (<strong>B</strong>), the z-score normalized data (<strong>C</strong>) and variance explained by the first few principal components (<strong>D</strong>) for the three data sets. The correlation matrices were generated using the absolute value of Pearson’s correlation coefficient <span class="math inline">\(r\)</span> to enhance readability. Black areas indicate no correlation, while white areas represent perfect correlation between bands — hence, the diagonal is always white.</figcaption>
</figure>
</div>
<p>Upon examining the pairs panel in <a href="#fig-PairsPanel">figure&nbsp;13</a>, the measurements reveal notable and positive correlations among wet weight, dry weight, and leaf area. Specifically, the correlation between the wet weight of the leaves and their corresponding leaf area is particularly solid (<span class="math inline">\(r=0.91\)</span>). This finding should hardly come as a surprise.</p>
<p>Additionally, it becomes evident that mass-based nitrogen (N<sub>%</sub>) and area-based nitrogen (N<sub>area</sub>) display a very low correlation (<span class="math inline">\(r = 0.093\)</span>). This observation is particularly surprising, given that the two traits can be converted into each other through leaf mass per area (LMA), as described by <a href="#eq-LMA">equation&nbsp;3</a>. Leaf mass per area, calculated from dry mass and leaf area, can be loosely interpreted as leaf thickness. It is strongly and positively correlated with N<sub>area</sub> (<span class="math inline">\(r = -0.85\)</span>), but exhibits a slight negative correlation with N<sub>%</sub> (<span class="math inline">\(r = -0.43\)</span>). This relationship can be attributed to a dilution effect caused by carbohydrates on nitrogen concentration (N<sub>%</sub>): higher carbohydrate concentrations result in a measured lower concentration of nitrogen, even if the absolute nitrogen content remains unchanged.</p>
<p>Consequently, <span class="citation" data-cites="baret2007quantification">Baret et al. (<a href="#ref-baret2007quantification" role="doc-biblioref">2007</a>)</span> suggested that N<sub>area</sub> should be the preferred metric. Nonetheless, the mass-based nitrogen N<sub>%</sub> remains the chief determinant in fertilization management decisions, as it is the most important part of <a href="#eq-NNI">equation&nbsp;1</a>. Therefore, I will primarily focus on the estimation of nitrogen concentration in leaves and canopy rather than the area-based nitrogen.</p>
<div id="fig-PairsPanel" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/PairsPanel.png" class="img-fluid figure-img" data-fig.pos="b"></p>
<figcaption class="figure-caption">Figure&nbsp;13: Pairs panel of the ground truth data, which includes measured leaf traits. The diagonal displays histograms for each variable, while the lower triangle presents pairwise scatterplots. The upper triangle showcases Pearson’s correlation coefficient <span class="math inline">\(r\)</span> and the corresponding p-values. Non-significant correlation coefficients and their p-values are printed in grey.</figcaption>
</figure>
</div>
</section>
</section>
<section id="sec-modelling" class="level1 page-columns page-full" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Technical description of the modelling</h1>
<p>In this chapter, the focus lies on exploring the process of modelling crop nitrogen at both the leaf and canopy levels. To guarantee a just and impartial comparison of the various modelling techniques, all models were provided with identical data — insofar they can make use of it. Furthermore, a machine learning paradigm was employed not solely for the training of machine learning models, but also for all models involved. As a result, certain parameters of the hand-crafted feature approach or the radiative transfer modelling approach were considered as hyperparameters and selected through cross-validation (this will be explained in more detail in <a href="#sec-hypertuning">section&nbsp;6.2</a>).</p>
<p>An extensive comparison was carried out, examining a total of 23 different models for their effectiveness. It is relevant to acknowledge that there is a considerable overlap among some of these models. A comprehensive list of the evaluated models can be found in <a href="#tbl-ModellingApproaches">table&nbsp;5</a>, where the method names in the third column are consistently used in subsequent plots and tables to simplify referencing and interpreting the results.</p>
<div id="tbl-ModellingApproaches" class="anchored">
<table class="table">
<caption>Table&nbsp;5: Here, an overview of the modelling techniques that were evaluated is presented. It is worth noting that the precise definitions of the vegetation indices mentioned below are available in <a href="#tbl-VegetationIndices">table&nbsp;1</a>.</caption>
<colgroup>
<col style="width: 30%">
<col style="width: 30%">
<col style="width: 40%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Approach</strong></th>
<th><strong>Group</strong></th>
<th><strong>Method</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Hand-crafted features</td>
<td>Single vegetation indices</td>
<td>NDVI<br>
NDRE<br>
EVI<br>
GCI<br>
GNDVI<br>
MCARI<br>
RECI<br>
TCARI<br>
WDRVI<br>
R434</td>
</tr>
<tr class="even">
<td></td>
<td>Stepwise selection</td>
<td>Combined vegetation indices<br>
Recursive feature selection<br>
Square feature selection</td>
</tr>
<tr class="odd">
<td>Machine learning</td>
<td>Regularized least squares</td>
<td>Lasso<br>
Ridge<br>
Elastic net</td>
</tr>
<tr class="even">
<td></td>
<td>Dimensionality reduction</td>
<td>Principal component<br>
Partial least squares</td>
</tr>
<tr class="odd">
<td></td>
<td>Tree-based learning</td>
<td>Random forests<br>
Extreme gradient boosting<br>
Cubist</td>
</tr>
<tr class="even">
<td>Radiative transfer modelling</td>
<td>Leaf-level<br>
Canopy-level</td>
<td><span class="smallcaps">Prospect Pro</span><br>
<span class="smallcaps">Less</span><br>
<span class="smallcaps">Less</span> + interpolation</td>
</tr>
</tbody>
</table>
</div>
<section id="modelling-with-hand-crafted-features" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="modelling-with-hand-crafted-features"><span class="header-section-number">6.1</span> Modelling with hand-crafted features</h2>
<p>This section presents the approaches taken to model crop nitrogen by employing vegetation indices. Ten relevant vegetation indices were computed according to the equations specified in <a href="#tbl-VegetationIndices">table&nbsp;1</a>. While a myriad of vegetation indices exist to model various plant traits, the focus in this thesis was placed on indices that have been utilized for modelling crop nitrogen in scientific literature, as documented by <span class="citation" data-cites="herrmann2010swir">Herrmann et al. (<a href="#ref-herrmann2010swir" role="doc-biblioref">2010</a>)</span> and <span class="citation" data-cites="tian2011assessing">Tian et al. (<a href="#ref-tian2011assessing" role="doc-biblioref">2011</a>)</span>. Upon calculating the vegetation indices, a simple linear regression model was fitted using ordinary least squares regression. Consequently, every model within the <em>single vegetation indices</em> group in <a href="#tbl-ModellingApproaches">table&nbsp;5</a> has only two parameters: an intercept (<span class="math inline">\(\beta_0\)</span>) and a slope (<span class="math inline">\(\beta_1\)</span>).</p>
<p>Moreover, multiple linear regression models were evaluated, with all of the vegetation indices initially available. To prevent overfitting, stepwise selection was employed to choose a suitable subset from all the vegetation indices. Utilizing the Bayesian information criterion (<a href="#eq-AIC">Equation&nbsp;9</a> with <span class="math inline">\(k = \ln n\)</span>) resulted in superior model performance compared to using the Akaike information criterion. The stepwise regression was executed with the <code>step</code> function from the R package <code>stats</code> <span class="citation" data-cites="R2022">(<a href="#ref-R2022" role="doc-biblioref">R Core Team 2022</a>)</span>, where sequential replacement was the specific strategy of choice (<code>direction = "both"</code>).</p>
<div style="page-break-after: always;"></div>
<p>Stepwise selection was also employed as a strategy to identify a suitable subset of bands for a multiple linear model. However, this approach was only used after implementing other algorithms that initially chose a smaller subset of bands from the original data. This was due to the declining performance of stepwise selection as the number of variables added to the model increases <span class="citation" data-cites="hastie2017extended">(<a href="#ref-hastie2017extended" role="doc-biblioref">Hastie et al. 2017</a>)</span>. Consequently, two algorithms that generate a prior subset of bands were examined in the context of this thesis: the recursive feature selection algorithm and the square feature selection algorithm. Both aim to identify bands that exhibit minimal correlation with one another, and both rely on the covariance matrix (<span class="math inline">\(\boldsymbol \Sigma\)</span>) of the z-score normalized data for this purpose.</p>
<p>The recursive feature selection algorithm is a straightforward method that iteratively selects bands with the highest pairwise covariance from <span class="math inline">\(\boldsymbol \Sigma\)</span>, eliminating the one with a higher shared covariance among all other bands. This process is repeated until the dataset is reduced to the desired <span class="math inline">\(d\)</span> dimensions. To refine this approach, <span class="math inline">\(d\)</span> was treated as a hyperparameter and chosen using cross-validation (<a href="#tbl-hyperparameters">Table&nbsp;6</a>).</p>
<div id="fig-FeatureSelection" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/CovarianceMatrix.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;14: The square feature selection algorithm selects bands based on the covariance matrix (<span class="math inline">\(\boldsymbol \Sigma\)</span>). It attempts to fit a square within the area of <span class="math inline">\(\boldsymbol \Sigma\)</span> that is larger than a predetermined threshold. In this instance, the square feature selection algorithm was applied to the canopy-level reflectance (<strong>A</strong>), leaf-level reflectance (<strong>B</strong>), and leaf-level transmittance (<strong>C</strong>), respectively. The threshold employed for this illustration was <span class="math inline">\(s = 0.95\)</span>, corresponding to the white areas on the matrix diagonals. The numbering of the squares refers to their ordered size. All axes on these plots indicate the wavelength in nanometres.</figcaption>
</figure>
</div>
<p>On the other hand, the square feature selection algorithm seeks to identify a square subset centred on the diagonal of <span class="math inline">\(\boldsymbol \Sigma\)</span>, with the constraint that the square may only contain covariance values larger than a specific threshold <span class="math inline">\(s\)</span> (<a href="#fig-FeatureSelection">Figure&nbsp;14</a>). The centre of the largest possible square corresponds to the selected band. All bands within that square are then removed from <span class="math inline">\(\mathbf X\)</span>, and the procedure is repeated <span class="math inline">\(d\)</span> times to find <span class="math inline">\(d\)</span> uncorrelated bands. As a result, in this implementation, both <span class="math inline">\(s\)</span> and <span class="math inline">\(d\)</span> were treated as hyperparameters that required fine-tuning. The optimal combinations of parameters derived from this fine-tuning process are presented in <a href="#tbl-hyperparameters">table&nbsp;6</a>.</p>
</section>
<section id="sec-hypertuning" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="sec-hypertuning"><span class="header-section-number">6.2</span> Training the machine learning models</h2>
<p>In general, numerous machine learning models possess the flexibility to fit any given data perfectly. However, the ultimate objective is not a perfect fit but to accurately approximate the true data generating function. To prevent overfitting, these models typically employ parameters that regulate their complexity or the scale of estimated parameters. These parameters, referred to as hyperparameters, are distinct from the model’s internal parameters, as they must be explicitly defined. The optimization of hyperparameters plays a crucial role in ensuring the effective performance of machine learning models. To circumvent overfitting, data is typically divided randomly into training and validation sets. Models are calibrated using the training set, while performance metrics are assessed on the validation set. This approach can lead to noisy performance estimates, particularly in smaller datasets, depending on the arbitrary data split. Consequently, cross-validation has become a widely adopted technique, wherein the dataset <span class="math inline">\(\mathcal D\)</span> is partitioned into <span class="math inline">\(K\)</span> folds. For each fold <span class="math inline">\(k \in \{1,2,\dots,K\}\)</span>, a model <span class="math inline">\(f\)</span> is trained based on <span class="math inline">\(\mathcal D_{\lnot k}\)</span> and validated on <span class="math inline">\(\mathcal D_{k}\)</span>. To further increase the stability of the results, this process may be repeated <span class="math inline">\(r\)</span> times <span class="citation" data-cites="stone1974cross">(<a href="#ref-stone1974cross" role="doc-biblioref">Stone 1974</a>)</span>. Ultimately, the average performance across all folds and repetitions can be employed to assess the model’s effectiveness.</p>
<p>The optimized hyperparameters for the machine learning models can be found in <a href="#tbl-hyperparameters">table&nbsp;6</a>. These models were developed utilizing the statistical software and programming language R, in conjunction with a variety of packages such as <code>stats</code>, <code>caret</code>, <code>pls</code>, <code>randomForest</code>, <code>xgboost</code>, and <code>Cubist</code> <span class="citation" data-cites="R2022 kuhn2022caret liland2022pls chen2022 kuhn2023 liaw2002randomForest">(<a href="#ref-liaw2002randomForest" role="doc-biblioref">Liaw and Wiener 2002</a>, <a href="#ref-chen2022" role="doc-biblioref">Chen et al. 2022</a>, <a href="#ref-kuhn2022caret" role="doc-biblioref">Kuhn 2022</a>, <a href="#ref-liland2022pls" role="doc-biblioref">Liland et al. 2022</a>, <a href="#ref-R2022" role="doc-biblioref">R Core Team 2022</a>, <a href="#ref-kuhn2023" role="doc-biblioref">Kuhn and Quinlan 2023</a>)</span>.</p>
<div id="tbl-hyperparameters" class="anchored">
<table class="table">
<caption>Table&nbsp;6: List of hyperparameters for the feature selection and machine learning algorithms. The optimized hyperparameters are individually provided for the leaf-level data (LL) both for area- and mass-based nitrogen, and for the canopy-level data (CL).</caption>
<colgroup>
<col style="width: 37%">
<col style="width: 27%">
<col style="width: 15%">
<col style="width: 12%">
<col style="width: 9%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Method</strong></th>
<th><strong>Parameter range</strong></th>
<th><strong>LL</strong> (N<sub>area</sub>)</th>
<th><strong>LL</strong> (N<sub>%</sub>)</th>
<th><strong>CL</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Recursive feature selection</td>
<td><span class="math inline">\(d \in [2,10]\)</span><br>
</td>
<td>9</td>
<td>8</td>
<td>4</td>
</tr>
<tr class="even">
<td>Square feature selection</td>
<td><span class="math inline">\(d \in [2,10]\)</span><br>
<span class="math inline">\(s \in [0.8,1]\)</span><br>
</td>
<td>7<br>
0.86<br>
</td>
<td>7<br>
0.92<br>
</td>
<td>6<br>
0.91</td>
</tr>
<tr class="odd">
<td>Lasso</td>
<td><span class="math inline">\(\lambda \in [10^{-5},1]\)</span><br>
</td>
<td>10<sup>-5</sup></td>
<td>10<sup>-5</sup></td>
<td>0.0063</td>
</tr>
<tr class="even">
<td>Ridge</td>
<td><span class="math inline">\(\lambda \in [10^{-5},1]\)</span><br>
</td>
<td>0.0141</td>
<td>10<sup>-5</sup></td>
<td>0.0063</td>
</tr>
<tr class="odd">
<td>Elastic-net</td>
<td><span class="math inline">\(\lambda \in [10^{-5},1]\)</span><br>
<span class="math inline">\(\alpha \in [0,1]\)</span><br>
</td>
<td>10<sup>-5</sup><br>
0.94</td>
<td>0.001<br>
1</td>
<td>0.0398<br>
0.1</td>
</tr>
<tr class="even">
<td>Principal components</td>
<td><span class="math inline">\(l \in [1,50]\)</span><br>
</td>
<td>19</td>
<td>19</td>
<td>5</td>
</tr>
<tr class="odd">
<td>Partial least squares</td>
<td><span class="math inline">\(l \in [1,50]\)</span><br>
</td>
<td>13</td>
<td>16</td>
<td>4</td>
</tr>
<tr class="even">
<td>Random forest</td>
<td><code>ntree</code> <span class="math inline">\(\in [50,5000]\)</span><br>
<code>nodesize</code> <span class="math inline">\(\in [1,5]\)</span><br>
</td>
<td>500<br>
3<br>
</td>
<td>500<br>
3<br>
</td>
<td>500<br>
5<br>
</td>
</tr>
<tr class="odd">
<td>Extreme gradient boosting</td>
<td><span class="math inline">\(\eta \in [0.01,1]\)</span><br>
<code>max_depth</code> <span class="math inline">\(\in [1,10]\)</span><br>
</td>
<td>0.3<br>
2<br>
</td>
<td>0.1<br>
2<br>
</td>
<td>0.1<br>
1<br>
</td>
</tr>
<tr class="even">
<td>Cubist</td>
<td><code>rules</code> <span class="math inline">\(\in [2,5]\)</span><br>
<code>committees</code> <span class="math inline">\(\in [1,20]\)</span></td>
<td>2<br>
10</td>
<td>2<br>
20</td>
<td>2<br>
20</td>
</tr>
</tbody>
</table>
</div>
</section>
<section id="running-the-radiative-transfer-models" class="level2 page-columns page-full" data-number="6.3">
<h2 data-number="6.3" class="anchored" data-anchor-id="running-the-radiative-transfer-models"><span class="header-section-number">6.3</span> Running the radiative transfer models</h2>
<p>Establishing the radiative transfer models proved to be the most labour-intensive task among all modelling approaches. This can be attributed to the highly specialized nature of these models, coupled with the scarcity of guidance available for their application, particularly at the canopy level.</p>
<section id="leaf-level-radiative-modelling" class="level3">
<h3 class="anchored" data-anchor-id="leaf-level-radiative-modelling">Leaf level radiative modelling</h3>
<p>At the leaf level, radiative transfer modelling was conducted utilizing <span class="smallcaps">Prospect-Pro</span>, which is available as one of the functions incorporated within the R package <code>prospect</code> <span class="citation" data-cites="feret2022package">(<a href="#ref-feret2022package" role="doc-biblioref">Féret and de Boissieu 2022a</a>)</span>. Generally, the most favourable results were obtained by employing the inversion technique of optimal spectral subdomains, as proposed by <span class="citation" data-cites="spafford2021spectral">Spafford et al. (<a href="#ref-spafford2021spectral" role="doc-biblioref">2021</a>)</span>. This method enhances model inversion by concentrating on spectral subdomains for each constituent and maximizing the fit within that range, rather than over the entire spectrum.</p>
<div id="tbl-range" class="anchored">
<table class="table">
<caption>Table&nbsp;7: Parameter search range for both a conservative parameter estimation and a relaxed parameter estimation with wider lower- and upper bounds.</caption>
<colgroup>
<col style="width: 36%">
<col style="width: 34%">
<col style="width: 27%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Parameter</strong></th>
<th><strong>Conservative</strong></th>
<th><strong>Relaxed</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Leaf structure parameter</td>
<td><span class="math inline">\(N\in[1.5,2.5]\)</span></td>
<td><span class="math inline">\(N\in[1,3]\)</span></td>
</tr>
<tr class="even">
<td>Chlorophyll content<br>
Carotenoid content<br>
Anthocyanin content<br>
Tannin content<br>
Equivalent water thickness<br>
Protein content<br>
Carbon-based constituents</td>
<td><code>CHL</code> <span class="math inline">\(\in[0,60]\)</span><br>
<code>CAR</code> <span class="math inline">\(\in[0,15]\)</span><br>
<code>ANT</code> <span class="math inline">\(\in[0,7.5]\)</span><br>
<code>BROWN</code> <span class="math inline">\(= 0\)</span><br>
<code>EWT</code> <span class="math inline">\(\in[0.004,0.012]\)</span><br>
<code>PROT</code> <span class="math inline">\(\in[0.0004,0.005]\)</span><br>
<code>CBC</code> <span class="math inline">\(\in[0.001,0.002]\)</span></td>
<td><code>CHL</code> <span class="math inline">\(\in[0,80]\)</span><br>
<code>CAR</code> <span class="math inline">\(\in[0,20]\)</span><br>
<code>ANT</code> <span class="math inline">\(\in[0,10]\)</span><br>
<code>BROWN</code> <span class="math inline">\(= 0\)</span><br>
<code>EWT</code> <span class="math inline">\(\in[0,0.03]\)</span><br>
<code>PROT</code> <span class="math inline">\(\in[0,0.02]\)</span><br>
<code>CBC</code> <span class="math inline">\(\in[0,0.02]\)</span></td>
</tr>
<tr class="odd">
<td>Maximum incident angle</td>
<td><span class="math inline">\(\alpha = 40\)</span></td>
<td><span class="math inline">\(\alpha = 40\)</span></td>
</tr>
</tbody>
</table>
</div>
<p>During the initial modelling phase, relatively restrictive box constraints were imposed on the parameters, meaning that no parameter was permitted to exceed the range specified in <a href="#tbl-range">table&nbsp;7</a>. Possible correlations among parameters were disregarded for constraining the search space. The conservative parameter range was derived from <span class="citation" data-cites="camino2021detection">Camino et al. (<a href="#ref-camino2021detection" role="doc-biblioref">2021</a>)</span>, who retrieved the same parameters from almonds for detecting <em>Xylella fastidiosa</em>.</p>
<p>Nonetheless, as many parameters retrieved by <span class="smallcaps">Prospect-Pro</span> in this phase resided at the lower or upper bounds, the parameter restrictions were relaxed in a subsequent phase. Parameters obtained under relaxed constraints generally resulted in a better alignment between simulated and observed data, as well as improved leaf-level predictions of foliar nitrogen concentration. Interestingly, reduced discrepancies between simulated and observed data did not consistently lead to enhanced nitrogen predictions. For instance, increasing the maximum incident angle <span class="math inline">\(\alpha\)</span> in <span class="smallcaps">Prospect-Pro</span> to <span class="math inline">\(\alpha = 80^\circ\)</span> decreased the difference between simulated and observed data, but worsened predictions of the foliar nitrogen concentration based on the retrieved parameters. Consequently, the maximum incident angle was set to <span class="math inline">\(\alpha = 40^\circ\)</span> as proposed by <span class="citation" data-cites="feret2008prospect">Féret et al. (<a href="#ref-feret2008prospect" role="doc-biblioref">2008</a>)</span>.</p>
<p>After successfully retrieving the plant parameters, they needed to be associated with the measured nitrogen labels. This was accomplished by employing stepwise selection to a multiple linear model, which allowed for the inclusion of all plant parameters as input variables. The fundamental assumption here is that the foliar nitrogen concentration can be modelled as a linear response to the concentrations of chlorophyll and protein.</p>
</section>
<section id="canopy-level-radiative-modelling" class="level3 page-columns page-full">
<h3 class="anchored" data-anchor-id="canopy-level-radiative-modelling">Canopy level radiative modelling</h3>
<p>To model radiative transfer at the canopy level, the ray tracing model <span class="smallcaps">Less</span> was adapted. This three-dimensional forest scene simulator offers a freely accessible graphical user interface (GUI) and can be downloaded via the model’s website (<a href="https://lessrt.org" class="uri">https://lessrt.org</a>). Unfortunately, <span class="smallcaps">Less</span> does not provide an application programming interface (API), limiting customization possibilities for the simulations.</p>
<p>A virtual scene of an almond tree, where reflectance may interact with neighbouring trees and the soil, was created using a single three-dimensional model of an almond tree. This model was obtained from the <a href="https://www.cgtrader.com/">cgtrader</a> platform, where it was sold by the anonymous artist <span class="smallcaps">XFrog</span>. <span class="smallcaps">Less</span> enables users to upload any <code>.obj</code> file and arrange the objects as desired. Thus, a typical orchard scene could be recreated in <span class="smallcaps">Less</span>, with a central focus tree surrounded by 24 other trees in a 5 × 5 grid design. The trees were aligned similarly to the average trees in the Westwind orchard, with an inter-row distance of 6.73 m and a distance of 4.56 m between trees along the rows. The generated data cubes had dimensions of 56 × 38 × 200 = 425’600 values, closely corresponding to the average dimensions of the cropped hyperspectral images. The simulated image was specified to be orthographic, akin to the georectified real image (<a href="#fig-lessmodelling">Figure&nbsp;15</a>).</p>
<div id="fig-lessmodelling" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/LESS.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;15: Illustration of the three-dimensional object of an almond tree used in the simulation (<strong>A</strong>) and the model’s alignment to replicate the orchard scene (<strong>B</strong>). To generate predictions with <span class="smallcaps">Less</span>, observed hyperspectral images (<strong>C</strong>) must be compared with simulated hyperspectral images (<strong>D</strong>). It is important to note that the two images shown here are RGB reconstructions derived from hyperspectral data.</figcaption>
</figure>
</div>
<p><span class="smallcaps">Less</span> requires defining camera settings that closely approximate the actual settings on the day of data capture. In order to recreate the conditions of the data capture, the camera distance to the ground was set at 100 m above ground level. Simulated spectral bands were configured to align with those from the hyperspectral data. The view zenith angle and view azimuth angle were set to 0° and 180°, respectively. The solar zenith angle was set to 20°, while the solar azimuth angle was set to 240°. Sky cloudiness was not considered.</p>
<p>The hyperspectral signatures of leaves in the scene were based on the simulated spectral reflectance and transmittance of <span class="smallcaps">Prospect-D</span>. Given that the hyperspectral data did not encompass the short-wave-infrared spectrum, there was no need for <span class="smallcaps">Prospect-Pro</span> and its ability to simulate absorption caused by protein. The soil in the simulated scene was assigned the same reflectance as the bare soil in the hyperspectral images of the orchard.</p>
<p><span class="smallcaps">Less</span> generates output files of simulated radiance in the <code>.bsq</code> format, which closely resembles the data captured by the hyperspectral camera. Consequently, the output of <span class="smallcaps">Less</span> had to undergo similar processing steps as real observations to obtain comparable results: calculating reflectance from radiance and irradiance, semantically segmenting the hyperspectral image, and ultimately computing the mean reflectance of the bright canopy. As a result, a single simulated hyperspectral image from <span class="smallcaps">Less</span> could be reduced to a vector of reflectance data — represented as one row in the generated look-up table.</p>
<p>The look-up table was constructed by running <span class="smallcaps">Less</span> for numerous parameter combinations (<a href="#tbl-LessParameters">Table&nbsp;8</a>), with the parameter ranges informed by the retrieved leaf-level ranges. The leaf area index (LAI) was the sole parameter not retrieved by <span class="smallcaps">Prospect-D</span>. This parameter was controlled exclusively by the size of the trees in the scene, resulting in a reduced leaf area over the same ground. All generated hyperspectral images were subsequently condensed to their reflectance vectors and stored as rows in a large matrix. This matrix effectively functions as the look-up table, as the exact parameters generating each reflectance entry are known and stored in a separate parameter matrix.</p>
<div id="tbl-LessParameters" class="anchored">
<table class="table">
<caption>Table&nbsp;8: The selected parameters for the look-up table generated using <span class="smallcaps">Less</span>. Each possible combination of these parameters was simulated once.</caption>
<colgroup>
<col style="width: 35%">
<col style="width: 65%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Parameter</strong></th>
<th><strong>Parameter range</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Chlorophyll</td>
<td><code>CHL</code> <span class="math inline">\(\in \{20, 22.5, 25, 27.5, 30, 32.5, 35, 37.5, 40, 42.5, 45\}\)</span></td>
</tr>
<tr class="even">
<td>Anthocyanin</td>
<td><code>ANT</code> <span class="math inline">\(\in \{2.75, 3.00, 3.25, 3.5, 3.75\}\)</span></td>
</tr>
<tr class="odd">
<td>Carotenoid</td>
<td><code>CAR</code> <span class="math inline">\(\in \{2.0, 3.0, 4.0, 5.0, 6.0\}\)</span></td>
</tr>
<tr class="even">
<td>Carbon-based constituents</td>
<td><code>CBC</code> <span class="math inline">\(\in \{0.006, 0.007, 0.008, 0.009, 0.010\}\)</span></td>
</tr>
<tr class="odd">
<td>Leaf area index</td>
<td><code>LAI</code> <span class="math inline">\(\in \{1,2,3,4,5,6,7,8,9,10,11,12,13,14,15 \}\)</span></td>
</tr>
</tbody>
</table>
</div>
<p>The look-up table <span class="math inline">\(\mathbf L\)</span> can be used to make a prediction for a single observation <span class="math inline">\(\mathbf x\)</span> by identifying the closest match between the observed and simulated reflectance based on a specific cost function <span class="math inline">\(J\)</span>. The most direct approach to this involves minimizing the Euclidean norm of the error between the <span class="math inline">\(i\)</span><sup>th</sup> simulation in the look-up table <span class="math inline">\(\mathbf L_{(i)}\)</span> and the observation <span class="math inline">\(\mathbf x\)</span>. <span id="eq-LUTminimization"><span class="math display">\[
J(i) = \| \mathbf L_{(i)} - \mathbf x \|_2^2
\qquad
i^* = \arg \min_{i \in \{1 .. d\}} J(i)
\tag{26}\]</span></span> In this context, <span class="math inline">\(i\)</span> represents the index of the simulated reflectance, and <span class="math inline">\(d\)</span> denotes the total number of simulations in the look-up table. To ensure a consistent treatment of the error, it is crucial that both the look-up table and the observed reflectance undergo z-score normalization prior to determining <span class="math inline">\(i^*\)</span>.</p>
<p>A straightforward method for performing regression using the look-up table <span class="math inline">\(\mathbf L\)</span> involves calculating the cost <span class="math inline">\(J\)</span> with respect to a single observation for each row in <span class="math inline">\(\mathbf L\)</span>, and then selecting <span class="math inline">\(i^*\)</span> associated with the lowest cost <span class="math inline">\(J(i)\)</span> as written in <a href="#eq-LUTminimization">equation&nbsp;26</a>. Given that the parameter matrix is <span class="math inline">\(\mathbf P\)</span>, the retrieved parameters are <span class="math inline">\(\mathbf P_{(i^*)}\)</span>, or the parameters in the <span class="math inline">\(i^*\)</span><sup>th</sup> row of <span class="math inline">\(\mathbf P\)</span>. This implies that the retrieved parameters may only be any combination of those listed in <a href="#tbl-LessParameters">table&nbsp;8</a>, resulting in discrete predictions.</p>
<div class="page-columns page-full"><p>To enable the retrieval of continuous parameters, a hyper-dimensional interpolation must be applied to the look-up table. This involves modelling the cost <span class="math inline">\(J\)</span> with respect to a single observation as a function of the parameters, an approach known as surrogate modelling <span class="citation" data-cites="gramacy2020surrogates">(<a href="#ref-gramacy2020surrogates" role="doc-biblioref">Gramacy 2020</a>)</span>. Gaussian processes would be the most mathematically elegant surrogate model; however, due to their computational inefficiency, a multicubic polynomial surrogate model was chosen instead.<a href="#fn6" class="footnote-ref" id="fnref6" role="doc-noteref"><sup>6</sup></a> To facilitate surrogate modelling, a subset of the parameter matrix <span class="math inline">\(\mathbf P\)</span> was created after computing <span class="math inline">\(J\)</span>, which includes only the closest neighbours of the found parameters <span class="math inline">\(\mathbf P_{(i^*)}\)</span>. The multicubic polynomial was fitted to this subset to model <span class="math inline">\(J\)</span> as a response to the parameters, allowing for easy minimization and therefore the retrieval of smooth parameters.</p><div class="no-row-height column-margin column-container"><li id="fn6"><p><sup>6</sup>&nbsp;Bear in mind that the surrogate model must be re-trained once for each observation, making computational efficiency an important aspect of this step.</p></li></div></div>
</section>
</section>
</section>
<section id="sec-results" class="level1" data-number="7">
<h1 data-number="7"><span class="header-section-number">7</span> Results of the model performances</h1>
<p>In this chapter, the predictive capabilities of all constructed models for estimating crop nitrogen in Californian almonds will be evaluated, both at the leaf and canopy levels. A clear performance metric is necessary for comparing different predictive models, summarizing the predictive power of a given model with respect to specific features and labels in a single number. A commonly used performance metric for regression models is the root mean squared error (RMSE, <a href="#eq-RegressionPerformanceMetrics">equation&nbsp;27</a>), which can be interpreted as the standard deviation of the error in the model’s predictions. However, RMSE is often challenging to interpret. Consequently, the primary performance metric reported here will be the coefficient of determination (R<sup>2</sup>, <a href="#eq-RegressionPerformanceMetrics">equation&nbsp;27</a>) as introduced by <span class="citation" data-cites="wright1921correlation">Wright (<a href="#ref-wright1921correlation" role="doc-biblioref">1921</a>)</span>. This metric ranges from <span class="math inline">\(-\infty\)</span> to 1, where a score of 1 signifies perfect predictions, and 0 is equivalent to the baseline model <span class="math inline">\(f(\mathbf x) = \bar y\)</span>, i.e., using the mean of the labels as the best guess for any given observation. A negative value of R<sup>2</sup> indicates that the model performance is worse than this baseline model. <span id="eq-RegressionPerformanceMetrics"><span class="math display">\[
\text{RMSE} = \sqrt{\frac{1}{n}\sum_{i=1}^n \left( y_i - \hat y_i \right)^2}
\qquad
\text{R}^2 =
1 - \frac{\sum_{i=1}^n \left( y_i - \hat y_i \right)^2}{\sum_{i=1}^n \left( y_i - \bar y \right)^2}
\tag{27}\]</span></span> Here, <span class="math inline">\(n\)</span> is the number of observations, <span class="math inline">\(y_i\)</span> is the <span class="math inline">\(i\)</span><sup>th</sup> label, <span class="math inline">\(\hat y_i\)</span> is the <span class="math inline">\(i\)</span><sup>th</sup> prediction, and <span class="math inline">\(\bar y\)</span> is the mean of the label vector <span class="math inline">\(\mathbf y\)</span>. The R<sup>2</sup> offers high interpretability without requiring further knowledge of the specific set of labels <span class="math inline">\(\mathbf y\)</span> <span class="citation" data-cites="chicco2021">(<a href="#ref-chicco2021" role="doc-biblioref">Chicco et al. 2021</a>)</span>, since it can also be seen as the proportion of variance in the label <span class="math inline">\(\mathbf y\)</span> that is explained by the model. While R<sup>2</sup> is included in many statistical models, its use for model selection has been criticized due to its tendency to overestimate a model’s fit <span class="citation" data-cites="willett1988another">(<a href="#ref-willett1988another" role="doc-biblioref">Willett and Singer 1988</a>)</span>. However, this issue is less about R<sup>2</sup> itself and more about over-fitting — a core concept in machine learning. One disadvantage of R<sup>2</sup> compared to the (root) mean squared error is its dependence on the range of the labels: the wider the range of <span class="math inline">\(\mathbf y\)</span>, the larger the R<sup>2</sup> value, without necessarily corresponding to a lower error. However, this only becomes an issue when comparing models tested on different datasets, which is not a concern in this case.</p>
<p>The results which are presented in the following two sections were all obtained from a 50 times repeated 20-fold cross-validation. This means, every model was re-run 1000 times on a different random subset of the data and evaluated on the rest. This large number of evaluations ensured that the results obtained are stable, as it leads to smaller standard errors and thus a more reliable inter-comparison of the models.</p>
<section id="performance-on-the-leaf-level" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="performance-on-the-leaf-level"><span class="header-section-number">7.1</span> Performance on the leaf level</h2>
<p>At the leaf level, two related prediction problems were examined: area-based nitrogen (N<sub>area</sub>) and mass-based nitrogen (N<sub>%</sub>). As demonstrated in <a href="#tbl-prospect">table&nbsp;2</a>, the leaf-level radiative transfer model <span class="smallcaps">Prospect-Pro</span> retrieves plant pigments in per-area units, i.e., independently of leaf thickness.</p>
<section id="sec-ResultsAreaBased" class="level3">
<h3 class="anchored" data-anchor-id="sec-ResultsAreaBased">Area-based nitrogen</h3>
<p>Indeed, <span class="smallcaps">Prospect-Pro</span> achieves the best results in predicting area-based nitrogen content with a significant margin (R<sup>2</sup> = 0.684). The next best model, the partial least squares regression model, trails <span class="smallcaps">Prospect-Pro</span> by 5.8 percentage points (R<sup>2</sup> = 0.626). Most of the implemented machine learning algorithms exceed an R<sup>2</sup> of 0.5, with the exception of cubist trees (<a href="#fig-modelPerformanceSRMNarea">Figure&nbsp;16</a>).</p>
<p>Interestingly, the square feature selection algorithm, which is relatively simple to implement, almost keeps pace with the best-performing machine learning methods (R<sup>2</sup> = 0.610). Single vegetation indices, on the other hand, do not correlate as well with area-based nitrogen. A single exception is the R434 which is the only vegetation index that utilizes information from blue bands (R<sup>2</sup> = 0.255). When combined, vegetation indices perform better, but this approach still falls short compared to most of the machine learning algorithms.</p>
</section>
<section id="sec-ResultsMassBased" class="level3">
<h3 class="anchored" data-anchor-id="sec-ResultsMassBased">Mass-based nitrogen</h3>
<p>Shifting from area-based nitrogen to mass-based nitrogen, the predictive capability of previously leading models generally declines (<a href="#fig-modelPerformanceSRM">Figure&nbsp;17</a>). <span class="smallcaps">Prospect-Pro</span> still achieves relatively good results (R<sup>2</sup> = 0.453), but is outperformed by one of the machine learning models: the partial least squares regression model (R<sup>2</sup> = 0.493). Regrettably, estimating leaf mass per area from <span class="smallcaps">Prospect-Pro</span> and using this estimation to convert units yielded worse results than directly applying a multiple linear regression model to the estimated (area-based) retrieved parameters. Meanwhile, directly using leaf mass per area to convert the retrieved parameters would have given the <span class="smallcaps">Prospect-Pro</span> an unjust advantage over the machine learning models. However, the promising results of partial least squares regression suggest some potential for improvement in <span class="smallcaps">Prospect-Pro</span> (or any related physics-based model) to predict the pigment concentrations.</p>
<p>Interestingly, the performance of ridge regression for mass-based nitrogen concentration drops much more than that of lasso regression, although ridge performed slightly worse than lasso on area-based nitrogen as well. Unlike ridge regression, which only shrinks linear model coefficients but never sets them to zero, lasso regression does precisely that: it sets the least important coefficients to zero, effectively leading to variable selection <span class="citation" data-cites="james2013introduction">(<a href="#ref-james2013introduction" role="doc-biblioref">James et al. 2013</a>)</span>. Hence, if lasso regression outperforms ridge regression, this hints to the existence of a small number of relevant bands, while others can be disregarded.</p>
<p>The R434 vegetation index, which performed best for area-based nitrogen, fares the worst for mass-based nitrogen. Conversely, all vegetation indices that performed worse than the baseline model <span class="math inline">\(f(\mathbf x) = \bar y\)</span> for area-based nitrogen significantly improved in the case of mass-based nitrogen. It is worth noting that vegetation indices, compared to other methods, exclusively utilize reflectance data. A possible explanation for their shortcomings in predicting area-based nitrogen is the correlation between transmittance and leaf mass per area, and subsequently any area-based trait.</p>
<p>Lastly, examining the standard deviation of cross-validated results reveals that the machine learning models in particular have very unstable outcomes (for both area-based and mass-based nitrogen), while <span class="smallcaps">Prospect-Pro</span> prediction results appear to be comparatively stable. This can be attributed to the effective degrees of freedom of the models, which is much lower for physics-based models than for any machine-learning model, since there are only a few parameters to be estimated from the data.</p>
<div id="fig-modelPerformanceSRMNarea" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/Results_SRM_N_area.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;16: Performance of the different modelling methods in predicting the area-based nitrogen (N<sub>area</sub>) on a leaf-level. The value of the performance metric R<sup>2</sup> was calculated based on a 50 times repeated 20-fold cross-validation, of which the mean (<span class="math inline">\(\mu\)</span>) and the standard deviation (<span class="math inline">\(\sigma\)</span>) are reported. The compact letter display (CLD) next to every bar is derived from a pairwise Wilcoxon rank sum tests. Exact numerical values of the results can be found in the appendix in <a href="#tbl-ResultsNarea">table&nbsp;10</a>.</figcaption>
</figure>
</div>
<div id="fig-modelPerformanceSRM" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/Results_SRM.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;17: Performance of the different modelling methods in predicting the mass-based nitrogen (N<sub>%</sub>) on a leaf-level. The value of the performance metric R<sup>2</sup> was calculated based on a 50 times repeated 20-fold cross-validation, of which the mean (<span class="math inline">\(\mu\)</span>) and the standard deviation (<span class="math inline">\(\sigma\)</span>) are reported. The compact letter display (CLD) next to every bar is derived from a pairwise Wilcoxon rank sum tests. Exact numerical values of the results can be found in the appendix in <a href="#tbl-ResultsNmass">table&nbsp;11</a>.</figcaption>
</figure>
</div>
</section>
</section>
<section id="performance-on-the-canopy-level" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="performance-on-the-canopy-level"><span class="header-section-number">7.2</span> Performance on the canopy level</h2>
<p>Focusing on the model performance results at the canopy level, the overall performance significantly declines compared to the leaf level (<a href="#fig-modelPerformanceHSI">Figure&nbsp;18</a>). At best, the models appear to explain only 26% of the variance in the measured foliar nitrogen concentration. In this context, dimensionality reduction techniques slightly underperform compared to both lasso regression and elastic-net (R<sup>2</sup> = 0.249 and R<sup>2</sup> = 0.250, respectively). The gap between the elastic-net and ridge regression (R<sup>2</sup> = 0.206) is especially intriguing, considering the mixing parameter (<span class="math inline">\(\alpha = 0.1\)</span>): the elastic-net coefficient penalization is a blend of 90% <span class="math inline">\(\ell^2\)</span>-norm and only 10% <span class="math inline">\(\ell^1\)</span>-norm (<a href="#tbl-hyperparameters">Table&nbsp;6</a>). Therefore, it leans heavily towards the penalization applied in ridge regression. However, the ability to completely eliminate coefficients appears to be highly beneficial. This outcome serves as an excellent illustration of the sensitivity of many machine learning models to minor alterations in their hyperparameters.</p>
<p>One of the most striking outcomes is the substantial difference the hyper-dimensional interpolation has on the <span class="smallcaps">Less</span> look-up table predictions. When the look-up table is used directly for predictions, the results are underwhelming (R<sup>2</sup> = 0.116). However, applying a surrogate model to achieve hyper-dimensional interpolation more than doubles the model performance (R<sup>2</sup> = 0.235). Consequently, by implementing the interpolation, the results produced by <span class="smallcaps">Less</span> become comparable to machine-learning methods, despite the relatively low resolution of certain parameters in the look-up table.</p>
<p>Ironically, the best-performing model at the canopy level is not the widely praised partial least squares regression model, but rather the combination of square feature selection and the widely criticized stepwise selection (R<sup>2</sup> = 0.256).</p>
<p>In contrast to most other tested methods, vegetation indices perform rather poorly when applied to canopy-level data. Out of ten assessed vegetation indices, only two (TCARI and R434) manage outperform the baseline model <span class="math inline">\(f(\mathbf x) = \bar y\)</span>. Notably, the R434 vegetation index, introduced by <span class="citation" data-cites="tian2011assessing">Tian et al. (<a href="#ref-tian2011assessing" role="doc-biblioref">2011</a>)</span>, is almost on par with the other more complex methods (R<sup>2</sup> = 0.198). It is the only vegetation index that utilizes information from blue bands (as to be seen in <a href="#tbl-VegetationIndices">table&nbsp;1</a>).</p>
<div id="fig-modelPerformanceHSI" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="resources_html/Results_HSI.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;18: Performance of the different modelling methods in predicting the mass-based nitrogen (N<sub>%</sub>) on a canopy-level. The value of the performance metric R<sup>2</sup> was calculated based on a 50 times repeated 20-fold cross-validation, of which the mean (<span class="math inline">\(\mu\)</span>) and the standard deviation (<span class="math inline">\(\sigma\)</span>) are reported. The compact letter display (CLD) next to every bar is derived from a pairwise Wilcoxon rank sum tests. Exact numerical values of the results can be found in the appendix in <a href="#tbl-ResultsHSI">table&nbsp;12</a>.</figcaption>
</figure>
</div>
</section>
</section>
<section id="sec-discussion" class="level1" data-number="8">
<h1 data-number="8"><span class="header-section-number">8</span> Discussion</h1>
<p>In this chapter, I will examine the results of this thesis in a broader context, addressing both the advantages and disadvantages of the various modelling approaches. Additionally, potential improvements will be explored, with a particular emphasis on radiative transfer models, but also aspects from which other methods could profit. Finally, certain peculiarities of the dataset, specifically nitrogen measurements, will be discussed and compared to similar research in the scientific literature.</p>
<section id="different-modelling-approaches" class="level2" data-number="8.1">
<h2 data-number="8.1" class="anchored" data-anchor-id="different-modelling-approaches"><span class="header-section-number">8.1</span> Different modelling approaches</h2>
<p>In this thesis, I compared various approaches to modelling foliar nitrogen concentration in Californian almonds, including vegetation indices and other hand-crafted features, diverse machine learning algorithms, and physics-based radiative transfer models. This model intercomparison highlighted the remarkable capability of machine learning models to identify patterns in the data and generate impressive predictions compared to other methods. Linear models, in particular, performed exceptionally well when combined with linear dimensionality reduction, regularization, or prior feature selection. The underwhelming performance of decision-tree-based models, capable of capturing non-linear relationships, likely stems from the absence of strong non-linear relationships in the data, combined with a low signal-to-noise ratio — a situation where decision-tree-based models easily overfit the data.</p>
<p>However, these data characteristics are highly dependent on the specific dataset, such as the collection time, nitrogen measurement range, and almond cultivar. As a result, the performance of machine learning algorithms can be highly inconsistent. One dataset may favour lasso regularization, while another may be better suited for partial least squares or extreme gradient boosting. This issue underscores the primary challenge of machine learning models in this context: the lack of generalizability. A model trained in one setting often loses predictive power in another <span class="citation" data-cites="verrelst2019quantifying">(<a href="#ref-verrelst2019quantifying" role="doc-biblioref">Verrelst et al. 2019</a>)</span>. Ideally, the training dataset should encompass all necessary variance, and the test set should reflect real-world deviations. However, acquiring such a dataset for agricultural practice would be extremely difficult.</p>
<p>In contrast, physics-based models promise to address this generalizability problem <span class="citation" data-cites="verrelst2019quantifying berger2020crop">(<a href="#ref-verrelst2019quantifying" role="doc-biblioref">Verrelst et al. 2019</a>, <a href="#ref-berger2020crop" role="doc-biblioref">Berger et al. 2020</a>)</span>. Identifying the correct functional form of a relationship enables models to reflect non-linear relationships without the need of giving the model more general flexibility. In this thesis, retrieved plant parameters, such as chlorophyll or carotenoid content, were linked to measured nitrogen concentrations. This connection only necessitates a handful of coefficients, contributing to the generalizability of physics-based models.</p>
<p>The radiative transfer model <span class="smallcaps">Prospect-Pro</span> can be viewed as a dimensionality reduction technique not completely unlike principal component analysis. However, compared to principal components, the latent variables identified by <span class="smallcaps">Prospect-Pro</span> have a physical meaning, mostly relating to the content of specific plant pigments responsible for observed reflectance (and transmittance). As a dimensionality reduction technique, <span class="smallcaps">Prospect-Pro</span> (and <span class="smallcaps">Less</span> by extension) simplifies modelling any plant trait of interest, not just crop nitrogen, representing the highest level of generalizability where the orchard, crop, and response of interest can all vary.</p>
<p>However, discovering the true physical relationship between variables is extremely labour-intensive and stands in a stark contrast to machine learning models, where data gathering is the most time-consuming aspect. However, once a functional form capturing the true relationship is found, physics-based models can enable even better predictions than machine learning models, as presented in <a href="#sec-ResultsAreaBased">section&nbsp;7.1.1</a>. Moreover, the physics-based modelling process enhances the understanding of nature, a benefit not offered by machine learning models. In particular, complex machine learning models, such as neural networks, are often referred to as black box models <span class="citation" data-cites="zhu2017deep rudin2019stop">(<a href="#ref-zhu2017deep" role="doc-biblioref">Zhu et al. 2017</a>, <a href="#ref-rudin2019stop" role="doc-biblioref">Rudin 2019</a>)</span>. While they may deliver impressive results, understanding the relationships they have learned can be extremely challenging.</p>
<p>The fact that machine learning models do not require prior knowledge and rely solely on data for making predictions makes them highly useful when combined with radiative transfer models. Machine learning models excel at detecting patterns and correlations in high-dimensional spaces, serving as valuable indicators for potential flaws or shortcomings in physics-based models. For instance, the superior performance of some machine learning models in predicting mass-based nitrogen on a leaf-level suggests a knowledge gap. Without machine learning models, determining what model accuracy can realistically be expected would be much more challenging.</p>
<p>The limitations of machine learning models can also be applied to vegetation indices. However, this vegetation indices generally performed poorly, especially when only using single vegetation indices. A primary advantage of vegetation indices is their computational efficiency and ease of implementation. However, this stems from a time when computational efficiency was a greater concern than it is today, meaning it is no longer a significant limitation. The results of this thesis suggest that single vegetation indices can hardly compete with either machine learning models or physics-based models. Nevertheless, new vegetation indices continue to be constructed using heuristics such as genetic algorithms <span class="citation" data-cites="costa2020new">(<a href="#ref-costa2020new" role="doc-biblioref">Costa et al. 2020</a>)</span>. In this approach, the authors essentially frame the problem in the same manner as a supervised machine learning problem, but they solve it in a computationally inefficient way. Similar to machine learning models, this solution may rely on fortuitous correlations and generally represents an inelegant approach given the widespread availability of machine learning algorithms.</p>
<p>It is crucial to repeatedly acknowledge that the distinction between manual feature selection, machine learning, and physics-based models is somewhat arbitrary and exhibits considerable overlap. For example, both vegetation indices and the parameters retrieved by radiative transfer models are still connected to nitrogen measurements via linear regression, which can be considered a machine learning model. This link function could also be any other model, as <span class="citation" data-cites="verrelst2019quantifying">Verrelst et al. (<a href="#ref-verrelst2019quantifying" role="doc-biblioref">2019</a>)</span> speaks of model hybridization — machine learning and radiative transfer models are fused. Consequently, machine learning is undoubtedly an invaluable tool for physics-based modelling.</p>
</section>
<section id="how-radiative-transfer-models-might-be-improved" class="level2" data-number="8.2">
<h2 data-number="8.2" class="anchored" data-anchor-id="how-radiative-transfer-models-might-be-improved"><span class="header-section-number">8.2</span> How radiative transfer models might be improved</h2>
<p>The machine learning models demonstrated slightly better performance than the radiative transfer models in predicting foliar nitrogen concentration, both at the canopy and leaf levels. This indicates a knowledge gap in the radiative transfer models.</p>
<p>To enhance the radiative transfer models employed, several adjustments could be contemplated. For the <span class="smallcaps">Prospect</span> model family, the chlorophyll parameter could be divided into two separate estimates for chlorophyll a and b concentrations. This distinction is crucial not only for more accurate chlorophyll content estimation but also for a better explanation of the observed spectral signature. <span class="citation" data-cites="li2018factors">Li et al. (<a href="#ref-li2018factors" role="doc-biblioref">2018</a>)</span> demonstrated significant differences in chlorophyll a to b ratios among deciduous trees, shrubs, herbs, and grasses. The chlorophyll a to b ratio is also time-dependent, suggesting that dynamic models providing prior information could be incorporated. Dynamic crop models could be implemented to account for the change of the chlorophyll a to b ratio over time. Additionally, such models could also account for the accumulation of carbon-based content over time, which dilutes leaf nitrogen and thus reduces its measured concentration, even though the absolute nitrogen content may remain constant.</p>
<p>In addition, prior information about leaf parameter distributions (instead of only box constraints) could be utilized for the inversion of the radiative transfer model, for instance, by adopting a Bayesian optimization method instead of a maximum likelihood estimation. While these approaches are considerably more computationally intensive, they offer the added benefit of uncertainty estimates for all parameters.</p>
<p>The implementation of a ray tracing model like <span class="smallcaps">Less</span>, as presented in this thesis, primarily serves as a proof-of-concept and offers numerous opportunities for improvement. For instance, trees could be generated rather than hand-crafted by utilizing models such as <span class="smallcaps">Helios</span> <span class="citation" data-cites="bailey2019helios">(<a href="#ref-bailey2019helios" role="doc-biblioref">Bailey 2019</a>)</span>. This approach would enable the more precise integration of plant parameters, such as leaf area index, tree size and geometry, and leaf inclination distribution function, all of which influence the spectral signature of trees <span class="citation" data-cites="guyot1989factors bailey2017rapid">(<a href="#ref-guyot1989factors" role="doc-biblioref">Guyot et al. 1989</a>, <a href="#ref-bailey2017rapid" role="doc-biblioref">Bailey and Mahaffee 2017</a>)</span>. Some of these parameters can be quickly measured using terrestrial LiDAR scanning to support modelling, as proposed by <span class="citation" data-cites="bailey2017rapid">Bailey and Mahaffee (<a href="#ref-bailey2017rapid" role="doc-biblioref">2017</a>)</span>. <span class="smallcaps">Less</span> simulates leaves as Lambertian planes, which reflect light equally in all directions — which constitutes a certain departure from reality. In fact, reflectance depends on the solar incidence angle and viewing angle, together referred to as the bidirectional reflectance distribution function. Therefore, a more realistic ray tracing model simulating canopy reflectance should incorporate this process as well. However, measuring the bidirectional reflectance distribution function presents challenges due to the need to evaluate every possible combination of viewing and incidence angles for each material modelled.</p>
<p>Finally, such comprehensive models would necessitate a large-scale sensitivity analysis to better understand the impact of parameters on the reflectance.</p>
</section>
<section id="general-possibilities-for-improvement" class="level2" data-number="8.3">
<h2 data-number="8.3" class="anchored" data-anchor-id="general-possibilities-for-improvement"><span class="header-section-number">8.3</span> General possibilities for improvement</h2>
<p>There are several avenues for improvement which could benefit both machine learning and radiative transfer modelling: the spectral range covered by the sensors could be expanded, spatial and temporal dimensions in the data could considered, and the pre-processing of the data could be improved.</p>
<p>Extending the spectral range of the hyperspectral camera may prove advantageous. The performance results of various models at the canopy level appear to be limited to a maximum of 25%. The canopy level data only spans from 400 to 900 nm in its spectral dimension. Consequently, models trained on data from the hyperspectral camera are confined to the direct relationship between nitrogen and chlorophyll, and any additional correlation with plant pigments absorbing in the visible light spectrum. Incorporating information from the short-wave-infrared region — which covers protein absorption bands — may significantly improve predictions <span class="citation" data-cites="camino2018improved">(<a href="#ref-camino2018improved" role="doc-biblioref">Camino et al. 2018</a>)</span>. While the correlation matrix in <a href="#fig-exploratory">figure&nbsp;12</a> indicates a weaker correlation between short-wave-infrared region reflectance and nitrogen compared to the visible light spectrum, it also reveals that the short-wave-infrared regions are almost entirely uncorrelated with visible light regions. Therefore, crop nitrogen can likely be better modelled when utilizing data from both relevant regions.</p>
<p>It is worth noting that although the spectral range covered seems important — particularly with respect to short-wave-infrared — a higher spectral resolution is likely unnecessary. The efficacy of many models requiring information from only a few bands suggests that a small fraction of the number of bands would be sufficient to adequately model crop nitrogen. For example, the best-performing model at the canopy level using the prior square feature selection algorithm only requires six bands for optimal performance (<a href="#tbl-hyperparameters">Table&nbsp;6</a>). Thus, adding only a few bands in the short-wave-infrared region may significantly enhance model performance.</p>
<p>Incorporating spatial information could further improve predictions. One possible approach involves learning the relationship between the shape and size of almond trees and their nitrogen foliar concentration, and using this knowledge to generate a synthetic labelled dataset. This dataset could enable the training of a convolutional neural network to learn the inverse function from spatial data. For example, similar research conducted by <span class="citation" data-cites="jin2021combining">Jin et al. (<a href="#ref-jin2021combining" role="doc-biblioref">2021</a>)</span> demonstrated the successful training of a convolutional neural network to estimate forest canopy cover from satellite images. The convolutional neural network was initially trained on data generated using the three-dimensional radiative transfer model <span class="smallcaps">Rapid</span> and subsequently transferred to real data. This approach was found to enhance accuracy in comparison to a similar model solely trained on real data.</p>
<p>In addition to enhancing spectral and spatial information, temporal information could be included. Satellite imagery might be particularly useful for this purpose, as it captures orchards at regular intervals and provides standardized data formats — especially in the context of future imaging spectroscopy satellite missions <span class="citation" data-cites="berger2020crop">(<a href="#ref-berger2020crop" role="doc-biblioref">Berger et al. 2020</a>)</span>.</p>
<p>Some final improvements could result from refining the data pre-processing steps applied to the hyperspectral images as described in <a href="#sec-HSIprocessing">section&nbsp;5.3</a>. For instance, using seam-carving for tree-cropping in a non-grid manner could result in more precise tree cropping and subsequently improve the prediction accuracy.</p>
</section>
<section id="addressing-the-quality-and-properties-of-the-data" class="level2" data-number="8.4">
<h2 data-number="8.4" class="anchored" data-anchor-id="addressing-the-quality-and-properties-of-the-data"><span class="header-section-number">8.4</span> Addressing the quality and properties of the data</h2>
<p>Thus far, the attention has been solely directed towards potential enhancements in modelling or expanding the spectral, spatial, or temporal range or resolution of the data. In contrast, the ground-truth data utilized throughout this thesis was — as its name suggests — assumed to be fixed and entirely accurate. This assumption does not truly represent the real-world situation, as the ground-truth data is likely to contain considerable noise. As a result, fitting models to such noisy data can be challenging and give a wrong impression of the model’s predictive power. After all, the notion that feeding low-quality data into a computational system leads to unfavourable outcomes has been recognized even before the emergence of modern computers.</p>
<blockquote class="blockquote">
<p><em>“On two occasions I have been asked [by members of Parliament], ‘Pray, Mr.&nbsp;Babbage, if you put into the machine wrong figures, will the right answers come out?’ … I am not able rightly to apprehend the kind of confusion of ideas that could provoke such a question.”</em></p>
<footer>
Charles Babbage
</footer>
</blockquote>
<p><span class="citation" data-cites="burke2021using">Burke et al. (<a href="#ref-burke2021using" role="doc-biblioref">2021</a>)</span> examined the implications of employing unreliable ground-truth data in remote sensing applications. Utilizing noisy data for calibrating models is not inherently problematic, provided that the data is unbiased. Nevertheless, a discrepancy between the measured nitrogen concentration and the actual tree-wide foliar nitrogen concentration can result in a substantial underestimation of model performance. But where could such noise in the labels come from?</p>
<p>First, the nitrogen measurements might contain substantial noise arising from the sample preparation, equipment failures, and also human error <span class="citation" data-cites="saez2013overview">(<a href="#ref-saez2013overview" role="doc-biblioref">Sáez-Plaza et al. 2013</a>)</span>. Thus, the nitrogen measurements themselves are not an absolute ground truth, even with respect to the leaves.</p>
<p>Second, there exists a considerable disconnect between the nitrogen concentration of the sampled leaves and that of the entire tree. Previous research has shown that there is a high within-tree variance of the foliar nitrogen concentration <span class="citation" data-cites="khalsa2018understanding">(<a href="#ref-khalsa2018understanding" role="doc-biblioref">Khalsa and Brown 2018</a>)</span>. Moreover, the tree crown typically contains more nitrogen than the bottom leaves due to its higher photosynthetic activity. Leaves sampled from a different area than that observed by the drones can also contribute to the noise.</p>
<p>The inadequate representation of the true tree-wide nitrogen content is further emphasized by the band-wise correlation with the nitrogen measurements (<a href="#fig-exploratory">Figure&nbsp;12</a>): although the band-wise correlation appears similar between the canopy and leaf-level data, the magnitude of the correlation is much greater at the leaf level (<span class="math inline">\(r = -0.48\)</span> and <span class="math inline">\(r = -0.32\)</span>, respectively). This strongly indicates a mismatch between leaf-level nitrogen measurements and the true tree-level foliar nitrogen concentration.</p>
<p>In summary, canopy-level reflectance data exhibits a weaker correlation with nitrogen concentration compared to leaf-level reflectance data, possibly due to the sampled leaves not accurately representing the true tree-wide nitrogen content. Consequently, the performance of canopy level models may be underestimated, with their actual performance likely being considerably better.</p>
</section>
<section id="range-of-the-nitrogen-measurements" class="level2" data-number="8.5">
<h2 data-number="8.5" class="anchored" data-anchor-id="range-of-the-nitrogen-measurements"><span class="header-section-number">8.5</span> Range of the nitrogen measurements</h2>
<p>The data utilized in this thesis was derived from a commercial almond orchard, which applied high nitrogen rates in line with recommendations by the Almond Board of California <span class="citation" data-cites="nitrogen2020">(<a href="#ref-nitrogen2020" role="doc-biblioref">Brown et al. 2020</a>)</span>. As a result, the measured foliar nitrogen concentrations almost all fell within a healthy range, between 1.96 and 3.17%, suggesting adequate nitrogen supply for most trees (<a href="#fig-PairsPanel">Figure&nbsp;13</a>). According to <span class="citation" data-cites="micke1996">Micke (<a href="#ref-micke1996" role="doc-biblioref">1996</a>)</span>, the optimal foliar nitrogen concentration would range between 2.2 and 2.5%, meaning that only a few almond trees would benefit from additional fertilization, while many could be considered slightly over-fertilized.</p>
<p>It is essential to note that almonds are alternate bearing crops. In 2022, the Westfield almond orchard experienced a weak bearing year, resulting in higher foliar nitrogen concentrations compared to a strong yielding year due to less foliar nitrogen being mobilized for fruit production. Conversely, the Westwind 2021 orchard had lower nitrogen levels, leading to a higher number of nitrogen-deficient trees. This factor significantly contributes to the varying predictive power of the same model from year to year.</p>
<p>Two critical implications arise from these observations concerning model evaluation. Firstly, the R<sup>2</sup> metric, the most intuitive measure of model performance, is influenced by the response variable’s range, as this range is part of its mathematical definition (<a href="#eq-RegressionPerformanceMetrics">Equation&nbsp;27</a>). This implies that a broader range of measured nitrogen concentration generally results in increased R<sup>2</sup> values and result in a heightened sense of accuracy. Secondly, crop nitrogen status becomes more predictable when leaf nitrogen concentration is low, and plants exhibit nitrogen deficiency. This is attributed to the visual cues of nitrogen deficiency, manifesting as reduced chlorophyll content and visible leaf yellowing <span class="citation" data-cites="hawkesford2012functions">(<a href="#ref-hawkesford2012functions" role="doc-biblioref">Hawkesford et al. 2012</a>)</span>. In contrast, nitrogen over-abundance leads to chlorophyll saturation without clear visual cues, making remote sensing detection more challenging <span class="citation" data-cites="munoz2013review">(<a href="#ref-munoz2013review" role="doc-biblioref">Muñoz-Huerta et al. 2013</a>)</span>.</p>
<p>Several remote sensing studies investigating nitrogen concentrations have been conducted in tandem with nitrogen trials, where the experimental design specifically foresees certain blocks to be fertilized much less than others, or even not to be fertilized at all. This approach generally creates a wider range of foliar nitrogen, particularly at the lower end, which does not accurately represent real farming conditions. Due to the previously stated reasons, this increase range of the nitrogen labels result in a heightened sense of model accuracy. For instance, <span class="citation" data-cites="wang2017non">Wang et al. (<a href="#ref-wang2017non" role="doc-biblioref">2017</a>)</span> predicted foliar nitrogen concentration in pear leaves from an experiment with application rates ranging between 0 and 990 kg N ha<sup>-1</sup> using spectrophotometric measurements. The authors achieved excellent predictions on their test data (R<sup>2</sup> = 0.85) and concluded that their method would enable relatively inexpensive nitrogen monitoring in pear orchards. Similar conclusions based on modelling results using data from fertilization trials can be found throughout the scientific literature and for various crops <span class="citation" data-cites="zha2020improving perich2021proximal pancorbo2021simultaneous">(<a href="#ref-zha2020improving" role="doc-biblioref">Zha et al. 2020</a>, <a href="#ref-pancorbo2021simultaneous" role="doc-biblioref">Pancorbo et al. 2021</a>, <a href="#ref-perich2021proximal" role="doc-biblioref">Perich et al. 2021</a>)</span>.</p>
<p>To ensure the development of remote sensing technologies applicable in commercial agriculture, it is important to test these technologies in settings that accurately reflect commercial farming conditions — especially when the conducted research promises to address the environmental problem of nitrogen leaching, rather than to detect plant stress from nitrogen deficiency. Consequently, further research focusing on remote sensing of crop nitrogen should focus on experimental settings that represent the reality of commercial farms.</p>
<!-- ## Additional remarks -->
<!-- - RS and variable rate fertilization is not the only solution to more sustainable nitrogen management. -->
<!-- - Increasingly popular: Under-sow to mitigate nitrogen leaching. Makes remote sensing more difficult, because undersow reflectance interferes with canopy reflectance. -->
<!-- - Most modern crop genotypes have been selected under non-limiting fertilization conditions [@presterl2003improving]. However, this is not the default condition of plants; many plants can tolerate low nitrogen concentrations -> still the need to adapt breeding/involvement of breeding. For a good evaluation, predicting nitrogen concentration in plants is crucial -> Hyperspectral analysis. -->
</section>
</section>
<section id="sec-conclusion" class="level1" data-number="9">
<h1 data-number="9"><span class="header-section-number">9</span> Conclusion</h1>
<p>In this study, we delved into the remote sensing of foliar nitrogen concentration in Californian almond trees, exploring methodologies that fall into three groups: hand-crafted features, machine learning, and radiative transfer models. Historically, hand-crafted features, particularly single vegetation indices, have been a prevalent approach for modeling crop nitrogen based on reflectance data. However, our findings reveal that vegetation indices fall short compared to the other examined techniques. Machine learning algorithms typically perform well on datasets they’ve been trained on but often lack generalizability due to their excessive adaptability and failure to utilize prior domain knowledge. Radiative transfer models present a promising solution to these limitations, offering a generalizable method for estimating plant characteristics through remote sensing and facilitating crop nitrogen prediction. These models present the challenge of model inversion, which can be addressed through mathematical optimization, utilization of look-up tables, or the application of machine learning techniques. Our study demonstrated that using a multicubic polynomial surrogate model for the cost function can significantly enhance predictions derived from a look-up table, nearly matching the performance of the best-performing machine learning models. Despite the slightly superior predictions generated by machine learning models, radiative transfer models could be enhanced by examining chlorophyll a and b independently at the leaf level or improving the representation of crops’ three-dimensional structure at the canopy level. Regarding spectral data, the range covered is more crucial than the spectral resolution, particularly the inclusion of the short-wave-infrared region for improving crop nitrogen estimation accuracy. Much of the research on remote sensing of crop nitrogen has been conducted in conjunction with nitrogen fertilization trials, potentially leading to an overestimation of model accuracy due to an artificially heightened range of measured nitrogen concentrations. Future research aimed at addressing nitrogen leaching through remote sensing should focus on experimental settings more closely resembling standard commercial agricultural practices.</p>
</section>
<section id="acknowledgement" class="level1 unnumbered">
<h1 class="unnumbered">Acknowledgement</h1>
<p>First and foremost, I would like to express my gratitude to my project supervisors, Prof.&nbsp;Dr.&nbsp;Achim Walter and Prof.&nbsp;Dr.&nbsp;Alireza Pourreza, both for their guidance and for granting me the opportunity to write my master’s thesis at the University of California in Davis. I would also like to express my appreciation to all the members of the <em>Digital Agriculture Laboratory</em> for their unwavering support and camaraderie. A special acknowledgement goes to Momtanu Chakraborty, Yuto Kamiya, and Hamid Jafarbiglu for their invaluable assistance in addressing technical and scientific queries. Finally, I am eternally grateful to my parents, Susannah and Armin, without whom this journey would not have been possible, and to Alice, who always stood by my side.</p>
</section>
<section id="appendix" class="level1 unnumbered">
<h1 class="unnumbered">Appendix</h1>
<p>This appendix features a list of abbreviations and symbols used throughout this thesis (<a href="#tbl-abbreviations">Table&nbsp;9</a>) as well as the numerical values of the performance results of the models tested (<a href="#tbl-ResultsNarea">Table&nbsp;10</a>, <a href="#tbl-ResultsNmass">11</a>, and <a href="#tbl-ResultsHSI">12</a>).</p>
<div id="tbl-abbreviations" class="anchored">
<table class="table">
<caption>Table&nbsp;9: List of abbreviations. The abbreviations used are informed by the closest literature.</caption>
<colgroup>
<col style="width: 28%">
<col style="width: 72%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Abbreviation</strong></th>
<th><strong>Meaning</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(\lambda\)</span></td>
<td>Wavelength</td>
</tr>
<tr class="even">
<td><span class="math inline">\(\boldsymbol \rho\)</span></td>
<td>Reflectance vector</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\rho_\lambda\)</span></td>
<td>Spectral reflectance at wavelength <span class="math inline">\(\lambda\)</span>.</td>
</tr>
<tr class="even">
<td><span class="math inline">\(\boldsymbol \tau\)</span></td>
<td>Transmittance vector</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\tau_\lambda\)</span></td>
<td>Spectral transmittance at wavelength <span class="math inline">\(\lambda\)</span>.</td>
</tr>
<tr class="even">
<td><span class="math inline">\(\text{N}_{\%}\)</span></td>
<td>Mass-based nitrogen, i.e., (foliar) nitrogen concentration.</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\text{N}_{\text {area}}\)</span></td>
<td>Area-based nitrogen.</td>
</tr>
<tr class="even">
<td>LMA</td>
<td>Leaf mass per area.</td>
</tr>
<tr class="odd">
<td>LAI</td>
<td>Leaf area index.</td>
</tr>
<tr class="even">
<td><span class="math inline">\(\mathbf X\)</span></td>
<td>Design matrix, matrix of input variables to a model.</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\mathbf x_{(i)}\)</span></td>
<td>The <span class="math inline">\(i\)</span><sup>th</sup> row of the design matrix <span class="math inline">\(\mathbf X\)</span>, i.e., one single observation.</td>
</tr>
<tr class="even">
<td><span class="math inline">\(\mathbf y\)</span></td>
<td>Response vector, vector of output variables from a model.</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\mathbf {\hat y}\)</span></td>
<td>Estimated values of <span class="math inline">\(\mathbf y\)</span>.</td>
</tr>
<tr class="even">
<td><span class="math inline">\({\bar y}\)</span></td>
<td>Arithmetic mean of all values in <span class="math inline">\(\mathbf y\)</span>.</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\mathcal D\)</span></td>
<td>The entire data set of <span class="math inline">\(\mathbf X\)</span> and <span class="math inline">\(\mathbf y\)</span>.</td>
</tr>
<tr class="even">
<td><span class="math inline">\(n\)</span></td>
<td>Number of observations in <span class="math inline">\(\mathcal D\)</span>.</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(d\)</span></td>
<td>Number of predictor variables, i.e., number of columns in <span class="math inline">\(\mathbf X\)</span>.</td>
</tr>
<tr class="even">
<td><span class="math inline">\(p\)</span></td>
<td>Number of parameters in a model.</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\boldsymbol \Sigma\)</span></td>
<td>Covariance matrix of <span class="math inline">\(\mathbf X\)</span>.</td>
</tr>
<tr class="even">
<td><span class="math inline">\(\boldsymbol \beta\)</span></td>
<td>Coefficient vector (vector of parameters of a linear model).</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\boldsymbol \theta\)</span></td>
<td>Parameter vector (vector of parameters of any model).</td>
</tr>
<tr class="even">
<td><span class="math inline">\(\mu_\mathbf{x}\)</span></td>
<td>Arithmetic mean of <span class="math inline">\(\mathbf{x}\)</span>.</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\sigma_\mathbf{x}\)</span></td>
<td>Standard deviation of <span class="math inline">\(\mathbf{x}\)</span>.</td>
</tr>
<tr class="even">
<td><span class="math inline">\(J\)</span></td>
<td>Cost function.</td>
</tr>
<tr class="odd">
<td><span class="smallcaps">Prospect</span></td>
<td>Leaf-level radiative transfer model.</td>
</tr>
<tr class="even">
<td><span class="smallcaps">Less</span></td>
<td>Canopy-level radiative transfer model.</td>
</tr>
</tbody>
</table>
</div>
<div style="page-break-after: always;"></div>
<div id="tbl-ResultsNarea" class="anchored">
<table class="table">
<caption>Table&nbsp;10: Performances of different modelling methods trying to predict nitrogen per area-based on spectroradiometric data on a leaf level. The root mean squared error (RMSE) and coefficient of determination (R<sup>2</sup>) were computed based on a 20-fold 50 times repeated cross validation, where the mean and the standard deviation are reported (<span class="math inline">\(\mu ± \sigma\)</span>). The compact letter display (CLD) is based on a pairwise Wilcoxon rank sum tests.</caption>
<colgroup>
<col style="width: 53%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"><strong>Method</strong></th>
<th style="text-align: left;"><strong>R<sup>2</sup></strong></th>
<th style="text-align: left;"><strong>RMSE</strong> [g mm<sup>-2</sup>]</th>
<th style="text-align: left;"><strong>CLD</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">NDVI</td>
<td style="text-align: left;">-0.016 ± 0.024</td>
<td style="text-align: left;">0.316 ± 0.0037</td>
<td style="text-align: left;">a</td>
</tr>
<tr class="even">
<td style="text-align: left;">NDRE</td>
<td style="text-align: left;">0.123 ± 0.007</td>
<td style="text-align: left;">0.293 ± 0.0011</td>
<td style="text-align: left;">b</td>
</tr>
<tr class="odd">
<td style="text-align: left;">EVI</td>
<td style="text-align: left;">0.115 ± 0.006</td>
<td style="text-align: left;">0.295 ± 0.001</td>
<td style="text-align: left;">c</td>
</tr>
<tr class="even">
<td style="text-align: left;">GCI</td>
<td style="text-align: left;">-0.055 ± 0.015</td>
<td style="text-align: left;">0.322 ± 0.0023</td>
<td style="text-align: left;">d</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GNDVI</td>
<td style="text-align: left;">-0.048 ± 0.013</td>
<td style="text-align: left;">0.321 ± 0.002</td>
<td style="text-align: left;">d</td>
</tr>
<tr class="even">
<td style="text-align: left;">MCARI</td>
<td style="text-align: left;">0.01 ± 0.014</td>
<td style="text-align: left;">0.312 ± 0.0021</td>
<td style="text-align: left;">e</td>
</tr>
<tr class="odd">
<td style="text-align: left;">RECI</td>
<td style="text-align: left;">0.128 ± 0.004</td>
<td style="text-align: left;">0.292 ± 8e-04</td>
<td style="text-align: left;">f</td>
</tr>
<tr class="even">
<td style="text-align: left;">TCARI</td>
<td style="text-align: left;">-0.016 ± 0.005</td>
<td style="text-align: left;">0.316 ± 7e-04</td>
<td style="text-align: left;">a</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WDRVI</td>
<td style="text-align: left;">0.011 ± 0.016</td>
<td style="text-align: left;">0.312 ± 0.0025</td>
<td style="text-align: left;">e</td>
</tr>
<tr class="even">
<td style="text-align: left;">R434</td>
<td style="text-align: left;">0.255 ± 0.008</td>
<td style="text-align: left;">0.27 ± 0.0014</td>
<td style="text-align: left;">g</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Combined vegetation indices</td>
<td style="text-align: left;">0.464 ± 0.019</td>
<td style="text-align: left;">0.229 ± 0.004</td>
<td style="text-align: left;">h</td>
</tr>
<tr class="even">
<td style="text-align: left;">Recursive feature selection</td>
<td style="text-align: left;">0.544 ± 0.01</td>
<td style="text-align: left;">0.211 ± 0.0024</td>
<td style="text-align: left;">i</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Square feature selection</td>
<td style="text-align: left;">0.61 ± 0.007</td>
<td style="text-align: left;">0.195 ± 0.0019</td>
<td style="text-align: left;">j</td>
</tr>
<tr class="even">
<td style="text-align: left;">Lasso</td>
<td style="text-align: left;">0.542 ± 0.04</td>
<td style="text-align: left;">0.212 ± 0.0089</td>
<td style="text-align: left;">i</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ridge</td>
<td style="text-align: left;">0.496 ± 0.007</td>
<td style="text-align: left;">0.222 ± 0.0015</td>
<td style="text-align: left;">k</td>
</tr>
<tr class="even">
<td style="text-align: left;">Elastic-net</td>
<td style="text-align: left;">0.537 ± 0.049</td>
<td style="text-align: left;">0.213 ± 0.0106</td>
<td style="text-align: left;">i</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Principal components</td>
<td style="text-align: left;">0.614 ± 0.021</td>
<td style="text-align: left;">0.195 ± 0.0052</td>
<td style="text-align: left;">jl</td>
</tr>
<tr class="even">
<td style="text-align: left;">Partial least squares</td>
<td style="text-align: left;">0.626 ± 0.02</td>
<td style="text-align: left;">0.192 ± 0.0052</td>
<td style="text-align: left;">l</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Random forest</td>
<td style="text-align: left;">0.492 ± 0.01</td>
<td style="text-align: left;">0.223 ± 0.0021</td>
<td style="text-align: left;">k</td>
</tr>
<tr class="even">
<td style="text-align: left;">Extreme gradient boosting</td>
<td style="text-align: left;">0.548 ± 0.03</td>
<td style="text-align: left;">0.21 ± 0.007</td>
<td style="text-align: left;">i</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Cubist</td>
<td style="text-align: left;">0.276 ± 0.046</td>
<td style="text-align: left;">0.266 ± 0.0083</td>
<td style="text-align: left;">g</td>
</tr>
<tr class="even">
<td style="text-align: left;"><span class="smallcaps">Prospect-Pro</span></td>
<td style="text-align: left;">0.684 ± 0.004</td>
<td style="text-align: left;">0.176 ± 0.0012</td>
<td style="text-align: left;">m</td>
</tr>
</tbody>
</table>
</div>
<div style="page-break-after: always;"></div>
<div id="tbl-ResultsNmass" class="anchored">
<table class="table">
<caption>Table&nbsp;11: Performances of different modelling methods trying to predict foliar nitrogen concentration based on spectroradiometric data on a leaf level. The root mean squared error (RMSE) and coefficient of determination (R<sup>2</sup>) were computed based on a 20-fold 50 times repeated cross validation, where the mean and the standard deviation are reported (<span class="math inline">\(\mu ± \sigma\)</span>). The compact letter display (CLD) is based on a pairwise Wilcoxon rank sum tests.</caption>
<colgroup>
<col style="width: 53%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"><strong>Method</strong></th>
<th style="text-align: left;"><strong>R<sup>2</sup></strong></th>
<th style="text-align: left;"><strong>RMSE</strong> [%]</th>
<th style="text-align: left;"><strong>CLD</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">NDVI</td>
<td style="text-align: left;">0.184 ± 0.005</td>
<td style="text-align: left;">0.22 ± 7e-04</td>
<td style="text-align: left;">a</td>
</tr>
<tr class="even">
<td style="text-align: left;">NDRE</td>
<td style="text-align: left;">0.177 ± 0.005</td>
<td style="text-align: left;">0.221 ± 7e-04</td>
<td style="text-align: left;">bc</td>
</tr>
<tr class="odd">
<td style="text-align: left;">EVI</td>
<td style="text-align: left;">0.177 ± 0.006</td>
<td style="text-align: left;">0.221 ± 7e-04</td>
<td style="text-align: left;">bc</td>
</tr>
<tr class="even">
<td style="text-align: left;">GCI</td>
<td style="text-align: left;">0.089 ± 0.006</td>
<td style="text-align: left;">0.233 ± 8e-04</td>
<td style="text-align: left;">d</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GNDVI</td>
<td style="text-align: left;">0.093 ± 0.007</td>
<td style="text-align: left;">0.232 ± 8e-04</td>
<td style="text-align: left;">d</td>
</tr>
<tr class="even">
<td style="text-align: left;">MCARI</td>
<td style="text-align: left;">0.151 ± 0.005</td>
<td style="text-align: left;">0.225 ± 7e-04</td>
<td style="text-align: left;">e</td>
</tr>
<tr class="odd">
<td style="text-align: left;">RECI</td>
<td style="text-align: left;">0.174 ± 0.007</td>
<td style="text-align: left;">0.222 ± 9e-04</td>
<td style="text-align: left;">b</td>
</tr>
<tr class="even">
<td style="text-align: left;">TCARI</td>
<td style="text-align: left;">0.148 ± 0.006</td>
<td style="text-align: left;">0.225 ± 8e-04</td>
<td style="text-align: left;">e</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WDRVI</td>
<td style="text-align: left;">0.177 ± 0.005</td>
<td style="text-align: left;">0.221 ± 6e-04</td>
<td style="text-align: left;">bc</td>
</tr>
<tr class="even">
<td style="text-align: left;">R434</td>
<td style="text-align: left;">0.065 ± 0.007</td>
<td style="text-align: left;">0.236 ± 9e-04</td>
<td style="text-align: left;">f</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Combined vegetation indices</td>
<td style="text-align: left;">0.14 ± 0.02</td>
<td style="text-align: left;">0.226 ± 0.0026</td>
<td style="text-align: left;">e</td>
</tr>
<tr class="even">
<td style="text-align: left;">Recursive feature selection</td>
<td style="text-align: left;">0.235 ± 0.006</td>
<td style="text-align: left;">0.213 ± 9e-04</td>
<td style="text-align: left;">g</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Square feature selection</td>
<td style="text-align: left;">0.22 ± 0.015</td>
<td style="text-align: left;">0.215 ± 0.0021</td>
<td style="text-align: left;">h</td>
</tr>
<tr class="even">
<td style="text-align: left;">Lasso</td>
<td style="text-align: left;">0.407 ± 0.012</td>
<td style="text-align: left;">0.188 ± 0.002</td>
<td style="text-align: left;">ij</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ridge</td>
<td style="text-align: left;">0.182 ± 0.009</td>
<td style="text-align: left;">0.221 ± 0.0012</td>
<td style="text-align: left;">ac</td>
</tr>
<tr class="even">
<td style="text-align: left;">Elastic-net</td>
<td style="text-align: left;">0.407 ± 0.01</td>
<td style="text-align: left;">0.188 ± 0.0016</td>
<td style="text-align: left;">i</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Principal components</td>
<td style="text-align: left;">0.417 ± 0.017</td>
<td style="text-align: left;">0.186 ± 0.0028</td>
<td style="text-align: left;">j</td>
</tr>
<tr class="even">
<td style="text-align: left;">Partial least squares</td>
<td style="text-align: left;">0.493 ± 0.022</td>
<td style="text-align: left;">0.174 ± 0.0037</td>
<td style="text-align: left;">k</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Random forest</td>
<td style="text-align: left;">0.27 ± 0.02</td>
<td style="text-align: left;">0.208 ± 0.0028</td>
<td style="text-align: left;">l</td>
</tr>
<tr class="even">
<td style="text-align: left;">Extreme gradient boosting</td>
<td style="text-align: left;">0.288 ± 0.05</td>
<td style="text-align: left;">0.206 ± 0.0072</td>
<td style="text-align: left;">l</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Cubist</td>
<td style="text-align: left;">0.384 ± 0.034</td>
<td style="text-align: left;">0.191 ± 0.0053</td>
<td style="text-align: left;">i</td>
</tr>
<tr class="even">
<td style="text-align: left;">Prospect-Pro</td>
<td style="text-align: left;">0.453 ± 0.005</td>
<td style="text-align: left;">0.18 ± 8e-04</td>
<td style="text-align: left;">m</td>
</tr>
</tbody>
</table>
</div>
<div style="page-break-after: always;"></div>
<div id="tbl-ResultsHSI" class="anchored">
<table class="table">
<caption>Table&nbsp;12: Performances of different modelling methods trying to predict foliar nitrogen concentration based on hyperspectral images. The root mean squared error (RMSE) and coefficient of determination (R<sup>2</sup>) were computed based on a 20-fold 50 times repeated cross validation, where the mean and the standard deviation are reported (<span class="math inline">\(\mu ± \sigma\)</span>). The compact letter display (CLD) is based on a pairwise Wilcoxon rank sum tests.</caption>
<colgroup>
<col style="width: 53%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"><strong>Method</strong></th>
<th style="text-align: left;"><strong>R<sup>2</sup></strong></th>
<th style="text-align: left;"><strong>RMSE</strong> [%]</th>
<th style="text-align: left;"><strong>CLD</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">NDVI</td>
<td style="text-align: left;">-0.026 ± 0.027</td>
<td style="text-align: left;">0.257 ± 0.0034</td>
<td style="text-align: left;">a</td>
</tr>
<tr class="even">
<td style="text-align: left;">NDRE</td>
<td style="text-align: left;">-0.003 ± 0.026</td>
<td style="text-align: left;">0.254 ± 0.0033</td>
<td style="text-align: left;">bc</td>
</tr>
<tr class="odd">
<td style="text-align: left;">EVI</td>
<td style="text-align: left;">-0.082 ± 0.011</td>
<td style="text-align: left;">0.264 ± 0.0013</td>
<td style="text-align: left;">d</td>
</tr>
<tr class="even">
<td style="text-align: left;">GCI</td>
<td style="text-align: left;">-0.01 ± 0.026</td>
<td style="text-align: left;">0.255 ± 0.0032</td>
<td style="text-align: left;">abc</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GNDVI</td>
<td style="text-align: left;">-0.012 ± 0.027</td>
<td style="text-align: left;">0.255 ± 0.0034</td>
<td style="text-align: left;">abc</td>
</tr>
<tr class="even">
<td style="text-align: left;">MCARI</td>
<td style="text-align: left;">-0.001 ± 0.019</td>
<td style="text-align: left;">0.254 ± 0.0024</td>
<td style="text-align: left;">b</td>
</tr>
<tr class="odd">
<td style="text-align: left;">RECI</td>
<td style="text-align: left;">-0.007 ± 0.023</td>
<td style="text-align: left;">0.254 ± 0.0029</td>
<td style="text-align: left;">abc</td>
</tr>
<tr class="even">
<td style="text-align: left;">TCARI</td>
<td style="text-align: left;">0.068 ± 0.006</td>
<td style="text-align: left;">0.245 ± 8e-04</td>
<td style="text-align: left;">e</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WDRVI</td>
<td style="text-align: left;">-0.021 ± 0.023</td>
<td style="text-align: left;">0.256 ± 0.0029</td>
<td style="text-align: left;">ac</td>
</tr>
<tr class="even">
<td style="text-align: left;">R434</td>
<td style="text-align: left;">0.198 ± 0.007</td>
<td style="text-align: left;">0.227 ± 9e-04</td>
<td style="text-align: left;">f</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Combined vegetation indices</td>
<td style="text-align: left;">0.237 ± 0.011</td>
<td style="text-align: left;">0.221 ± 0.0015</td>
<td style="text-align: left;">g</td>
</tr>
<tr class="even">
<td style="text-align: left;">Recursive feature selection</td>
<td style="text-align: left;">0.142 ± 0.022</td>
<td style="text-align: left;">0.235 ± 0.003</td>
<td style="text-align: left;">h</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Square feature selection</td>
<td style="text-align: left;">0.256 ± 0.014</td>
<td style="text-align: left;">0.218 ± 0.0021</td>
<td style="text-align: left;">i</td>
</tr>
<tr class="even">
<td style="text-align: left;">Lasso</td>
<td style="text-align: left;">0.249 ± 0.012</td>
<td style="text-align: left;">0.22 ± 0.0018</td>
<td style="text-align: left;">i</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ridge</td>
<td style="text-align: left;">0.206 ± 0.008</td>
<td style="text-align: left;">0.226 ± 0.0012</td>
<td style="text-align: left;">j</td>
</tr>
<tr class="even">
<td style="text-align: left;">Elastic-net</td>
<td style="text-align: left;">0.25 ± 0.012</td>
<td style="text-align: left;">0.219 ± 0.0017</td>
<td style="text-align: left;">i</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Principal components</td>
<td style="text-align: left;">0.236 ± 0.013</td>
<td style="text-align: left;">0.221 ± 0.0019</td>
<td style="text-align: left;">g</td>
</tr>
<tr class="even">
<td style="text-align: left;">Partial least squares</td>
<td style="text-align: left;">0.228 ± 0.013</td>
<td style="text-align: left;">0.223 ± 0.0019</td>
<td style="text-align: left;">g</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Random forest</td>
<td style="text-align: left;">0.102 ± 0.032</td>
<td style="text-align: left;">0.24 ± 0.0042</td>
<td style="text-align: left;">k</td>
</tr>
<tr class="even">
<td style="text-align: left;">Extreme gradient boosting</td>
<td style="text-align: left;">0.127 ± 0.018</td>
<td style="text-align: left;">0.237 ± 0.0024</td>
<td style="text-align: left;">hl</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Cubist</td>
<td style="text-align: left;">0.175 ± 0.013</td>
<td style="text-align: left;">0.23 ± 0.0018</td>
<td style="text-align: left;">m</td>
</tr>
<tr class="even">
<td style="text-align: left;"><span class="smallcaps">Less</span></td>
<td style="text-align: left;">0.116 ± 0.011</td>
<td style="text-align: left;">0.237 ± 0.0015</td>
<td style="text-align: left;">kl</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><span class="smallcaps">Less</span> + interpolation</td>
<td style="text-align: left;">0.235 ± 0.009</td>
<td style="text-align: left;">0.22 ± 0.0013</td>
<td style="text-align: left;">g</td>
</tr>
</tbody>
</table>
</div>
</section>
<section id="references" class="level1 unnumbered">


</section>


<div id="quarto-appendix"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-line-spacing="2" role="list">
<div id="ref-almondsustainability2020" class="csl-entry" role="listitem">
ABC. 2020. <a href="https://www.almonds.com/sites/default/files/2020-12/2020%20Growing%20Good%20Almond%20Sustainability.pdf">Growing good – almond sustainability</a>. Almond Board of California, 1150 Ninth Street, Suite 1500, Modesto, CA95354.
</div>
<div id="ref-akaike1998information" class="csl-entry" role="listitem">
Akaike, H. 1998. <a href="https://doi.org/10.1007/978-1-4612-1694-0_15">Information theory and an extension of the maximum likelihood principle</a>. Selected papers of Hirotugu Akaike:199–213.
</div>
<div id="ref-allen1973transmission" class="csl-entry" role="listitem">
Allen, W. A. 1973. <a href="https://doi.org/10.1364/JOSA.63.000664">Transmission of isotropic light across a dielectric surface in two and three dimensions</a>. JOSA 63:664–666.
</div>
<div id="ref-allen1969interaction" class="csl-entry" role="listitem">
Allen, W. A., H. W. Gausman, A. J. Richardson, and J. R. Thomas. 1969. <a href="https://doi.org/10.1364/JOSA.59.001376">Interaction of isotropic light with a compact plant leaf</a>. JOSA 59:1376–1379.
</div>
<div id="ref-allen1973willstatter" class="csl-entry" role="listitem">
Allen, W. A., H. Gausman, and A. Richardson. 1973. <a href="https://doi.org/10.1364/AO.12.002448">Willst<span>ä</span>tter-stoll theory of leaf reflectance evaluated by ray tracing</a>. Applied Optics 12:2448–2453.
</div>
<div id="ref-asner2009airborne" class="csl-entry" role="listitem">
Asner, G. P., and R. E. Martin. 2009. <a href="https://doi.org/10.1890/070152">Airborne spectranomics: Mapping canopy chemical and taxonomic diversity in tropical forests</a>. Frontiers in Ecology and the Environment 7:269–276.
</div>
<div id="ref-atzberger2011confining" class="csl-entry" role="listitem">
Atzberger, C., K. Richter, F. Vuolo, R. Darvishzadeh, and M. Schlerf. 2011. <a href="https://doi.org/10.1117/12.898479">Why confining to vegetation indices? Exploiting the potential of improved spectral observations using radiative transfer models</a>. Pages 263–278 Remote sensing for agriculture, ecosystems, and hydrology XIII. SPIE.
</div>
<div id="ref-bailey2019helios" class="csl-entry" role="listitem">
Bailey, B. N. 2019. <a href="https://doi.org/10.3389/fpls.2019.01185">Helios: A scalable 3D plant and environmental biophysical modeling framework</a>. Frontiers in Plant Science 10:1185.
</div>
<div id="ref-bailey2017rapid" class="csl-entry" role="listitem">
Bailey, B. N., and W. F. Mahaffee. 2017. <a href="https://doi.org/10.1016/j.rse.2017.03.011">Rapid measurement of the three-dimensional distribution of leaf orientation and the leaf angle probability density function using terrestrial LiDAR scanning</a>. Remote Sensing of Environment 194:63–76.
</div>
<div id="ref-bailey2020one" class="csl-entry" role="listitem">
Bailey, B. N., M. A. Ponce de León, and E. S. Krayenhoff. 2020. <a href="https://doi.org/10.5194/gmd-13-4789-2020">One-dimensional models of radiation transfer in heterogeneous canopies: A review, re-evaluation, and improved model</a>. Geoscientific Model Development 13:4789–4808.
</div>
<div id="ref-baret1997estimation" class="csl-entry" role="listitem">
Baret, F., and T. Fourty. 1997. <a href="https://doi.org/10.1051/agro:19970903">Estimation of leaf water content and specific leaf weight from reflectance and transmittance measurements</a>. Agronomie 17:455–464.
</div>
<div id="ref-baret2007quantification" class="csl-entry" role="listitem">
Baret, F., V. Houlès, and M. Guerif. 2007. <a href="https://doi.org/10.1093/jxb/erl231">Quantification of plant stress using remote sensing observations and crop models: The case of nitrogen management</a>. Journal of experimental botany 58:869–880.
</div>
<div id="ref-baret1992modeled" class="csl-entry" role="listitem">
Baret, F., S. Jacquemoud, G. Guyot, and C. Leprieur. 1992. <a href="https://doi.org/10.1016/0034-4257(92)90073-S">Modeled analysis of the biophysical nature of spectral shifts and comparison with information content of broad bands</a>. Remote Sensing of Environment 41:133–142.
</div>
<div id="ref-baret1993soil" class="csl-entry" role="listitem">
Baret, F., S. Jacquemoud, and J. Hanocq. 1993. <a href="https://doi.org/10.1080/02757259309532166">The soil line concept in remote sensing</a>. Remote Sensing Reviews 7:65–82.
</div>
<div id="ref-barnes2000coincident" class="csl-entry" role="listitem">
Barnes, E., T. Clarke, S. Richards, P. Colaizzi, J. Haberland, M. Kostrzewski, P. Waller, C. Choi, E. Riley, T. Thompson, and others. 2000. Coincident detection of crop water stress, nitrogen status and canopy density using ground based multispectral data. Page 6 Proceedings of the fifth international conference on precision agriculture, bloomington, MN, USA.
</div>
<div id="ref-beer1852bestimmung" class="csl-entry" role="listitem">
Beer, A. 1852. <span class="nocase">Bestimmung der absorption des rothen lichts in farbigen flussigkeiten</span>. Ann. Physik 162:78–88.
</div>
<div id="ref-bennett2003optimization" class="csl-entry" role="listitem">
Bennett, K., and M. Embrechts. 2003. An optimization perspective on kernel partial least squares regression. Nato Science Series sub series III computer and systems sciences 190:227–250.
</div>
<div id="ref-berger2018evaluation" class="csl-entry" role="listitem">
Berger, K., C. Atzberger, M. Danner, G. D’Urso, W. Mauser, F. Vuolo, and T. Hank. 2018. <a href="https://doi.org/10.3390/rs10010085">Evaluation of the PROSAIL model capabilities for future hyperspectral model environments: A review study</a>. Remote Sensing 10:85.
</div>
<div id="ref-berger2020crop" class="csl-entry" role="listitem">
Berger, K., J. Verrelst, J.-B. Feret, Z. Wang, M. Wocher, M. Strathmann, M. Danner, W. Mauser, and T. Hank. 2020. <a href="https://doi.org/10.1016/j.rse.2020.111758">Crop nitrogen monitoring: Recent progress and principal developments in the context of imaging spectroscopy missions</a>. Remote Sensing of Environment 242:111758.
</div>
<div id="ref-bioucas2013hyperspectral" class="csl-entry" role="listitem">
Bioucas-Dias, J. M., A. Plaza, G. Camps-Valls, P. Scheunders, N. Nasrabadi, and J. Chanussot. 2013. <a href="https://doi.org/10.1109/MGRS.2013.2244672">Hyperspectral remote sensing data analysis and future challenges</a>. IEEE Geoscience and remote sensing magazine 1:6–36.
</div>
<div id="ref-bjerga2018california" class="csl-entry" role="listitem">
Bjerga, A., D. Cohen, and C. Hoffman. 2018. California almonds are back after four years of brutal drought. Bloomberg.
</div>
<div id="ref-bloem2005influence" class="csl-entry" role="listitem">
Bloem, E., S. Haneklaus, and E. Schnug. 2005. <a href="https://doi.org/10.1081/PLN-200026433">Influence of nitrogen and sulfur fertilization on the alliin content of onions and garlic</a>. Journal of Plant Nutrition 27:1827–1839.
</div>
<div id="ref-blumthaler1997increase" class="csl-entry" role="listitem">
Blumthaler, M., W. Ambach, and R. Ellinger. 1997. <a href="https://doi.org/10.1016/S1011-1344(96)00018-8">Increase in solar UV radiation with altitude</a>. Journal of photochemistry and Photobiology B: Biology 39:130–134.
</div>
<div id="ref-bolster1996determination" class="csl-entry" role="listitem">
Bolster, K. L., M. E. Martin, and J. D. Aber. 1996. <a href="https://doi.org/10.1139/x26-068">Determination of carbon fraction and nitrogen concentration in tree foliage by near infrared reflectances: A comparison of statistical methods</a>. Canadian journal of forest research 26:590–600.
</div>
<div id="ref-borchers2022pracma" class="csl-entry" role="listitem">
Borchers, H. W. 2022. <a href="https://CRAN.R-project.org/package=pracma">Pracma: Practical numerical math functions</a>.
</div>
<div id="ref-breiman2001" class="csl-entry" role="listitem">
Breiman, L. 2001. <a href="https://doi.org/10.1023/A:1010933404324">Random forests</a>. Machine Learning 45:5–32.
</div>
<div id="ref-nitrogen2020" class="csl-entry" role="listitem">
Brown, P. H., S. Saa, S. Muhammad, and S. D. Khalsa. 2020. Nitrogen best management practices. Almond Board of California, 1150 Ninth Street, Suite 1500, Modesto, CA 95354.
</div>
<div id="ref-bruno2021outlook" class="csl-entry" role="listitem">
Bruno, E. M., B. Goodrich, and R. J. Sexton. 2021. The outlook for california’s almond market. Calif. Almond Acreage Rep 24:9–11.
</div>
<div id="ref-bruno2005crc" class="csl-entry" role="listitem">
Bruno, T. J., and P. D. Svoronos. 2005. CRC handbook of fundamental spectroscopic correlation charts. CRC Press.
</div>
<div id="ref-bubeck2023sparks" class="csl-entry" role="listitem">
Bubeck, S., V. Chandrasekaran, R. Eldan, J. Gehrke, E. Horvitz, E. Kamar, P. Lee, Y. T. Lee, Y. Li, S. Lundberg, and others. 2023. <a href="https://doi.org/10.48550/arXiv.2303.12712">Sparks of artificial general intelligence: Early experiments with gpt-4</a>. arXiv preprint arXiv:2303.12712.
</div>
<div id="ref-burke2021using" class="csl-entry" role="listitem">
Burke, M., A. Driscoll, D. B. Lobell, and S. Ermon. 2021. <a href="https://doi.org/10.1126/science.abe8628">Using satellite imagery to understand and promote sustainable development</a>. Science 371:eabe8628.
</div>
<div id="ref-Byrnes2009" class="csl-entry" role="listitem">
Byrnes, J. 2009. <a href="https://doi.org/10.1007/978-1-4020-9253-4">Unexploded ordnance detection and mitigation</a>. Springer Science &amp; Business Media.
</div>
<div id="ref-camino2021detection" class="csl-entry" role="listitem">
Camino, C., R. Calderón, S. Parnell, H. Dierkes, Y. Chemin, M. Román-Écija, M. Montes-Borrego, B. B. Landa, J. A. Navas-Cortes, P. J. Zarco-Tejada, and others. 2021. <a href="https://doi.org/10.1016/j.rse.2021.112420">Detection of xylella fastidiosa in almond orchards by synergic use of an epidemic spread model and remotely sensed plant traits</a>. Remote Sensing of Environment 260:112420.
</div>
<div id="ref-camino2018improved" class="csl-entry" role="listitem">
Camino, C., V. González-Dugo, P. Hernández, J. Sillero, and P. J. Zarco-Tejada. 2018. <a href="https://doi.org/10.1016/j.jag.2018.04.013">Improved nitrogen retrievals with airborne-derived fluorescence and plant traits quantified from VNIR-SWIR hyperspectral imagery in the context of precision agriculture</a>. International journal of applied earth observation and geoinformation 70:105–117.
</div>
<div id="ref-camps2011remote" class="csl-entry" role="listitem">
Camps-Valls, G., D. Tuia, L. Gómez-Chova, S. Jiménez, and J. Malo. 2011. <a href="https://doi.org/10.1007/978-3-031-02247-0_1">Remote sensing image processing</a>. Synthesis Lectures on Image, Video, and Multimedia Processing 5:1–192.
</div>
<div id="ref-castaldi2016data" class="csl-entry" role="listitem">
Castaldi, F., A. Castrignanò, and R. Casa. 2016. <a href="https://doi.org/10.1080/01431161.2016.1212423">A data fusion and spatial data analysis approach for the estimation of wheat grain nitrogen uptake from satellite data</a>. International Journal of Remote Sensing 37:4317–4336.
</div>
<div id="ref-almonds2020" class="csl-entry" role="listitem">
CDFA. 2020. California agriculture exports 2019-2020. <span>California Department of Food and Argiculture</span>, 1220 N Street, Sacramento, CA 95812.
</div>
<div id="ref-ceresimaging" class="csl-entry" role="listitem">
Ceres Imaging. 2022. <a href="https://www.ceresimaging.net/water-stress-index">Water stress index</a>. Webpage.
</div>
<div id="ref-chapin1987plant" class="csl-entry" role="listitem">
Chapin, F. S., A. J. Bloom, C. B. Field, and R. H. Waring. 1987. <a href="https://doi.org/10.2307/1310177">Plant responses to multiple environmental factors</a>. Bioscience 37:49–57.
</div>
<div id="ref-chen2016xgboost" class="csl-entry" role="listitem">
Chen, T., and C. Guestrin. 2016. XGBoost: A scalable tree boosting system. Pages 785–794 Proceedings of the 22nd acm sigkdd international conference on knowledge discovery and data mining.
</div>
<div id="ref-chen2022" class="csl-entry" role="listitem">
Chen, T., T. He, M. Benesty, V. Khotilovich, Y. Tang, H. Cho, K. Chen, R. Mitchell, I. Cano, T. Zhou, M. Li, J. Xie, M. Lin, Y. Geng, Y. Li, and J. Yuan. 2022. <a href="https://CRAN.R-project.org/package=xgboost">Xgboost: Extreme gradient boosting</a>.
</div>
<div id="ref-cheng2014estimating" class="csl-entry" role="listitem">
Cheng, X., G. Yang, X. Xu, T. Chen, Z. Li, H. Feng, D. Wang, and others. 2014. <a href="https://doi.org/10.3964/j.issn.1000-0593(2014)12-3391-06">Estimating canopy water content in wheat based on new vegetation water index</a>. Spectroscopy and Spectral Analysis 34:3391–3396.
</div>
<div id="ref-chicco2021" class="csl-entry" role="listitem">
Chicco, D., M. J. Warrens, and G. Jurman. 2021. <a href="https://doi.org/10.7717/peerj-cs.623">The coefficient of determination r-squared is more informative than <span>SMAPE</span>, <span>MAE</span>, <span>MAPE</span>, <span>MSE</span> and <span>RMSE</span> in regression analysis evaluation</a>. Computer Science 7:e623.
</div>
<div id="ref-chlingaryan2018machine" class="csl-entry" role="listitem">
Chlingaryan, A., S. Sukkarieh, and B. Whelan. 2018. <a href="https://doi.org/10.1016/j.compag.2018.05.012">Machine learning approaches for crop yield prediction and nitrogen status estimation in precision agriculture: A review</a>. Computers and Electronics in Agriculture 151:61–69.
</div>
<div id="ref-ciani2020comparison" class="csl-entry" role="listitem">
Ciani, A. J., C. H. Grein, W. L. Sarney, S. P. Svensson, D. V. Donetski, and G. L. Belenky. 2020. <a href="https://doi.org/10.1063/5.0017924">A comparison of indium arsenide antimonide and mercury cadmium telluride as long wavelength infrared detector materials</a>. Journal of Applied Physics 128:075704.
</div>
<div id="ref-costa2020new" class="csl-entry" role="listitem">
Costa, L., L. Nunes, and Y. Ampatzidis. 2020. <a href="https://doi.org/10.1016/j.compag.2020.105334">A new visible band index (vNDVI) for estimating NDVI values on RGB images utilizing genetic algorithms</a>. Computers and Electronics in Agriculture 172:105334.
</div>
<div id="ref-cui2017comparison" class="csl-entry" role="listitem">
Cui, S., and K. Zhou. 2017. <a href="https://doi.org/10.1007/s12145-016-0281-3">A comparison of the predictive potential of various vegetation indices for leaf chlorophyll content</a>. Earth Science Informatics 10:169–181.
</div>
<div id="ref-daughtry2000estimating" class="csl-entry" role="listitem">
Daughtry, C. S., C. Walthall, M. Kim, E. B. De Colstoun, and J. McMurtrey Iii. 2000. <a href="https://doi.org/10.1016/S0034-4257(00)00113-9">Estimating corn leaf chlorophyll concentration from leaf and canopy reflectance</a>. Remote Sensing of Environment 74:229–239.
</div>
<div id="ref-daun2017" class="csl-entry" role="listitem">
Daun, K. J. 2017. <a href="https://doi.org/10.1007/978-3-319-32003-8_64-1">Inverse problems in radiative transfer</a>. Pages 1–51. Springer International Publishing.
</div>
<div id="ref-de2016leaf" class="csl-entry" role="listitem">
de la Riva, E. G., M. Olmo, H. Poorter, J. L. Ubera, and R. Villar. 2016. <a href="https://doi.org/10.1371/journal.pone.0148788">Leaf mass per area (LMA) and its relationship with leaf structure and anatomy in 34 mediterranean woody species along a water availability gradient</a>. PloS one 11:e0148788.
</div>
<div id="ref-degrave2022magnetic" class="csl-entry" role="listitem">
Degrave, J., F. Felici, J. Buchli, M. Neunert, B. Tracey, F. Carpanese, T. Ewalds, R. Hafner, A. Abdolmaleki, D. de Las Casas, and others. 2022. <a href="https://doi.org/10.1038/s41586-021-04301-9">Magnetic control of tokamak plasmas through deep reinforcement learning</a>. Nature 602:414–419.
</div>
<div id="ref-doktor2014extraction" class="csl-entry" role="listitem">
Doktor, D., A. Lausch, D. Spengler, and M. Thurner. 2014. <a href="https://doi.org/10.3390/rs61212247">Extraction of plant physiological status from hyperspectral signatures using machine learning methods</a>. Remote Sensing 6:12247–12274.
</div>
<div id="ref-california2015california" class="csl-entry" role="listitem">
DWR. 2015. California climate science and data for water resources management. California Department of Water Resources, 715 P Street Sacramento, CA 95814.
</div>
<div id="ref-watersupply2022" class="csl-entry" role="listitem">
DWR. 2022. California’s water supply strategy – adapting to a hotter, drier future. California Department of Water Resources, 715 P Street, Sacramento, CA 95814.
</div>
<div id="ref-dyson2021" class="csl-entry" role="listitem">
Dyson, M. 2021. <a href="https://www.idtechex.com/en/research-article/competing-short-wavelength-infra-red-sensing-technologies/24918">Competing short-wavelength infra-red sensing technologies</a>. Blog post.
</div>
<div id="ref-elmasry2010principles" class="csl-entry" role="listitem">
El Masry, G., and D.-W. Sun. 2010. <a href="https://doi.org/10.1016/C2009-0-01853-4">Principles of hyperspectral imaging technology</a>. Pages 3–43 Hyperspectral imaging for food quality analysis and control. Elsevier.
</div>
<div id="ref-evans1989photosynthesis" class="csl-entry" role="listitem">
Evans, J. R. 1989. <a href="https://doi.org/10.1007/BF00377192">Photosynthesis and nitrogen relationships in leaves of C3 plants</a>. Oecologia 78:9–19.
</div>
<div id="ref-feret2021prospect" class="csl-entry" role="listitem">
Féret, J.-B., K. Berger, F. De Boissieu, and Z. Malenovskỳ. 2021. <a href="https://doi.org/10.1016/j.rse.2020.112173">PROSPECT-PRO for estimating content of nitrogen-containing leaf proteins and other carbon-based constituents</a>. Remote Sensing of Environment 252:112173.
</div>
<div id="ref-feret2022prosail" class="csl-entry" role="listitem">
Féret, J.-B., and F. de Boissieu. 2022b. <a href="https://gitlab.com/jbferet/prosail">Prosail: PROSAIL leaf and canopy radiative transfer model and inversion routines</a>.
</div>
<div id="ref-feret2022package" class="csl-entry" role="listitem">
Féret, J.-B., and F. de Boissieu. 2022a. <a href="https://gitlab.com/jbferet/prospect">Prospect: PROSPECT leaf radiative transfer model and inversion routines</a>.
</div>
<div id="ref-feret2008prospect" class="csl-entry" role="listitem">
Féret, J.-B., C. François, G. P. Asner, A. A. Gitelson, R. E. Martin, L. P. Bidel, S. L. Ustin, G. Le Maire, and S. Jacquemoud. 2008. <a href="https://doi.org/10.1016/j.rse.2008.02.012">PROSPECT-4 and 5: Advances in the leaf optical properties model separating photosynthetic pigments</a>. Remote Sensing of Environment 112:3030–3043.
</div>
<div id="ref-fourty1998spectral" class="csl-entry" role="listitem">
Fourty, T., and F. Baret. 1998. <a href="https://doi.org/10.1080/014311698215441">On spectral estimates of fresh leaf biochemistry</a>. International Journal of Remote Sensing 19:1283–1297.
</div>
<div id="ref-fourty1996leaf" class="csl-entry" role="listitem">
Fourty, T., F. Baret, S. Jacquemoud, G. Schmuck, and J. Verdebout. 1996. <a href="https://doi.org/10.1016/0034-4257(95)00234-0">Leaf optical properties with explicit description of its biochemical composition: Direct and inverse problems</a>. Remote Sensing of Environment 56:104–117.
</div>
<div id="ref-fresnel1821" class="csl-entry" role="listitem">
Fresnel, A. 1821. Note sur le calcul des teintes que la polarisation d<span>é</span>veloppe dans leslames cristallis<span>é</span>es. Annales de Chimie et de Physique 17:102–111.
</div>
<div id="ref-fridman2019" class="csl-entry" role="listitem">
Fridman, L. 2019. <a href="https://youtu.be/O5xeyoRL95U?t=115">Deep learning basics</a>. Lecture.
</div>
<div id="ref-fu2020winter" class="csl-entry" role="listitem">
Fu, Y., G. Yang, Z. Li, X. Song, Z. Li, X. Xu, P. Wang, and C. Zhao. 2020. <a href="https://doi.org/10.3390/rs12223778">Winter wheat nitrogen status estimation using UAV-based RGB imagery and gaussian processes regression</a>. Remote Sensing 12:3778.
</div>
<div id="ref-gao2009atmospheric" class="csl-entry" role="listitem">
Gao, B.-C., M. J. Montes, C. O. Davis, and A. F. Goetz. 2009. <a href="https://doi.org/10.1016/j.rse.2007.12.015">Atmospheric correction algorithms for hyperspectral remote sensing data of land and ocean</a>. Remote Sensing of Environment 113:S17–S24.
</div>
<div id="ref-gauss1809theoria" class="csl-entry" role="listitem">
Gauss, C. F. 1809. Theoria motus corporum coelestium in sectionibus conicis solem ambientium auctore carolo friderico gauss. Sumtibus Frid. Perthes et IH Besser.
</div>
<div id="ref-geladi1986partial" class="csl-entry" role="listitem">
Geladi, P., and B. R. Kowalski. 1986. <a href="https://doi.org/10.1016/0003-2670(86)80028-9">Partial least-squares regression: A tutorial</a>. Analytica chimica acta 185:1–17.
</div>
<div id="ref-gewali2018machine" class="csl-entry" role="listitem">
Gewali, U. B., S. T. Monteiro, and E. Saber. 2018. <a href="https://doi.org/10.48550/arXiv.1802.08701">Machine learning based hyperspectral image analysis: A survey</a>. arXiv.
</div>
<div id="ref-gitelson2004wide" class="csl-entry" role="listitem">
Gitelson, A. A. 2004. <a href="https://doi.org/10.1078/0176-1617-01176">Wide dynamic range vegetation index for remote quantification of biophysical characteristics of vegetation</a>. Journal of Plant Physiology 161:165–173.
</div>
<div id="ref-gitelson2003relationships" class="csl-entry" role="listitem">
Gitelson, A. A., Y. Gritz, and M. N. Merzlyak. 2003. <a href="https://doi.org/10.1078/0176-1617-00887">Relationships between leaf chlorophyll content and spectral reflectance and algorithms for non-destructive chlorophyll assessment in higher plant leaves</a>. Journal of Plant Physiology 160:271–282.
</div>
<div id="ref-gitelson1996use" class="csl-entry" role="listitem">
Gitelson, A. A., Y. J. Kaufman, and M. N. Merzlyak. 1996. <a href="https://doi.org/10.1016/S0034-4257(96)00072-7">Use of a green channel in remote sensing of global vegetation from EOS-MODIS</a>. Remote Sensing of Environment 58:289–298.
</div>
<div id="ref-expint" class="csl-entry" role="listitem">
Goulet, V. 2016. <a href="https://cran.r-project.org/package=expint">Expint: Exponential integral and incomplete gamma function</a>.
</div>
<div id="ref-gramacy2020surrogates" class="csl-entry" role="listitem">
Gramacy, R. B. 2020. Surrogates: <span>G</span>aussian process modeling, design and &nbsp;optimization for the applied sciences. Chapman Hall/CRC, Boca Raton, Florida.
</div>
<div id="ref-grossman1996critique" class="csl-entry" role="listitem">
Grossman, Y., S. Ustin, S. Jacquemoud, E. Sanderson, G. Schmuck, and J. Verdebout. 1996. <a href="https://doi.org/10.1016/0034-4257(95)00235-9">Critique of stepwise multiple linear regression for the extraction of leaf biochemistry information from leaf reflectance data</a>. Remote Sensing of Environment 56:182–193.
</div>
<div id="ref-guerrero2021current" class="csl-entry" role="listitem">
Guerrero, A., S. De Neve, and A. M. Mouazen. 2021. <a href="https://doi.org/10.1016/bs.agron.2021.02.001">Current sensor technologies for in situ and on-line measurement of soil nitrogen for variable rate fertilization: A review</a>. Advances in Agronomy 168:1–38.
</div>
<div id="ref-guo2018review" class="csl-entry" role="listitem">
Guo, Y., Y. Liu, T. Georgiou, and M. S. Lew. 2018. <a href="https://doi.org/10.1007/s13735-017-0141-z">A review of semantic segmentation using deep neural networks</a>. International journal of multimedia information retrieval 7:87–93.
</div>
<div id="ref-guyot1989factors" class="csl-entry" role="listitem">
Guyot, G., D. Guyon, and J. Riom. 1989. <a href="https://doi.org/10.1080/10106048909354217">Factors affecting the spectral response of forest canopies: A review</a>. Geocarto International 4:3–18.
</div>
<div id="ref-haboudane2002integrated" class="csl-entry" role="listitem">
Haboudane, D., J. R. Miller, N. Tremblay, P. J. Zarco-Tejada, and L. Dextraze. 2002. <a href="https://doi.org/10.1016/S0034-4257(02)00018-4">Integrated narrow-band vegetation indices for prediction of crop chlorophyll content for application to precision agriculture</a>. Remote Sensing of Environment 81:416–426.
</div>
<div id="ref-hagen2013review" class="csl-entry" role="listitem">
Hagen, N. A., and M. W. Kudenov. 2013. <a href="https://doi.org/10.1117/1.OE.52.9.090901">Review of snapshot spectral imaging technologies</a>. Optical Engineering 52:090901.
</div>
<div id="ref-hank2019spaceborne" class="csl-entry" role="listitem">
Hank, T. B., K. Berger, H. Bach, J. G. Clevers, A. Gitelson, P. Zarco-Tejada, and W. Mauser. 2019. <a href="https://doi.org/10.1007/s10712-018-9492-0">Spaceborne imaging spectroscopy for sustainable agriculture: Contributions and challenges</a>. Surveys in Geophysics 40:515–551.
</div>
<div id="ref-hansen2003reflectance" class="csl-entry" role="listitem">
Hansen, P., and J. Schjoerring. 2003. <a href="https://doi.org/10.1016/S0034-4257(03)00131-7">Reflectance measurement of canopy biomass and nitrogen status in wheat crops using normalized difference vegetation indices and partial least squares regression</a>. Remote Sensing of Environment 86:542–553.
</div>
<div id="ref-hastie2017extended" class="csl-entry" role="listitem">
Hastie, T., R. Tibshirani, and R. J. Tibshirani. 2017. <a href="https://doi.org/10.48550/arXiv.1707.08692">Extended comparisons of best subset selection, forward stepwise selection, and the lasso</a>. arXiv preprint arXiv:1707.08692.
</div>
<div id="ref-hawkesford2012functions" class="csl-entry" role="listitem">
Hawkesford, M., W. Horst, T. Kichey, H. Lambers, J. Schjoerring, I. S. Møller, and P. White. 2012. <a href="https://doi.org/10.1016/B978-0-12-384905-2.00006-6">Functions of macronutrients</a>. Pages 135–189 Marschner’s mineral nutrition of higher plants. Elsevier.
</div>
<div id="ref-herrmann2010swir" class="csl-entry" role="listitem">
Herrmann, I., A. Karnieli, D. Bonfil, Y. Cohen, and V. Alchanatis. 2010. <a href="https://doi.org/10.1080/01431160903283892">SWIR-based spectral indices for assessing nitrogen content in potato fields</a>. International Journal of Remote Sensing 31:5127–5143.
</div>
<div id="ref-holthaus2015" class="csl-entry" role="listitem">
Holthaus, E. 2015. 10 percent of california’s water goes to almond farming. That’s nuts.’. Slate Magazine.
</div>
<div id="ref-homolova2013review" class="csl-entry" role="listitem">
Homolová, L., Z. Malenovskỳ, J. G. Clevers, G. Garcı́a-Santos, and M. E. Schaepman. 2013. <a href="https://doi.org/10.1016/j.ecocom.2013.06.003">Review of optical-based remote sensing for plant trait mapping</a>. Ecological Complexity 15:1–16.
</div>
<div id="ref-hosgood1995leaf" class="csl-entry" role="listitem">
Hosgood, B., S. Jacquemoud, G. Andreoli, J. Verdebout, G. Pedrini, and G. Schmuck. 1995. Leaf optical properties experiment 93 (LOPEX93). Report EUR 16095.
</div>
<div id="ref-huete1988soil" class="csl-entry" role="listitem">
Huete, A. R. 1988. <a href="https://doi.org/10.1016/0034-4257(88)90106-X">A soil-adjusted vegetation index (SAVI)</a>. Remote Sensing of Environment 25:295–309.
</div>
<div id="ref-huete2002overview" class="csl-entry" role="listitem">
Huete, A., K. Didan, T. Miura, E. P. Rodriguez, X. Gao, and L. G. Ferreira. 2002. <a href="https://doi.org/10.1016/S0034-4257(02)00096-2">Overview of the radiometric and biophysical performance of the MODIS vegetation indices</a>. Remote Sensing of Environment 83:195–213.
</div>
<div id="ref-hunt2013visible" class="csl-entry" role="listitem">
Hunt Jr, E. R., P. C. Doraiswamy, J. E. McMurtrey, C. S. Daughtry, E. M. Perry, and B. Akhmedov. 2013. <a href="https://doi.org/10.1016/j.jag.2012.07.020">A visible band index for remote sensing leaf chlorophyll content at the canopy scale</a>. International journal of applied earth observation and geoinformation 21:103–112.
</div>
<div id="ref-ihuoma2017recent" class="csl-entry" role="listitem">
Ihuoma, S. O., and C. A. Madramootoo. 2017. <a href="https://doi.org/10.1016/j.compag.2017.07.026">Recent advances in crop water stress detection</a>. Computers and Electronics in Agriculture 141:267–275.
</div>
<div id="ref-inoue2012diagnostic" class="csl-entry" role="listitem">
Inoue, Y., E. Sakaiya, Y. Zhu, and W. Takahashi. 2012. <a href="https://doi.org/10.1016/j.rse.2012.08.026">Diagnostic mapping of canopy nitrogen content in rice based on hyperspectral measurements</a>. Remote Sensing of Environment 126:210–221.
</div>
<div id="ref-ioffe2015batch" class="csl-entry" role="listitem">
Ioffe, S., and C. Szegedy. 2015. Batch normalization: Accelerating deep network training by reducing internal covariate shift. Pages 448–456 International conference on machine learning. pmlr.
</div>
<div id="ref-ISO9288" class="csl-entry" role="listitem">
ISO. 2022. Thermal insulation — heat transfer by radiation — vocabulary. Standard, International Organization for Standardization, Geneva, CH.
</div>
<div id="ref-jackson1981canopy" class="csl-entry" role="listitem">
Jackson, R. D., S. Idso, R. Reginato, and P. Pinter Jr. 1981. <a href="https://doi.org/10.1029/WR017i004p01133">Canopy temperature as a crop water stress indicator</a>. Water resources research 17:1133–1138.
</div>
<div id="ref-jacquemoud1992utilisation" class="csl-entry" role="listitem">
Jacquemoud, S. 1992. Utilisation de la haute resolution spectrale pour l’etude des couverts vegetaux: Developpement d’un modele de reflectance spectrale. PhD thesis, Universit<span>é</span> Paris Diderot-Paris 7.
</div>
<div id="ref-jacquemoud2000comparison" class="csl-entry" role="listitem">
Jacquemoud, S., C. Bacour, H. Poilve, and J.-P. Frangi. 2000. <a href="https://doi.org/10.1016/S0034-4257(00)00139-5">Comparison of four radiative transfer models to simulate plant canopies reflectance: Direct and inverse mode</a>. Remote Sensing of Environment 74:471–481.
</div>
<div id="ref-jacquemoud1990prospect" class="csl-entry" role="listitem">
Jacquemoud, S., and F. Baret. 1990. <a href="https://doi.org/10.1016/0034-4257(90)90100-Z">PROSPECT: A model of leaf optical properties spectra</a>. Remote Sensing of Environment 34:75–91.
</div>
<div id="ref-jacquemoud1996estimating" class="csl-entry" role="listitem">
Jacquemoud, S., S. Ustin, J. Verdebout, G. Schmuck, G. Andreoli, and B. Hosgood. 1996. <a href="https://doi.org/10.1016/0034-4257(95)00238-3">Estimating leaf biochemistry using the PROSPECT leaf optical properties model</a>. Remote Sensing of Environment 56:194–202.
</div>
<div id="ref-jacquemoud2009prospect" class="csl-entry" role="listitem">
Jacquemoud, S., W. Verhoef, F. Baret, C. Bacour, P. Zarco-Tejada, G. Asner, C. François, and S. Ustin. 2009. <a href="https://doi.org/10.1016/j.rse.2008.01.026">PROSPECT + SAIL models: A review of use for vegetation characterization</a>. Remote Sensing of Environment 113:S56–S66.
</div>
<div id="ref-jafarbiglu2022comprehensive" class="csl-entry" role="listitem">
Jafarbiglu, H., and A. Pourreza. 2022. <a href="https://doi.org/10.1016/j.compag.2022.106844">A comprehensive review of remote sensing platforms, sensors, and applications in nut crops</a>. Computers and Electronics in Agriculture 197:106844.
</div>
<div id="ref-jafarbiglu2023impact" class="csl-entry" role="listitem">
Jafarbiglu, H., and A. Pourreza. 2023. <a href="https://doi.org/10.1016/j.isprsjprs.2022.12.002">Impact of sun-view geometry on canopy spectral reflectance variability</a>. ISPRS Journal of Photogrammetry and Remote Sensing 196:270–286.
</div>
<div id="ref-james2013introduction" class="csl-entry" role="listitem">
James, G., D. Witten, T. Hastie, and R. Tibshirani. 2013. <a href="https://doi.org/10.1007/978-1-0716-1418-1">An introduction to statistical learning</a>. Springer.
</div>
<div id="ref-LessUserManual2022" class="csl-entry" role="listitem">
Jianbo, Q. 2022. LESS user’s manual. 2.0 edition.
</div>
<div id="ref-jin2021combining" class="csl-entry" role="listitem">
Jin, D., J. Qi, H. Huang, and L. Li. 2021. <a href="https://doi.org/10.1109/JSTARS.2021.3122509">Combining 3D radiative transfer model and convolutional neural network to accurately estimate forest canopy cover from very high-resolution satellite images</a>. IEEE Journal of Selected Topics in Applied Earth Observations and Remote Sensing 14:10953–10963.
</div>
<div id="ref-jin2014newly" class="csl-entry" role="listitem">
Jin, X., Z. Li, H. Feng, X. Xu, and G. Yang. 2014. <a href="https://doi.org/10.1109/JSTARS.2014.2360069">Newly combined spectral indices to improve estimation of total leaf chlorophyll content in cotton</a>. IEEE Journal of Selected Topics in Applied Earth Observations and Remote Sensing 7:4589–4600.
</div>
<div id="ref-jin2020advancing" class="csl-entry" role="listitem">
Jin, Y., B. Chen, B. D. Lampinen, and P. H. Brown. 2020. <a href="https://doi.org/10.3389/fpls.2020.00290">Advancing agricultural production with machine learning analytics: Yield determinants for california’s almond orchards</a>. Frontiers in Plant Science 11:290.
</div>
<div id="ref-johnson2016crop" class="csl-entry" role="listitem">
Johnson, M. D., W. W. Hsieh, A. J. Cannon, A. Davidson, and F. Bédard. 2016. Crop yield forecasting on the canadian prairies by remotely sensed vegetation indices and machine learning methods. Agricultural and forest meteorology 218:74–84.
</div>
<div id="ref-jordan1969derivation" class="csl-entry" role="listitem">
Jordan, C. F. 1969. <a href="https://doi.org/10.2307/1936256">Derivation of leaf-area index from quality of light on the forest floor</a>. Ecology 50:663–666.
</div>
<div id="ref-jordan2015machine" class="csl-entry" role="listitem">
Jordan, M. I., and T. M. Mitchell. 2015. <a href="https://doi.org/10.1126/science.aaa8415">Machine learning: Trends, perspectives, and prospects</a>. Science 349:255–260.
</div>
<div id="ref-jumper2021highly" class="csl-entry" role="listitem">
Jumper, J., R. Evans, A. Pritzel, T. Green, M. Figurnov, O. Ronneberger, K. Tunyasuvunakool, R. Bates, A. Žı́dek, A. Potapenko, and others. 2021. <a href="https://doi.org/10.1038/s41586-021-03819-2">Highly accurate protein structure prediction with AlphaFold</a>. Nature 596:583–589.
</div>
<div id="ref-justes1994determination" class="csl-entry" role="listitem">
Justes, E., B. Mary, J.-M. Meynard, J.-M. Machet, and L. Thelier-Huché. 1994. <a href="https://doi.org/10.1006/anbo.1994.1133">Determination of a critical nitrogen dilution curve for winter wheat crops</a>. Annals of botany 74:397–407.
</div>
<div id="ref-kapoor1997nitrate" class="csl-entry" role="listitem">
Kapoor, A., and T. Viraraghavan. 1997. <a href="https://doi.org/10.1061/(ASCE)0733-9372(1997)123:4(371)">Nitrate removal from drinking water</a>. Journal of environmental engineering 123:371–380.
</div>
<div id="ref-kaufman1992atmospherically" class="csl-entry" role="listitem">
Kaufman, Y. J., and D. Tanre. 1992. <a href="https://doi.org/10.1109/36.134076">Atmospherically resistant vegetation index (ARVI) for EOS-MODIS</a>. IEEE Transactions on Geoscience and Remote Sensing 30:261–270.
</div>
<div id="ref-khalsa2018understanding" class="csl-entry" role="listitem">
Khalsa, S. D. S., and P. H. Brown. 2018. <a href="https://doi.org/10.17660/ActaHortic.2019.1253.28">Understanding nitrogen cycling in an irrigated deciduous permanent crop</a>. Pages 207–212 XXX international horticultural congress IHC2018: International symposium on water and nutrient relations and management of 1253.
</div>
<div id="ref-kimes2000inversion" class="csl-entry" role="listitem">
Kimes, D., Y. Knyazikhin, J. Privette, A. Abuelgasim, and F. Gao. 2000. <a href="https://doi.org/10.1080/02757250009532396">Inversion methods for physically-based models</a>. Remote Sensing Reviews 18:381–439.
</div>
<div id="ref-kokaly2009characterizing" class="csl-entry" role="listitem">
Kokaly, R. F., G. P. Asner, S. V. Ollinger, M. E. Martin, and C. A. Wessman. 2009. <a href="https://doi.org/10.1016/j.rse.2008.10.018">Characterizing canopy biochemistry from imaging spectroscopy and its application to ecosystem studies</a>. Remote Sensing of Environment 113:S78–S91.
</div>
<div id="ref-kuhn2022caret" class="csl-entry" role="listitem">
Kuhn, M. 2022. <a href="https://CRAN.R-project.org/package=caret">Caret: Classification and regression training</a>.
</div>
<div id="ref-kuhn2023" class="csl-entry" role="listitem">
Kuhn, M., and R. Quinlan. 2023. <a href="https://CRAN.R-project.org/package=Cubist">Cubist: Rule- and instance-based regression modeling</a>.
</div>
<div id="ref-lambert1760photometria" class="csl-entry" role="listitem">
Lambert, J. H. 1760. Photometria sive de mensura et gradibus luminis, colorum et umbrae. Sumptibus vidvae E. Klett, typis CP Detleffsen.
</div>
<div id="ref-legendre1806nouvelles" class="csl-entry" role="listitem">
Legendre, A. M. 1806. Nouvelles m<span>é</span>thodes pour la d<span>é</span>termination des orbites des com<span>è</span>tes; par AM legendre... chez Firmin Didot, libraire pour lew mathematiques, la marine, l&nbsp;<span>…</span>.
</div>
<div id="ref-li2018factors" class="csl-entry" role="listitem">
Li, Y., N. He, J. Hou, L. Xu, C. Liu, J. Zhang, Q. Wang, X. Zhang, and X. Wu. 2018. <a href="https://doi.org/10.3389/fevo.2018.00064">Factors influencing leaf chlorophyll content in natural forests at the biome scale</a>. Frontiers in Ecology and Evolution 6:64.
</div>
<div id="ref-liaw2002randomForest" class="csl-entry" role="listitem">
Liaw, A., and M. Wiener. 2002. <a href="https://CRAN.R-project.org/doc/Rnews/">Classification and regression by randomForest</a>. R News 2:18–22.
</div>
<div id="ref-liland2022pls" class="csl-entry" role="listitem">
Liland, K. H., B.-H. Mevik, and R. Wehrens. 2022. <a href="https://CRAN.R-project.org/package=pls">Pls: Partial least squares and principal component regression</a>.
</div>
<div id="ref-lin2022generalized" class="csl-entry" role="listitem">
Lin, D., M. Kettunen, B. Bitterli, J. Pantaleoni, C. Yuksel, and C. Wyman. 2022. <a href="https://doi.org/3528223.3530158">Generalized resampled importance sampling: Foundations of ReSTIR</a>. ACM Transactions on Graphics (TOG) 41:1–23.
</div>
<div id="ref-liu2022partial" class="csl-entry" role="listitem">
Liu, C., X. Zhang, T. T. Nguyen, J. Liu, T. Wu, E. Lee, and X. M. Tu. 2022. <a href="https://doi.org/10.1136/gpsych-2021-100662">Partial least squares regression and principal component analysis: Similarity and differences between two popular variable reduction approaches</a>. General Psychiatry 35.
</div>
<div id="ref-malenovsky2013retrieval" class="csl-entry" role="listitem">
Malenovskỳ, Z., L. Homolová, R. Zurita-Milla, P. Lukeš, V. Kaplan, J. Hanuš, J.-P. Gastellu-Etchegorry, and M. E. Schaepman. 2013. <a href="https://doi.org/10.1016/j.rse.2012.12.015">Retrieval of spruce leaf chlorophyll content from airborne image data using continuum removal and radiative transfer</a>. Remote Sensing of Environment 131:85–102.
</div>
<div id="ref-markov1912wahrscheinlichkeitsrechnung" class="csl-entry" role="listitem">
Markov, A. A. 1912. Wahrscheinlichkeitsrechnung. BG Teubner.
</div>
<div id="ref-mcculloch1943logical" class="csl-entry" role="listitem">
McCulloch, W. S., and W. Pitts. 1943. A logical calculus of the ideas immanent in nervous activity. The bulletin of mathematical biophysics 5:115–133.
</div>
<div id="ref-mehmood2016diversity" class="csl-entry" role="listitem">
Mehmood, T., and B. Ahmed. 2016. <a href="https://doi.org/10.1002/cem.2762">The diversity in the applications of partial least squares: An overview</a>. Journal of Chemometrics 30:4–17.
</div>
<div id="ref-micke1996" class="csl-entry" role="listitem">
Micke, W. 1996. Almond production manual. Regents of the University of California.
</div>
<div id="ref-moghimi2020novel" class="csl-entry" role="listitem">
Moghimi, A., A. Pourreza, G. Zuniga-Ramirez, L. E. Williams, and M. W. Fidelibus. 2020. <a href="https://doi.org/10.3390/rs12213515">A novel machine learning approach to estimate grapevine leaf nitrogen concentration using aerial multispectral imagery</a>. Remote Sensing 12:3515.
</div>
<div id="ref-mount2016water" class="csl-entry" role="listitem">
Mount, J., and E. Hanak. 2016. Water use in california. Public Policy Institute of California, San Francisco, California 2.
</div>
<div id="ref-munoz2013review" class="csl-entry" role="listitem">
Muñoz-Huerta, R. F., R. G. Guevara-Gonzalez, L. M. Contreras-Medina, I. Torres-Pacheco, J. Prado-Olivarez, and R. V. Ocampo-Velazquez. 2013. <a href="https://doi.org/10.3390/s130810823">A review of methods for sensing the nitrogen status in plants: Advantages, disadvantages and recent advances</a>. sensors 13:10823–10843.
</div>
<div id="ref-NREL" class="csl-entry" role="listitem">
National Renewable Energy Laboratory. 2022. <a href="https://www.nrel.gov/grid/solar-resource/spectra-astm-e490.html">2000 ASTM standard extraterrestrial spectrum reference e-490-00</a>. Webpage.
</div>
<div id="ref-nawi2013effect" class="csl-entry" role="listitem">
Nawi, N. M., W. H. Atomi, and M. Z. Rehman. 2013. The effect of data pre-processing on optimized training of artificial neural networks. Procedia Technology 11:32–39.
</div>
<div id="ref-stockfish2023" class="csl-entry" role="listitem">
Noda, T. R. A. M. C. A. J. K. A. G. L. A. Y. N. A. M. I. A. H. 2023. Stockfish. https://github.com/official-stockfish/Stockfish; GitHub.
</div>
<div id="ref-ohyama2010nitrogen" class="csl-entry" role="listitem">
Ohyama, T. 2010. Nitrogen as a major essential element of plants. Nitrogen assimilation in plants 37:2–17.
</div>
<div id="ref-omidi2020ensemble" class="csl-entry" role="listitem">
Omidi, R., A. Moghimi, A. Pourreza, M. El-Hadedy, and A. S. Eddin. 2020. <a href="https://doi.org/10.1109/ICMLA51294.2020.00054">Ensemble hyperspectral band selection for detecting nitrogen status in grape leaves</a>. IEEE.
</div>
<div id="ref-openai2023" class="csl-entry" role="listitem">
OpenAI. 2023. <a href="https://cdn.openai.com/papers/gpt-4.pdf">GPT-4 technical report</a>. OpenAI.
</div>
<div id="ref-padilla2018different" class="csl-entry" role="listitem">
Padilla, F. M., R. de Souza, M. T. Peña-Fleitas, M. Gallardo, C. Gimenez, and R. B. Thompson. 2018. <a href="https://doi.org/10.3389/fpls.2018.01752">Different responses of various chlorophyll meters to increasing nitrogen supply in sweet pepper</a>. Frontiers in plant science 9:1752.
</div>
<div id="ref-pancorbo2021simultaneous" class="csl-entry" role="listitem">
Pancorbo, J., C. Camino, M. Alonso-Ayuso, M. Raya-Sereno, I. Gonzalez-Fernandez, J. L. Gabriel, P. J. Zarco-Tejada, and M. Quemada. 2021. <a href="https://doi.org/10.1016/j.eja.2021.126287">Simultaneous assessment of nitrogen and water status in winter wheat using hyperspectral and thermal sensors</a>. European Journal of Agronomy 127:126287.
</div>
<div id="ref-pathak2018climate" class="csl-entry" role="listitem">
Pathak, T. B., M. L. Maskey, J. A. Dahlberg, F. Kearns, K. M. Bali, and D. Zaccaria. 2018. <a href="https://doi.org/10.3390/agronomy8030025">Climate change trends and impacts on california agriculture: A detailed review</a>. Agronomy 8:25.
</div>
<div id="ref-peddie2019ray" class="csl-entry" role="listitem">
Peddie, J. 2019. <a href="https://doi.org/10.1007/978-3-030-17490-3">Ray tracing: A tool for all</a>. Springer.
</div>
<div id="ref-penuelas1994reflectance" class="csl-entry" role="listitem">
Peñuelas, J., J. Gamon, A. Fredeen, J. Merino, and C. Field. 1994. <a href="https://doi.org/10.1016/0034-4257(94)90136-8">Reflectance indices associated with physiological changes in nitrogen-and water-limited sunflower leaves</a>. Remote Sensing of Environment 48:135–146.
</div>
<div id="ref-perich2021proximal" class="csl-entry" role="listitem">
Perich, G., P. Meyer, A. Wieser, and F. Liebisch. 2021. <a href="https://doi.org/10.1109/WHISPERS52202.2021.9484031">Proximal and drone based hyperspectral sensing for crop nitrogen status detection in historic field trials</a>. Pages 1–5 2021 11th workshop on hyperspectral imaging and signal processing: Evolution in remote sensing (WHISPERS). IEEE.
</div>
<div id="ref-pierce1999aspects" class="csl-entry" role="listitem">
Pierce, F. J., and P. Nowak. 1999. <a href="https://doi.org/10.1016/S0065-2113(08)60513-1">Aspects of precision agriculture</a>. Advances in Agronomy 67:1–85.
</div>
<div id="ref-plackett1949historical" class="csl-entry" role="listitem">
Plackett, R. L. 1949. <a href="https://doi.org/10.2307/2332682">A historical note on the method of least squares</a>. Biometrika 36:458–460.
</div>
<div id="ref-QGIS" class="csl-entry" role="listitem">
QGIS Development Team. 2022. <a href="https://www.qgis.org">QGIS geographic information system</a>. QGIS Association.
</div>
<div id="ref-qi1994modified" class="csl-entry" role="listitem">
Qi, J., A. Chehbouni, A. R. Huete, Y. H. Kerr, and S. Sorooshian. 1994. <a href="https://doi.org/10.1016/0034-4257(94)90134-1">A modified soil adjusted vegetation index</a>. Remote Sensing of Environment 48:119–126.
</div>
<div id="ref-qi2023fast" class="csl-entry" role="listitem">
Qi, J., J. Jiang, K. Zhou, D. Xie, and H. Huang. 2023. <a href="https://doi.org/10.34133/remotesensing.0017">Fast and accurate simulation of canopy reflectance under wavelength-dependent optical properties using a semi-empirical 3D radiative transfer model</a>. Journal of Remote Sensing 3:0017.
</div>
<div id="ref-qi2017large" class="csl-entry" role="listitem">
Qi, J., D. Xie, D. Guo, and G. Yan. 2017. <a href="https://doi.org/10.1109/JSTARS.2017.2714423">A large-scale emulation system for realistic three-dimensional (3-d) forest simulation</a>. IEEE Journal of Selected Topics in Applied Earth Observations and Remote Sensing 10:4834–4843.
</div>
<div id="ref-qi20223d" class="csl-entry" role="listitem">
Qi, J., D. Xie, J. Jiang, and H. Huang. 2022. <a href="https://doi.org/10.1016/j.rse.2022.113301">3D radiative transfer modeling of structurally complex forest canopies through a lightweight boundary-based description of leaf clusters</a>. Remote Sensing of Environment 283:113301.
</div>
<div id="ref-qi2019less" class="csl-entry" role="listitem">
Qi, J., D. Xie, T. Yin, G. Yan, J.-P. Gastellu-Etchegorry, L. Li, W. Zhang, X. Mu, and L. K. Norford. 2019. <a href="https://doi.org/10.1016/j.rse.2018.11.036">LESS: LargE-scale remote sensing data and image simulation framework over heterogeneous 3D scenes</a>. Remote Sensing of Environment 221:695–706.
</div>
<div id="ref-R2022" class="csl-entry" role="listitem">
R Core Team. 2022. <a href="https://www.R-project.org/">R: A language and environment for statistical computing</a>. R Foundation for Statistical Computing, Vienna, Austria.
</div>
<div id="ref-raun1999fertilizer" class="csl-entry" role="listitem">
Raun, W. R., G. Johnson, and R. Westerman. 1999. <a href="https://doi.org/10.2136/sssaj1999.03615995006300030030x">Fertilizer nitrogen recovery in long-term continuous winter wheat</a>. Soil Science Society of America Journal 63:645–650.
</div>
<div id="ref-ravikanth2017extraction" class="csl-entry" role="listitem">
Ravikanth, L., D. S. Jayas, N. D. White, P. G. Fields, and D.-W. Sun. 2017. <a href="https://doi.org/10.1007/s11947-016-1817-8">Extraction of spectral information from hyperspectral data and application of hyperspectral imaging for food and agricultural products</a>. Food and bioprocess technology 10:1–33.
</div>
<div id="ref-ren1996approach" class="csl-entry" role="listitem">
Ren-hua, Z., N. Rao, and K. Liao. 1996. Approach for a vegetation index resistant to atmospheric effect. Journal of Integrative Plant Biology 38.
</div>
<div id="ref-richardson1977distinguishing" class="csl-entry" role="listitem">
Richardson, A. J., and C. Wiegand. 1977. Distinguishing vegetation from soil background information. Photogrammetric engineering and remote sensing 43:1541–1552.
</div>
<div id="ref-Rohde2013" class="csl-entry" role="listitem">
Rohde, R. A. 2013. <a href="https://commons.wikimedia.org/wiki/File:Solar_Spectrum.png#/media/File:Solar_spectrum_en.svg">Spectrum of solar radiation</a>. Webpage.
</div>
<div id="ref-rombach2021highresolution" class="csl-entry" role="listitem">
Rombach, R., A. Blattmann, D. Lorenz, P. Esser, and B. Ommer. 2021. <a href="https://arxiv.org/abs/2112.10752">High-resolution image synthesis with latent diffusion models</a>.
</div>
<div id="ref-rosenblatt1958perceptron" class="csl-entry" role="listitem">
Rosenblatt, F. 1958. <a href="https://doi.org/10.1037/h0042519">The perceptron: A probabilistic model for information storage and organization in the brain.</a> Psychological review 65:386.
</div>
<div id="ref-rosipal2006overview" class="csl-entry" role="listitem">
Rosipal, R., and N. Krämer. 2006. <a href="https://doi.org/10.1007/11752790_2">Overview and recent advances in partial least squares</a>. Pages 34–51 Subspace, latent structure and feature selection: Statistical and optimization perspectives workshop, SLSFS 2005, bohinj, slovenia, february 23-25, 2005, revised selected papers. Springer.
</div>
<div id="ref-rouse1973monitoring" class="csl-entry" role="listitem">
Rouse Jr, J. W., R. H. Haas, J. Schell, and D. Deering. 1973. <a href="https://ntrs.nasa.gov/citations/19740022555">Monitoring the vernal advancement and retrogradation (green wave effect) of natural vegetation</a>. Texas A&amp;M University Remote Sensing Center.
</div>
<div id="ref-rudin2019stop" class="csl-entry" role="listitem">
Rudin, C. 2019. <a href="https://doi.org/10.1038/s42256-019-0048-x">Stop explaining black box machine learning models for high stakes decisions and use interpretable models instead</a>. Nature machine intelligence 1:206–215.
</div>
<div id="ref-saez2013overview" class="csl-entry" role="listitem">
Sáez-Plaza, P., M. J. Navas, S. Wybraniec, T. Michałowski, and A. G. Asuero. 2013. <a href="https://doi.org/10.1080/10408347.2012.751787">An overview of the kjeldahl method of nitrogen determination. Part II. Sample preparation, working scale, instrumental finish, and quality control</a>. Critical Reviews in Analytical Chemistry 43:224–272.
</div>
<div id="ref-samuel1959machine" class="csl-entry" role="listitem">
Samuel, A. L. 1959. Machine learning. The Technology Review 62:42–45.
</div>
<div id="ref-schirrmann2016monitoring" class="csl-entry" role="listitem">
Schirrmann, M., A. Giebel, F. Gleiniger, M. Pflanz, J. Lentschke, and K.-H. Dammer. 2016. <a href="https://doi.org/10.3390/rs8090706">Monitoring agronomic parameters of winter wheat crops with low-cost UAV imagery</a>. Remote Sensing 8:706.
</div>
<div id="ref-schowengerdt2006remote" class="csl-entry" role="listitem">
Schowengerdt, R. A. 2006. Remote sensing: Models and methods for image processing. Elsevier.
</div>
<div id="ref-selan2005using" class="csl-entry" role="listitem">
Selan, J. 2005. Using lookup tables to accelerate color transformations. GPU Gems 2:381–392.
</div>
<div id="ref-serrano2002remote" class="csl-entry" role="listitem">
Serrano, L., J. Penuelas, and S. L. Ustin. 2002. <a href="https://doi.org/10.1016/S0034-4257(02)00011-1">Remote sensing of nitrogen and lignin in mediterranean vegetation from AVIRIS data: Decomposing biochemical from structural signals</a>. Remote sensing of Environment 81:355–364.
</div>
<div id="ref-shiklomanov2021cutting" class="csl-entry" role="listitem">
Shiklomanov, A. N., M. C. Dietze, I. Fer, T. Viskari, and S. P. Serbin. 2021. <a href="https://doi.org/10.5194/gmd-14-2603-2021">Cutting out the middleman: Calibrating and validating a dynamic vegetation model (ED2-PROSPECT5) using remotely sensed surface reflectance</a>. Geoscientific Model Development 14:2603–2633.
</div>
<div id="ref-silver2017mastering" class="csl-entry" role="listitem">
Silver, D., J. Schrittwieser, K. Simonyan, I. Antonoglou, A. Huang, A. Guez, T. Hubert, L. Baker, M. Lai, A. Bolton, and others. 2017. <a href="https://doi.org/10.1038/nature24270">Mastering the game of go without human knowledge</a>. Nature 550:354–359.
</div>
<div id="ref-sishodia2020applications" class="csl-entry" role="listitem">
Sishodia, R. P., R. L. Ray, and S. K. Singh. 2020. <a href="https://doi.org/10.3390/rs12193136">Applications of remote sensing in precision agriculture: A review</a>. Remote Sensing 12:3136.
</div>
<div id="ref-smart2014nitrogen" class="csl-entry" role="listitem">
Smart, D., D. Schellenberg, S. Saa Silva, S. Muhammad, B. Sanden, and P. Brown. 2014. Nitrogen use efficiency of california almond orchards using advanced farming practices. Page 487 EGU general assembly conference abstracts.
</div>
<div id="ref-sonobe2018crop" class="csl-entry" role="listitem">
Sonobe, R., Y. Yamaya, H. Tani, X. Wang, N. Kobayashi, and K. Mochizuki. 2018. <a href="https://doi.org/10.1117/1.JRS.12.026019">Crop classification from sentinel-2-derived vegetation indices using ensemble learning</a>. Journal of Applied Remote Sensing 12:026019–026019.
</div>
<div id="ref-spafford2021spectral" class="csl-entry" role="listitem">
Spafford, L., G. Le Maire, A. MacDougall, F. De Boissieu, and J.-B. Feret. 2021. <a href="https://doi.org/10.1016/j.rse.2020.112176">Spectral subdomains and prior estimation of leaf structure improves PROSPECT inversion on reflectance or transmittance alone</a>. Remote Sensing of Environment 252:112176.
</div>
<div id="ref-stark2008continuing" class="csl-entry" role="listitem">
Stark, C. H., and K. G. Richards. 2008. The continuing challenge of agricultural nitrogen loss to the environment in the context of global change and advancing research. Dynamic Soil, Dynamic Plant 2:1–12.
</div>
<div id="ref-stern1964transmission" class="csl-entry" role="listitem">
Stern, F. 1964. <a href="https://doi.org/10.1364/AO.3.000111">Transmission of isotropic radiation across an interface between two dielectrics</a>. Applied Optics 3:111–113.
</div>
<div id="ref-stickler2016educational" class="csl-entry" role="listitem">
Stickler, G., and L. Kyle. 2016. Educational brief – solar radiation and the earth system. National Aeronautics and Space Administration.
</div>
<div id="ref-stone1974cross" class="csl-entry" role="listitem">
Stone, M. 1974. <a href="https://doi.org/10.1111/j.2517-6161.1974.tb00994.x">Cross-validatory choice and assessment of statistical predictions</a>. Journal of the royal statistical society: Series B (Methodological) 36:111–133.
</div>
<div id="ref-takayama2016optimal" class="csl-entry" role="listitem">
Takayama, T., and A. Iwasaki. 2016. <a href="https://doi.org/10.5194/isprsannals-III-8-101-2016">Optimal wavelength selection on hyperspectral data with fused lasso for biomass estimation of tropical rainforests</a>. ISPRS Annals of Photogrammetry, Remote Sensing &amp; Spatial Information Sciences 3.
</div>
<div id="ref-thenkabail2013selection" class="csl-entry" role="listitem">
Thenkabail, P. S., I. Mariotto, M. K. Gumma, E. M. Middleton, D. R. Landis, and K. F. Huemmrich. 2013. <a href="https://doi.org/10.1109/JSTARS.2013.2252601">Selection of hyperspectral narrowbands (HNBs) and composition of hyperspectral twoband vegetation indices (HVIs) for biophysical characterization and discrimination of crop types using field reflectance and hyperion/EO-1 data</a>. IEEE Journal of Selected Topics in Applied Earth Observations and Remote Sensing 6:427–439.
</div>
<div id="ref-thomas1972estimating" class="csl-entry" role="listitem">
Thomas, J. R., and G. F. Oerther. 1972. <a href="https://doi.org/10.2134/agronj1972.00021962006400010004x">Estimating nitrogen content of sweet pepper leaves by reflectance measurements</a>. Agronomy journal 64:11–13.
</div>
<div id="ref-tian2011assessing" class="csl-entry" role="listitem">
Tian, Y., X. Yao, J. Yang, W. Cao, D. Hannaway, and Y. Zhu. 2011. <a href="https://doi.org/10.1016/j.fcr.2010.11.002">Assessing newly developed and published vegetation indices for estimating rice leaf nitrogen concentration with ground-and space-based hyperspectral reflectance</a>. Field Crops Research 120:299–310.
</div>
<div id="ref-tibshirani1996" class="csl-entry" role="listitem">
Tibshirani, R. 1996. <a href="https://doi.org/10.1111/j.2517-6161.1996.tb02080.x">Regression shrinkage and selection via the lasso</a>. Journal of the Royal Statistical Society. Series B (Methodological) 58:267–288.
</div>
<div id="ref-almonds2021" class="csl-entry" role="listitem">
USDA. 2021. California almond acreage report. United States Department of Agriculture, Pacific Region, P.O. Box 1258, Sacramento, CA 95812.
</div>
<div id="ref-van2018occam" class="csl-entry" role="listitem">
Van den Berg, H. 2018. <a href="https://doi.org/10.3184/003685018X15295002645082">Occam’s razor: From ockham’s via moderna to modern data science</a>. Science progress 101:261–272.
</div>
<div id="ref-verger2011optimal" class="csl-entry" role="listitem">
Verger, A., F. Baret, and F. Camacho. 2011. <a href="https://doi.org/10.1016/j.rse.2010.09.012">Optimal modalities for radiative transfer-neural network estimation of canopy biophysical characteristics: Evaluation over an agricultural area with CHRIS/PROBA observations</a>. Remote Sensing of Environment 115:415–426.
</div>
<div id="ref-verhoef1984light" class="csl-entry" role="listitem">
Verhoef, W. 1984. <a href="https://doi.org/10.1016/0034-4257(84)90057-9">Light scattering by leaf layers with application to canopy reflectance modeling: The SAIL model</a>. Remote Sensing of Environment 16:125–141.
</div>
<div id="ref-verhoef1985earth" class="csl-entry" role="listitem">
Verhoef, W. 1985. <a href="https://doi.org/10.1016/0034-4257(85)90072-0">Earth observation modeling based on layer scattering matrices</a>. Remote Sensing of Environment 17:165–178.
</div>
<div id="ref-verhoef2007unified" class="csl-entry" role="listitem">
Verhoef, W., L. Jia, Q. Xiao, and Z. Su. 2007. <a href="https://doi.org/10.1109/TGRS.2007.895844">Unified optical-thermal four-stream radiative transfer theory for homogeneous vegetation canopies</a>. IEEE Transactions on geoscience and remote sensing 45:1808–1822.
</div>
<div id="ref-verrelst2016active" class="csl-entry" role="listitem">
Verrelst, J., S. Dethier, J. P. Rivera, J. Munoz-Mari, G. Camps-Valls, and J. Moreno. 2016. <a href="https://doi.org/10.1109/LGRS.2016.2560799">Active learning methods for efficient hybrid biophysical variable retrieval</a>. IEEE Geoscience and Remote Sensing Letters 13:1012–1016.
</div>
<div id="ref-verrelst2019quantifying" class="csl-entry" role="listitem">
Verrelst, J., Z. Malenovskỳ, C. Van der Tol, G. Camps-Valls, J.-P. Gastellu-Etchegorry, P. Lewis, P. North, and J. Moreno. 2019. <a href="https://doi.org/10.1007/s10712-018-9478-y">Quantifying vegetation biophysical variables from imaging spectroscopy data: A review on retrieval methods</a>. Surveys in Geophysics 40:589–629.
</div>
<div id="ref-verrelst2017global" class="csl-entry" role="listitem">
Verrelst, J., and J. Rivera. 2017. <a href="https://doi.org/10.1016/B978-0-12-803011-0.00016-1">A global sensitivity analysis toolbox to quantify drivers of vegetation radiative transfer models</a>. Pages 319–339 Sensitivity analysis in earth observation modelling. Elsevier.
</div>
<div id="ref-verrelst2015replacing" class="csl-entry" role="listitem">
Verrelst, J., J. P. Rivera, J. Gómez-Dans, G. Camps-Valls, and J. Moreno. 2015. <a href="https://doi.org/10.1109/IGARSS.2015.7325843">Replacing radiative transfer models by surrogate approximations through machine learning</a>. Pages 633–636 2015 IEEE international geoscience and remote sensing symposium (IGARSS). IEEE.
</div>
<div id="ref-nitrogen2012" class="csl-entry" role="listitem">
Viers, J. H., D. Liptzin, T. S. Rosenstock, V. B. Jensen, A. D. Hollander, A. McNally, A. M. King, G. Kourakos, E. M. Lopez, N. D. L. Mora, A. FryjoffHung, H. C. Kristin N. Dzurella, S. Laybourne, C. McKenney, J. Darby, J. F. Quinn, and T. Harter. 2012. Assessing nitrate in california’s drinking water with a focus on tulare lake basin and salinas valley groundwater. <span>Center for Watershed Sciences, University of California, Davis</span>.
</div>
<div id="ref-vineview" class="csl-entry" role="listitem">
VineView. 2022. <a href="https://vineview.com/data-products/water-index/">Water index</a>. Webpage.
</div>
<div id="ref-vinyals2019grandmaster" class="csl-entry" role="listitem">
Vinyals, O., I. Babuschkin, W. M. Czarnecki, M. Mathieu, A. Dudzik, J. Chung, D. H. Choi, R. Powell, T. Ewalds, P. Georgiev, and others. 2019. <a href="https://doi.org/10.1038/s41586-019-1724-z">Grandmaster level in StarCraft II using multi-agent reinforcement learning</a>. Nature 575:350–354.
</div>
<div id="ref-vitousek1997human" class="csl-entry" role="listitem">
Vitousek, P. M., J. D. Aber, R. W. Howarth, G. E. Likens, P. A. Matson, D. W. Schindler, W. H. Schlesinger, and D. G. Tilman. 1997. <a href="https://doi.org/10.1890/1051-0761(1997)007[0737:HAOTGN]2.0.CO;2">Human alteration of the global nitrogen cycle: Sources and consequences</a>. Ecological applications 7:737–750.
</div>
<div id="ref-wade2006z" class="csl-entry" role="listitem">
Wade, T., S. Sommer, and others. 2006. A to z GIS, an illustrated dictionary of geographic information systems. Esri Press.
</div>
<div id="ref-walter2017smart" class="csl-entry" role="listitem">
Walter, A., R. Finger, R. Huber, and N. Buchmann. 2017. <a href="https://doi.org/10.1073/pnas.1707462114">Smart farming is key to developing sustainable agriculture</a>. Proceedings of the National Academy of Sciences 114:6148–6150.
</div>
<div id="ref-wang2017non" class="csl-entry" role="listitem">
Wang, J., C. Shen, N. Liu, X. Jin, X. Fan, C. Dong, and Y. Xu. 2017. <a href="https://doi.org/10.3390/s17030538">Non-destructive evaluation of the leaf nitrogen concentration by in-field visible/near-infrared spectroscopy in pear orchards</a>. Sensors 17:538.
</div>
<div id="ref-webb2013assessment" class="csl-entry" role="listitem">
Webb, J., P. Sørensen, G. Velthof, B. Amon, M. Pinto, L. Rodhe, E. Salomon, N. Hutchings, P. Burczyk, and J. Reid. 2013. <a href="https://doi.org/10.1016/B978-0-12-407247-3.00007-X">An assessment of the variation of manure nitrogen efficiency throughout europe and an appraisal of means to increase manure-n efficiency</a>. Advances in Agronomy 119:371–442.
</div>
<div id="ref-weber2016extended" class="csl-entry" role="listitem">
Weber, A., M. Benecke, J. Wendler, A. Sieck, D. Hübner, H. Figgemeier, and R. Breiter. 2016. <a href="https://doi.org/10.1117/12.2223737">Extended SWIR imaging sensors for hyperspectral imaging applications</a>. Pages 42–56 Image sensing technologies: Materials, devices, systems, and applications III. SPIE.
</div>
<div id="ref-wheeler2011" class="csl-entry" role="listitem">
Wheeler, R. 2011. <a href="https://upload.wikimedia.org/wikipedia/commons/0/06/Leaf_Tissue_Structure.svg">Diagram of the internal structure of a leaf</a>. Webpage.
</div>
<div id="ref-almonds2016" class="csl-entry" role="listitem">
WIFSS. 2016. <a href="https://www.wifss.ucdavis.edu/wp-content/uploads/2016/10/Almonds_PDF.pdf">Almonds</a>. <span>Western Institute for Food Safety and Security</span>, 1477 Drew Ave, Davis, CA 95618.
</div>
<div id="ref-willett1988another" class="csl-entry" role="listitem">
Willett, J. B., and J. D. Singer. 1988. <a href="https://doi.org/10.1080/00031305.1988.10475573">Another cautionary note about r<span class="math inline">\(^{2}\)</span>: Its use in weighted least-squares regression analysis</a>. The American Statistician 42:236–238.
</div>
<div id="ref-wold1984collinearity" class="csl-entry" role="listitem">
Wold, S., A. Ruhe, H. Wold, and W. Dunn Iii. 1984. <a href="https://doi.org/10.1137/0905052">The collinearity problem in linear regression. The partial least squares (PLS) approach to generalized inverses</a>. SIAM Journal on Scientific and Statistical Computing 5:735–743.
</div>
<div id="ref-wolfert2017big" class="csl-entry" role="listitem">
Wolfert, S., L. Ge, C. Verdouw, and M.-J. Bogaardt. 2017. <a href="https://doi.org/10.1016/j.agsy.2017.01.023">Big data in smart farming–a review</a>. Agricultural Systems 153:69–80.
</div>
<div id="ref-wright2004worldwide" class="csl-entry" role="listitem">
Wright, I. J., P. B. Reich, M. Westoby, D. D. Ackerly, Z. Baruch, F. Bongers, J. Cavender-Bares, T. Chapin, J. H. Cornelissen, M. Diemer, and others. 2004. <a href="https://doi.org/10.1038/nature02403">The worldwide leaf economics spectrum</a>. Nature 428:821–827.
</div>
<div id="ref-wright1921correlation" class="csl-entry" role="listitem">
Wright, S. 1921. Correlation and causation. Journal of Agricultural Research 20:557–585.
</div>
<div id="ref-wyman2021rearchitecting" class="csl-entry" role="listitem">
Wyman, C., and A. Panteleev. 2021. Rearchitecting spatiotemporal resampling for production. Pages 23–41 Proceedings of the conference on high-performance graphics.
</div>
<div id="ref-xue2017significant" class="csl-entry" role="listitem">
Xue, J., and B. Su. 2017. <a href="https://doi.org/10.1155/2017/1353691">Significant remote sensing vegetation indices: A review of developments and applications</a>. Journal of sensors 2017:1–17.
</div>
<div id="ref-yan2009linear" class="csl-entry" role="listitem">
Yan, X., and X. Su. 2009. <a href="https://doi.org/10.1142/6986">Linear regression analysis: Theory and computing</a>. world scientific.
</div>
<div id="ref-yang2020spart" class="csl-entry" role="listitem">
Yang, P., C. van der Tol, T. Yin, and W. Verhoef. 2020. <a href="https://doi.org/10.1016/j.rse.2020.111870">The SPART model: A soil-plant-atmosphere radiative transfer model for satellite measurements in the solar spectrum</a>. Remote Sensing of Environment 247:111870.
</div>
<div id="ref-yang2008mechanisms" class="csl-entry" role="listitem">
Yang, X., X. Wu, H. Hao, and Z. He. 2008. <a href="https://doi.org/10.1631/jzus.B0710626">Mechanisms and assessment of water eutrophication</a>. Journal of zhejiang university Science B 9:197–209.
</div>
<div id="ref-yokoya2013hyperspectral" class="csl-entry" role="listitem">
Yokoya, N., and A. Iwasaki. 2013. Hyperspectral and multispectral data fusion mission on hyperspectral imager suite (HISUI). Pages 4086–4089 2013 IEEE international geoscience and remote sensing symposium-IGARSS. IEEE.
</div>
<div id="ref-zha2020improving" class="csl-entry" role="listitem">
Zha, H., Y. Miao, T. Wang, Y. Li, J. Zhang, W. Sun, Z. Feng, and K. Kusnierek. 2020. <a href="https://doi.org/10.3390/rs12020215">Improving unmanned aerial vehicle remote sensing-based rice nitrogen nutrition index prediction with machine learning</a>. Remote Sensing 12:215.
</div>
<div id="ref-zhang2002precision" class="csl-entry" role="listitem">
Zhang, N., M. Wang, and N. Wang. 2002. <a href="https://doi.org/10.1016/S0168-1699(02)00096-0">Precision agriculture—a worldwide overview</a>. Computers and electronics in agriculture 36:113–132.
</div>
<div id="ref-zhang2005estimating" class="csl-entry" role="listitem">
Zhang, Q., X. Xiao, B. Braswell, E. Linder, F. Baret, and B. Moore III. 2005. <a href="https://doi.org/10.1016/j.rse.2005.09.009">Estimating light absorption by chlorophyll, leaf and canopy in a deciduous broadleaf forest using MODIS data and a radiative transfer model</a>. Remote Sensing of Environment 99:357–371.
</div>
<div id="ref-zhang2019california" class="csl-entry" role="listitem">
Zhang, Z., Y. Jin, B. Chen, and P. Brown. 2019. <a href="https://doi.org/10.3389/fpls.2019.00809">California almond yield prediction at the orchard level with a machine learning approach</a>. Frontiers in plant science 10:809.
</div>
<div id="ref-zheng2009retrieving" class="csl-entry" role="listitem">
Zheng, G., and M. Moskal. 2009. <a href="https://doi.org/10.3390/s90402719">Retrieving leaf area index (LAI) using remote sensing: Theories, methods and sensors</a>. Sensors 9:2719–2745.
</div>
<div id="ref-zheng2018evaluation" class="csl-entry" role="listitem">
Zheng, H., T. Cheng, D. Li, X. Zhou, X. Yao, Y. Tian, W. Cao, and Y. Zhu. 2018. <a href="https://doi.org/10.3390/rs10060824">Evaluation of RGB, color-infrared and multispectral images acquired from unmanned aerial systems for the estimation of nitrogen accumulation in rice</a>. Remote Sensing 10:824.
</div>
<div id="ref-zhu2017deep" class="csl-entry" role="listitem">
Zhu, X. X., D. Tuia, L. Mou, G.-S. Xia, L. Zhang, F. Xu, and F. Fraundorfer. 2017. <a href="https://doi.org/10.1109/MGRS.2017.2762307">Deep learning in remote sensing: A comprehensive review and list of resources</a>. IEEE Geoscience and Remote Sensing Magazine 5:8–36.
</div>
<div id="ref-zou2005regularization" class="csl-entry" role="listitem">
Zou, H., and T. Hastie. 2005. <a href="https://doi.org/10.1111/j.1467-9868.2005.00503.x">Regularization and variable selection via the elastic net</a>. Journal of the royal statistical society: series B (statistical methodology) 67:301–320.
</div>
</div></section><section class="quarto-appendix-contents"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{oswald2023,
  author = {Oswald, Damian},
  title = {Remote {Sensing} of {Foliar} {Nitrogen} in {Californian}
    {Almonds}},
  date = {2023-04-28},
  langid = {en},
  abstract = {Nitrogen is a crucial plant nutrient, but both its
    under-availability and over-abundance can harm crop yields as well
    as the environment and reduce water supply quality. This is
    particularly important for almond orchards in California, where
    nitrogen is applied in large quantities due to the crop’s
    profitability. Remote sensing techniques show promise for
    quantifying foliar nitrogen concentration in crops, which can inform
    fertilization decisions and potentially reduce the risks of
    over-fertilization. In this thesis, I explore and test various
    remote sensing approaches for estimating the foliar nitrogen
    concentration in Californian almonds using spectroradiometric
    measurements at the leaf level and hyperspectral imaging at the
    canopy level. I evaluate numerous classical vegetation indices,
    various machine learning models, and two physics-informed radiative
    transfer models. Radiative transfer modelling, although less
    investigated in scientific literature, demonstrates potential as
    both a robust and generalizable technique, yielding results
    comparable to the best-performing machine learning models.}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-oswald2023" class="csl-entry quarto-appendix-citeas" role="listitem">
Oswald, D. 2023, April 28. Remote Sensing of Foliar Nitrogen in
Californian Almonds.
</div></div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>